<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Olink DS team" />

<meta name="date" content="2024-06-27" />

<title>Olink® Analyze Vignette</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Olink® Analyze Vignette</h1>
<h4 class="author">Olink DS team</h4>
<h4 class="date">2024-06-27</h4>


<div id="TOC">
<ul>
<li><a href="#installation" id="toc-installation">Installation</a></li>
<li><a href="#list-of-functions" id="toc-list-of-functions">List of
functions</a></li>
<li><a href="#usage" id="toc-usage">Usage</a>
<ul>
<li><a href="#load-the-library" id="toc-load-the-library">Load the
library</a></li>
<li><a href="#introduction-to-olink-npx-data-format" id="toc-introduction-to-olink-npx-data-format">Introduction to Olink NPX
data format</a></li>
</ul></li>
<li><a href="#preprocessing" id="toc-preprocessing">Preprocessing</a>
<ul>
<li><a href="#read-npx-data-read_npx" id="toc-read-npx-data-read_npx">Read NPX data (read_NPX)</a></li>
<li><a href="#randomize-samples-on-plate-olink_plate_randomizer" id="toc-randomize-samples-on-plate-olink_plate_randomizer">Randomize
samples on plate (olink_plate_randomizer)</a></li>
<li><a href="#select-bridge-samples-olink_bridgeselector" id="toc-select-bridge-samples-olink_bridgeselector">Select bridge
samples (olink_bridgeselector)</a></li>
<li><a href="#normalizing-npx-data-olink_normalization" id="toc-normalizing-npx-data-olink_normalization">Normalizing NPX data
(olink_normalization)</a></li>
<li><a href="#integrating-explore-npx-lod-olink_lod" id="toc-integrating-explore-npx-lod-olink_lod">Integrating Explore NPX
LOD (olink_lod)</a></li>
</ul></li>
<li><a href="#statistical-analysis" id="toc-statistical-analysis">Statistical analysis</a>
<ul>
<li><a href="#t-test-analysis-olink_ttest" id="toc-t-test-analysis-olink_ttest">T-test analysis
(olink_ttest)</a></li>
<li><a href="#mann-whitney-u-test-analysis-olink_wilcox" id="toc-mann-whitney-u-test-analysis-olink_wilcox">Mann-Whitney U Test
analysis (olink_wilcox)</a></li>
<li><a href="#analysis-for-variance-anova-olink_anova" id="toc-analysis-for-variance-anova-olink_anova">Analysis for variance
(ANOVA) (olink_anova)</a></li>
<li><a href="#post-hoc-anova-analysis-olink_anova_posthoc" id="toc-post-hoc-anova-analysis-olink_anova_posthoc">Post-hoc ANOVA
analysis (olink_anova_posthoc)</a></li>
<li><a href="#one-way-non-parametric-test-olink_one_non_parametric" id="toc-one-way-non-parametric-test-olink_one_non_parametric">One way
non-parametric test (olink_one_non_parametric)</a></li>
<li><a href="#post-hoc-one-way-non-parametric-analysis-olink_one_non_parametric_posthoc" id="toc-post-hoc-one-way-non-parametric-analysis-olink_one_non_parametric_posthoc">Post-hoc
one way non-parametric analysis
(olink_one_non_parametric_posthoc)</a></li>
<li><a href="#regression-models-for-ordinal-data-olink_ordinalregression" id="toc-regression-models-for-ordinal-data-olink_ordinalregression">Regression
models for ordinal data (olink_ordinalRegression)</a></li>
<li><a href="#post-hoc-of-regression-models-for-ordinal-data-analysis-olink_ordinalregression_posthoc" id="toc-post-hoc-of-regression-models-for-ordinal-data-analysis-olink_ordinalregression_posthoc">Post-hoc
of regression models for ordinal data analysis
(olink_ordinalRegression_posthoc)</a></li>
<li><a href="#linear-mixed-effects-model-analysis-olink_lmer" id="toc-linear-mixed-effects-model-analysis-olink_lmer">Linear mixed
effects model analysis (olink_lmer)</a></li>
<li><a href="#post-hoc-linear-mixed-effects-model-analysis-olink_lmer_posthoc" id="toc-post-hoc-linear-mixed-effects-model-analysis-olink_lmer_posthoc">Post-hoc
linear mixed effects model analysis (olink_lmer_posthoc)</a></li>
<li><a href="#pathway-enrichment-olink_pathway_enrichment" id="toc-pathway-enrichment-olink_pathway_enrichment">Pathway Enrichment
(olink_pathway_enrichment)</a></li>
</ul></li>
<li><a href="#exploratory-analysis" id="toc-exploratory-analysis">Exploratory analysis</a>
<ul>
<li><a href="#principal-components-analysis-pca-plot-olink_pca_plot" id="toc-principal-components-analysis-pca-plot-olink_pca_plot">Principal
components analysis (PCA) plot (olink_pca_plot)</a></li>
<li><a href="#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap-olink_umap_plot" id="toc-uniform-manifold-approximation-and-projection-for-dimension-reduction-umap-olink_umap_plot">Uniform
Manifold Approximation and Projection for Dimension Reduction (UMAP)
(olink_umap_plot)</a></li>
</ul></li>
<li><a href="#visualization" id="toc-visualization">Visualization</a>
<ul>
<li><a href="#boxplots-for-outcomes-olink_boxplot" id="toc-boxplots-for-outcomes-olink_boxplot">Boxplots for outcomes
(olink_boxplot)</a></li>
<li><a href="#boxplots-for-qc-olink_dist_plot" id="toc-boxplots-for-qc-olink_dist_plot">Boxplots for QC
(olink_dist_plot)</a></li>
<li><a href="#point-range-plot-for-lmer-olink_lmer_plot" id="toc-point-range-plot-for-lmer-olink_lmer_plot">Point-range plot for
LMER (olink_lmer_plot)</a></li>
<li><a href="#heatmap-for-visualizing-pathway-enrichment-olink_pathway_heatmap" id="toc-heatmap-for-visualizing-pathway-enrichment-olink_pathway_heatmap">Heatmap
for visualizing pathway enrichment (olink_pathway_heatmap)</a></li>
<li><a href="#bargraph-for-visualizing-pathway-enrichment-olink_pathway_visualization" id="toc-bargraph-for-visualizing-pathway-enrichment-olink_pathway_visualization">Bargraph
for visualizing pathway enrichment
(olink_pathway_visualization)</a></li>
<li><a href="#scatterplot-for-qc-olink_qc_plot" id="toc-scatterplot-for-qc-olink_qc_plot">Scatterplot for QC
(olink_qc_plot)</a></li>
<li><a href="#heatmap-olink_heatmap_plot" id="toc-heatmap-olink_heatmap_plot">Heatmap
(olink_heatmap_plot)</a></li>
<li><a href="#plot-results-of-t-test-olink_volcano_plot" id="toc-plot-results-of-t-test-olink_volcano_plot">Plot results of
t-test (olink_volcano_plot)</a></li>
<li><a href="#theming-function-set_plot_theme" id="toc-theming-function-set_plot_theme">Theming function
(set_plot_theme)</a></li>
<li><a href="#color-theming-olink_color_discrete-olink_color_gradient-olink_fill_discrete-olink_fill_gradient" id="toc-color-theming-olink_color_discrete-olink_color_gradient-olink_fill_discrete-olink_fill_gradient">Color
theming (olink_color_discrete, olink_color_gradient,
olink_fill_discrete, olink_fill_gradient)</a></li>
</ul></li>
</ul>
</div>

<p>Olink® Analyze is an R package that provides a versatile toolbox to
enable fast and easy handling of Olink® NPX data for your proteomics
research. Olink® Analyze provides functions for using Olink data,
including functions for importing Olink® NPX datasets, as well as
quality control (QC) plot functions and functions for various
statistical tests. This package is meant to provide a convenient
pipeline for your Olink NPX data analysis.</p>
<div id="installation" class="section level1">
<h1>Installation</h1>
<p>You can install Olink® Analyze from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;OlinkAnalyze&quot;</span>)</span></code></pre></div>
</div>
<div id="list-of-functions" class="section level1">
<h1>List of functions</h1>
<p><strong>Preprocessing</strong></p>
<ul>
<li><em>read_NPX</em> Function to read NPX data into long format</li>
<li><em>olink_plate_randomizer</em> Randomize samples on plate</li>
<li><em>olink_bridgeselector</em> Select bridge samples</li>
<li><em>olink_normalization</em> Normalization of all proteins (by
OlinkID)</li>
</ul>
<p><strong>Statistical analysis</strong></p>
<ul>
<li><em>olink_ttest</em> Function which performs a t-test per
protein</li>
<li><em>olink_wilcox</em> Function which performs a Mann-Whitney U Test
per protein</li>
<li><em>olink_anova</em> Function which performs an ANOVA per
protein</li>
<li><em>olink_anova_posthoc</em> Function which performs an ANOVA
post-hoc test per protein</li>
<li><em>olink_one_non_parametric</em> Function which performs a
Kruskal-Wallis Test or Friedman Test per protein</li>
<li><em>olink_one_non_parametric_posthoc</em> Function which performs
post-hoc test for one way non-parametric test</li>
<li><em>olink_ordinalRegression</em> Function which performs an ordinal
regression per protein</li>
<li><em>olink_ordinalRegression_posthoc</em> Function which performs an
ordinal regression post-hoc test per protein</li>
<li><em>olink_lmer</em> Function which performs a linear mixed model per
protein</li>
<li><em>olink_lmer_posthoc</em> Function which performs a linear mixed
model post-hoc per protein</li>
<li><em>olink_pathway_enrichment</em> Function which performs GSEA or
ORA pathway enrichment using outcome from other statistical tests</li>
</ul>
<p><strong>Visualization</strong></p>
<ul>
<li><em>olink_boxplot</em> Function which plots boxplots of a selected
variable</li>
<li><em>olink_dist_plot</em> Function to plot the NPX distribution by
panel</li>
<li><em>olink_lmer_plot</em> Function which performs a point-range plot
per protein on a linear mixed model</li>
<li><em>olink_pathway_visualization</em> Function which plots a bar
graph for pathways of interest</li>
<li><em>olink_pathway_heatmap</em> Function which plots estimates of
proteins associated with pathways of interest</li>
<li><em>olink_pca_plot</em> Function to plot a PCA of the data</li>
<li><em>olink_qc_plot</em> Function to plot an overview of a sample
cohort per Panel</li>
<li><em>olink_volcano_plot</em> Easy volcano plot with Olink theme</li>
<li><em>olink_heatmap_plot</em> Function which generate a heatmap over
all proteins</li>
<li><em>set_plot_theme</em> Function to set plot theme</li>
</ul>
<p><strong>Sample datasets</strong></p>
<ul>
<li><em>npx_data1</em> NPX Data in Long format</li>
<li><em>npx_data2</em> NPX Data in Long format, Follow-up</li>
<li><em>manifest</em> A sample manifest including Sample ID, Subject ID
and clinical variables</li>
</ul>
</div>
<div id="usage" class="section level1">
<h1>Usage</h1>
<div id="load-the-library" class="section level2">
<h2>Load the library</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Load OlinkAnalyze</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(OlinkAnalyze)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co"># Load other libraries used in Vignette</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="fu">library</span>(stringr)</span></code></pre></div>
</div>
<div id="introduction-to-olink-npx-data-format" class="section level2">
<h2>Introduction to Olink NPX data format</h2>
<p>The package contains two test data files named npx_data1 and
npx_data2. These are synthetic datasets that resemble Olink® data
accompanied by clinical variables. Olink® data in that are delivered in
long format or imported with the function <em>read_NPX</em> (that
converts the data into a long format) contain the following columns:</p>
<ul>
<li><strong>SampleID</strong> <em>&lt;chr&gt;</em>: Sample names or
IDs.</li>
<li><strong>Index</strong> <em>&lt;int&gt;</em>: Unique number for each
SampleID. It is used to make up for non unique sample IDs.</li>
<li><strong>OlinkID</strong> <em>&lt;chr&gt;</em>: Unique ID for each
assay assigned by Olink. In case the assay is included in more than one
panels it will have a different OlinkID in each one.</li>
<li><strong>UniProt</strong> <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li><strong>Assay</strong> <em>&lt;chr&gt;</em>: Common gene name for
the assay.</li>
<li><strong>MissingFreq</strong> <em>&lt;dbl&gt;</em>: Missing frequency
for the OlinkID, i.e. frequency of samples with NPX value below limit of
detection (LOD).</li>
<li><strong>Panel</strong> <em>&lt;chr&gt;</em>: Olink Panel that
samples ran on. Read more about Olink Panels here: <a href="https://olink.com/products-services/">https://olink.com/products-services/</a>.</li>
<li><strong>Panel_Version</strong> <em>&lt;chr&gt;</em>: Version of the
panel. A new panel version might include some different or improved
assays.</li>
<li><strong>PlateID</strong> <em>&lt;chr&gt;</em>: Name of the
plate.</li>
<li><strong>QC_Warning</strong> <em>&lt;chr&gt;</em>: Indication whether
the sample passed Olink QC. Read more here: <a href="https://olink.com/faq/how-is-quality-control-of-the-data-performed/">https://olink.com/faq/how-is-quality-control-of-the-data-performed/</a>.</li>
<li><strong>LOD</strong> <em>&lt;dbl&gt;</em>: Limit of detection (LOD)
is the minimum level of an individual protein that can be measured. LOD
is defined as 3 times the standard deviation over background.</li>
<li><strong>NPX</strong> <em>&lt;dbl&gt;</em>: Normalized Protein
eXpression, is Olink®’s unit of protein expression level in a
log<sub>2</sub> scale. The majority of the functions of this package use
NPX values for calculations. Read more about NPX here: <a href="https://olink.com/faq/what-is-npx/">https://olink.com/faq/what-is-npx/</a>.</li>
</ul>
<p><strong>Note:</strong> There are 5 additional variables in the sample
datasets npx_data1 and npx_data2 that include clinical or other
information, namely: Subject <em>&lt;chr&gt;</em>, Treatment
<em>&lt;chr&gt;</em>, Site <em>&lt;chr&gt;</em>, Time
<em>&lt;chr&gt;</em>, Project <em>&lt;chr&gt;</em>.</p>
</div>
</div>
<div id="preprocessing" class="section level1">
<h1>Preprocessing</h1>
<div id="read-npx-data-read_npx" class="section level2">
<h2>Read NPX data (read_NPX)</h2>
<p>The read_NPX function imports an NPX file of wide format that has
been exported from Olink® and converts the data into the (preferred by
R) long format. The wide format is the most common way Olink® delivers
data for Olink® Target 96, however, for data analysis a long format is
preferred. No prior alterations to the NPX output file should be made
for this function to work as expected.</p>
<div id="function-arguments" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>filename: Path to the NPX output file.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_NPX</span>(<span class="st">&quot;~/NPX_file_location.xlsx&quot;</span>)</span></code></pre></div>
</div>
<div id="function-output" class="section level3">
<h3>Function output</h3>
<p>A tibble in long format containing:</p>
<ul>
<li>SampleID: Sample names or IDs.</li>
<li>Index: Unique number for each SampleID. It is used to make up for
non unique sample IDs.</li>
<li>OlinkID: Unique ID for each assay assigned by Olink. In case the
assay is included in more than one panels it will have a different
OlinkID in each one.</li>
<li>UniProt: UniProt ID.</li>
<li>Assay: Common gene name for the assay.</li>
<li>MissingFreq: Missing frequency for the OlinkID, i.e. frequency of
samples with NPX value below limit of detection (LOD).</li>
<li>Panel: Olink Panel that samples ran on. Read more about Olink Panels
here: <a href="https://olink.com/products-services/">https://olink.com/products-services/</a>.</li>
<li>Panel_Version: Version of the panel. A new panel version might
include some different or improved assays.</li>
<li>PlateID: Name of the plate.</li>
<li>QC_Warning: Indication whether the sample passed Olink QC. Read more
here: <a href="https://olink.com/faq/how-is-quality-control-of-the-data-performed/">https://olink.com/faq/how-is-quality-control-of-the-data-performed/</a>.<br />
</li>
<li>LOD: Limit of detection (LOD) is the minimum level of an individual
protein that can be measured. LOD is defined as 3 times the standard
deviation over background.</li>
<li>NPX: Normalized Protein eXpression, is Olink’s unit of protein
expression level in a log<sub>2</sub> scale. The majority of the
functions of this package use NPX values for calculations. Read more
about NPX here: <a href="https://olink.com/faq/what-is-npx/">https://olink.com/faq/what-is-npx/</a>.</li>
</ul>
</div>
</div>
<div id="randomize-samples-on-plate-olink_plate_randomizer" class="section level2">
<h2>Randomize samples on plate (olink_plate_randomizer)</h2>
<p>The olink_plate_randomizer function randomly assigns samples to a
plate well with the option to keep the same individuals on the same
plate. Olink® does not recommend to force balance based on other
clinical variables.</p>
<div id="function-arguments-1" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>Manifest: tibble/data frame in long format containing all sample
ID’s. Sample ID column should be named SampleID.</li>
<li>PlateSize: Integer, either 96 or 48. 96 is default and should be
used for Olink® Target 96 and Olink® Explore projects. For Olink® Target
48 projects, use 48.</li>
<li>SubjectColumn: (Optional) Column name of the subject ID column.
Cannot contain missing values. If provided, subjects are kept on the
same plate. This argument is used for longitudinal studies and must be a
separate column from the SampleID column.</li>
<li>iterations: Number of iterations for fitting subjects on the same
plate.</li>
<li>available.spots: (Optional) Integer. Number of wells available on
each plate. Maximum 40 for Olink® Target 48 and 88 for Olink® Target
96/Explore. Can also take a vector equal to the number of plates to be
used indicating the number of wells available on each plate.</li>
<li>seed: Seed to set. Highly recommend setting it for
reproducibility.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">olink_plate_randomizer</span>(manifest, </span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>                       <span class="at">SubjectColumn =</span><span class="st">&quot;SubjectID&quot;</span>,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>                       <span class="at">seed=</span><span class="dv">111</span>)</span></code></pre></div>
</div>
<div id="function-output-1" class="section level3">
<h3>Function output</h3>
<p>A tibble including SampleID, SubjectID etc. assigned to well
positions.</p>
</div>
</div>
<div id="select-bridge-samples-olink_bridgeselector" class="section level2">
<h2>Select bridge samples (olink_bridgeselector)</h2>
<p>The bridge selection function selects a number of bridge samples
based on the input data. Bridge samples are used to normalize two
dataframes/projects that have been ran at different time points, hence,
a batch effect is expected. It select samples that fulfill certain
criteria that include good detectability, passing quality control and
covering a wide range of data points. When possible the function
recommends 8-16 bridge samples.</p>
<p>Bridge sample selection strategy: Start by choosing samples with at
most 10% missingness (sampleMissingFreq = 0.1), and in case there are
not enough samples to output, increase the threshold to 20%
(sampleMissingFreq = 0.2).</p>
<div id="function-arguments-2" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: tibble/data frame in long format such as produced by the
read_NPX function.</li>
<li>sampleMissingFreq: The threshold for sample wise missingness.</li>
<li>n: Number of bridge samples to be selected.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># Select overlapping samples</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="fu">olink_bridgeselector</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>                     <span class="at">sampleMissingFreq =</span> <span class="fl">0.1</span>,</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>                     <span class="at">n =</span> <span class="dv">8</span>)</span></code></pre></div>
</div>
<div id="function-output-2" class="section level3">
<h3>Function output</h3>
<p>Tibble with sample IDs and mean NPX for the pre-defined number of
bridging samples.</p>
</div>
</div>
<div id="normalizing-npx-data-olink_normalization" class="section level2">
<h2>Normalizing NPX data (olink_normalization)</h2>
<p>The olink_normalization is a function used to normalize NPX values
between two different dataframes/projects which have been ran at
different times. Commonly, there is a shift in (mean) NPX values between
runs, however, the spread of the data remains the same. This is why
normalization between dataframes/projects is required. When
normalization is performed, one of the two provided dataframes/projects
shall be used as a reference. If two dataframes/projects have been
normalized to one another, Olink® by default uses the chronologically
older one as reference. The function handles three different types of
normalization:</p>
<ul>
<li><strong>Bridging normalization</strong>: One of the dataframes is
adjusted to another using overlapping samples (bridge samples). The
overlapping samples should have the same IDs between dataframes, and
adjustment is made using the median of the paired differences between
the bridge samples. The two dataframes are provided as the inputs df1
and df2, while the one being used as reference is specified by the
reference_project and the overlapping samples are specified by the
overlapping_samples_df1. Only overlapping_samples_df1 should be provided
regardless of which dataframe is used as reference_project.</li>
<li><strong>Subset normalization</strong>: A subset of samples is used
to normalize two dataframes, one of which is used as a
reference_project. Adjustment is made using the differences of medians
between the sample subsets from the two dataframes. Both
overlapping_samples_df1 and overlapping_samples_df2 should be provided
as input. The sample IDs do not need to overlap. A special case of
subset normalization is where all samples (except control samples and
samples with QC warning) from df1 are used as input in
overlapping_samples_df1 and all samples from df2 are used as input in
overlapping_samples_df2. This is useful if no bridge samples was
included and one can assume that the distribution of the two datasets
should be very similar.</li>
<li><strong>Reference median normalization</strong>: Works only on one
dataframe. This is effectively subset normalization, but using
difference of medians to pre-recorded median values. df1,
overlapping_samples_df1 and reference_medians need to be specified.
Adjustment of df1 is made using the differences in median between the
overlapping samples and the reference medians.</li>
</ul>
<div id="function-arguments-3" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df1: First dataframe to be used in normalization (required).</li>
<li>df2: Second dataframe to be used in normalization.</li>
<li>overlapping_samples_df1: Samples to be used for adjustment factor
calculation in df1 (required).</li>
<li>overlapping_samples_df2: Samples to be used for adjustment factor
calculation in df2.</li>
<li>df1_project_nr: Project name of first dataset.</li>
<li>df2_project_nr: Project name of second dataset.</li>
<li>reference_project: Project name of reference_project. Needs to be
the same as either df1_project_nr or df2_project_nr. The project to
which the second project is to be adjusted to.</li>
<li>reference_medians: Dataframe which needs to contain columns
“OlinkID”, and “Reference_NPX”. Used for reference median
normalization.</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># Find overlapping samples</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>overlap_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(npx_data1<span class="sc">$</span>SampleID, npx_data2<span class="sc">$</span>SampleID) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(., <span class="st">&#39;CONTROL_SAMPLE&#39;</span>)) <span class="sc">%&gt;%</span> <span class="co">#Remove control samples</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  <span class="fu">pull</span>(.)</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co"># Perform Bridging normalization</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="fu">olink_normalization</span>(<span class="at">df1 =</span> npx_data1, </span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>                    <span class="at">df2 =</span> npx_data2, </span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>                    <span class="at">overlapping_samples_df1 =</span> overlap_samples,</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>                    <span class="at">df1_project_nr =</span> <span class="st">&#39;20200001&#39;</span>,</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>                    <span class="at">df2_project_nr =</span> <span class="st">&#39;20200002&#39;</span>,</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>                    <span class="at">reference_project =</span> <span class="st">&#39;20200001&#39;</span>)</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co"># Example of using all samples for normalization</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>subset_df1 <span class="ot">&lt;-</span> npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a>  <span class="fu">group_by</span>(SampleID) <span class="sc">%&gt;%</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">all</span>(QC_Warning <span class="sc">==</span> <span class="st">&#39;Pass&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(SampleID, <span class="st">&#39;CONTROL_SAMPLE&#39;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a>  <span class="fu">pull</span>(SampleID) <span class="sc">%&gt;%</span> </span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a>subset_df2 <span class="ot">&lt;-</span> npx_data2 <span class="sc">%&gt;%</span>   </span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a>  <span class="fu">group_by</span>(SampleID) <span class="sc">%&gt;%</span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">all</span>(QC_Warning <span class="sc">==</span> <span class="st">&#39;Pass&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(SampleID, <span class="st">&#39;CONTROL_SAMPLE&#39;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a>  <span class="fu">pull</span>(SampleID) <span class="sc">%&gt;%</span> </span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a><span class="fu">olink_normalization</span>(<span class="at">df1 =</span> npx_data1, </span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a>                    <span class="at">df2 =</span> npx_data2, </span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a>                    <span class="at">overlapping_samples_df1 =</span> subset_df1,</span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a>                    <span class="at">overlapping_samples_df2 =</span> subset_df2,</span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a>                    <span class="at">df1_project_nr =</span> <span class="st">&#39;20200001&#39;</span>,</span>
<span id="cb6-34"><a href="#cb6-34" tabindex="-1"></a>                    <span class="at">df2_project_nr =</span> <span class="st">&#39;20200002&#39;</span>,</span>
<span id="cb6-35"><a href="#cb6-35" tabindex="-1"></a>                    <span class="at">reference_project =</span> <span class="st">&#39;20200001&#39;</span>)</span></code></pre></div>
</div>
<div id="function-output-3" class="section level3">
<h3>Function output</h3>
<p>A tibble of NPX data in long format containing normalized NPX values,
including adjustment factors:</p>
<ul>
<li>SampleID: Sample names or IDs.</li>
<li>Index: Unique number for each SampleID. It is used to make up for
non unique sample IDs.</li>
<li>OlinkID: Unique ID for each assay assigned by Olink®. In case the
assay is included in more than one panels it will have a different
OlinkID in each one.</li>
<li>UniProt: UniProt ID.</li>
<li>Assay: Common gene name for the assay.</li>
<li>MissingFreq: Missing frequency for the OlinkID, i.e. frequency of
samples with NPX value below limit of detection (LOD).</li>
<li>Panel: Olink Panel that samples ran on. Read more about Olink Panels
here: <a href="https://olink.com/products-services/">https://olink.com/products-services/</a>.</li>
<li>Panel_Version: Version of the panel. A new panel version might
include some different or improved assays.</li>
<li>PlateID: Name of the plate.</li>
<li>QC_Warning: Indication whether the sample passed Olink QC. Read more
here: <a href="https://olink.com/faq/how-is-quality-control-of-the-data-performed/">https://olink.com/faq/how-is-quality-control-of-the-data-performed/</a>.</li>
<li>LOD: Limit of detection (LOD) is the minimum level of an individual
protein that can be measured. LOD is defined as 3 times the standard
deviation over background.</li>
<li>NPX: Normalized Protein eXpression, is Olink®’s unit of protein
expression level in a log<sub>2</sub> scale. The majority of the
functions of this package use NPX values for calculations. Read more
about NPX here: <a href="https://olink.com/faq/what-is-npx/">https://olink.com/faq/what-is-npx/</a>.</li>
<li>Project: Name given from the dataframe of origin.</li>
<li>Adj_factor: Adjustment factor, i.e. how much was added to or
subtracted from the original NPX value.</li>
</ul>
</div>
</div>
<div id="integrating-explore-npx-lod-olink_lod" class="section level2">
<h2>Integrating Explore NPX LOD (olink_lod)</h2>
<p>The olink_lod function adds LOD information to an Explore HT or
Explore 3072 NPX dataframe. This function can incorporate LOD based on
either an Explore dataset’s negative controls or using predetermined
fixed LOD values, which can be downloaded from the Document Download
Center at <a href="https://olink.com">olink.com</a>. The default LOD
calculation method is based off of the negative controls. If an NPX file
is intensity normalized, both intensity normalized and PC normalized
LODs are provided.</p>
<div id="function-arguments-4" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>data: Explore HT or Explore 3072 tibble/data frame in parquet file
format such as produced by the read_NPX function.</li>
<li>lod_file_path: Path to fixed LOD file from Olink website. This is
only applicable if using the fixed LOD method.</li>
<li>lod_method: String of LOD method name. Must be either “NCLOD”
(default) or “FixedLOD”.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># Integrating negative control LOD into Explore NPX dataset</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>explore_npx <span class="ot">&lt;-</span> <span class="fu">read_NPX</span>(<span class="st">&quot;~/Explore_NPX_file.parquet&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="fu">olink_lod</span>(explore_npx, <span class="at">lod_method =</span> <span class="st">&quot;NCLOD&quot;</span>)</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co"># Integrating fixed LOD into Explore NPX dataset - note that these are NOT real fixed LOD values</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>fixedLOD_filepath <span class="ot">&lt;-</span> <span class="st">&quot;~/ExploreHT_fixedLOD.csv&quot;</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>explore_npx <span class="ot">&lt;-</span> <span class="fu">read_NPX</span>(<span class="st">&quot;~/Explore_NPX_file.parquet&quot;</span>)</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="fu">olink_lod</span>(explore_npx, <span class="at">lod_file_path =</span> fixedLOD_filepath, <span class="at">lod_method =</span> <span class="st">&quot;FixedLOD&quot;</span>)</span></code></pre></div>
</div>
<div id="function-output-4" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li><strong>SampleID</strong> <em>&lt;chr&gt;</em>: Sample names or
IDs.</li>
<li><strong>SampleType</strong> <em>&lt;chr&gt;</em>: Sample type.
Indicates whether a sample is a study sample or a type of Olink control
sample.</li>
<li><strong>WellID</strong> <em>&lt;chr&gt;</em>: Well location in the
plate.</li>
<li><strong>PlateID</strong> <em>&lt;chr&gt;</em>: Name of the
plate.</li>
<li><strong>DataAnalysisRefID</strong> <em>&lt;chr&gt;</em>: Version of
the panel (Explore 3072) or block (Explore HT). A new panel or block
version might include some different or improved assays.</li>
<li><strong>OlinkID</strong> <em>&lt;chr&gt;</em>: Unique ID for each
assay assigned by Olink. In case the assay is included in more than one
panels it will have a different OlinkID in each one.</li>
<li><strong>UniProt</strong> <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li><strong>Assay</strong> <em>&lt;chr&gt;</em>: Common gene name for
the assay.</li>
<li><strong>AssayType</strong> <em>&lt;chr&gt;</em>: Assay type.
Indicates whether an assay is a panel assay or an Olink control
assay.</li>
<li><strong>Panel</strong> <em>&lt;chr&gt;</em>: Olink Panel that
samples ran on. Read more about Olink Panels here: <a href="https://olink.com/products-services/" class="uri">https://olink.com/products-services/</a>.</li>
<li><strong>Block</strong> <em>&lt;chr&gt;</em>: Olink Block that
samples ran on.</li>
<li><strong>Count</strong> <em><int></em>: Raw counts generated during
sequencing.</li>
<li><strong>ExtNPX</strong> <em><num></em>: Extension normalized NPX
value that is used in NPX calculation. Read more about ExtNPX here: <a href="https://olink.com/faq/how-is-the-npx-value-calculated-in-explore/" class="uri">https://olink.com/faq/how-is-the-npx-value-calculated-in-explore/</a></li>
<li><strong>NPX</strong> <em><num></em>: Normalized Protein eXpression,
is Olink’s unit of protein expression level in a log2 scale. The
majority of the functions of this package use NPX values for
calculations. If Explore data is PC normalized, NPX reflects the PC
normalized value. If Explore data is intensity normalized NPX, NPX
reflects the intensity normalized value. Read more about NPX here: <a href="https://olink.com/faq/what-is-npx/" class="uri">https://olink.com/faq/what-is-npx/</a>.</li>
<li><strong>Normalization</strong> <em>&lt;chr&gt;</em>: The
normalization method used.</li>
<li><strong>PCNormalizedNPX</strong> <em><num></em>: Normalized Protein
eXpression, is Olink’s unit of protein expression level in a log2 scale.
The majority of the functions of this package use NPX values for
calculations. Regardless of normalization method, this column always
reflects PC normalized NPX values. Read more about NPX here: <a href="https://olink.com/faq/what-is-npx/" class="uri">https://olink.com/faq/what-is-npx/</a>.</li>
<li><strong>AssayQC</strong> <em>&lt;chr&gt;</em>:Indicates whether an
assay contains a QC warning.</li>
<li><strong>SampleQC</strong> <em>&lt;chr&gt;</em>: Indicates whether a
sample contains a QC warning within a block.</li>
<li><strong>ExploreVersion</strong> <em>&lt;chr&gt;</em>: The version of
the Explore software library that was used to produce the data
file.</li>
<li><strong>LOD</strong> <em><num></em>: LOD added by olink_lod
function. If Explore dataset is intensity normalized, this will reflect
the intensity normalized LOD. If Explore dataset is PC normalized, this
will reflect the PC normalized LOD.</li>
<li><strong>PCNormalizedLOD</strong> <em><num></em>: LOD added by
olink_lod function. This will always reflect the PC normalized LOD,
regardless of the normalization method applied to the Explore
dataset.</li>
<li><strong>LOD</strong> <em><num></em>: LOD added by olink_lod
function. If Explore dataset is intensity normalized, this will reflect
the intensity normalized LOD. If Explore dataset is PC normalized, this
will reflect the PC normalized LOD.</li>
<li><strong>PCNormalizedLOD</strong> <em><num></em>: LOD added by
olink_lod function. This will always reflect the PC normalized LOD,
regardless of the normalization method applied to the Explore
dataset.</li>
</ul>
</div>
</div>
</div>
<div id="statistical-analysis" class="section level1">
<h1>Statistical analysis</h1>
<div id="t-test-analysis-olink_ttest" class="section level2">
<h2>T-test analysis (olink_ttest)</h2>
<p>The olink_ttest function performs a Welch 2-sample t-test or paired
t-test at confidence level 0.95 for every protein (by OlinkID) for a
given grouping variable using the function <em>t.test</em> from the R
library <em>stats</em> and corrects for multiple testing using the
Benjamini-Hochberg method (“fdr”) using the function <em>p.adjust</em>
from the R library <em>stats</em>. Adjusted p-values are logically
evaluated towards adjusted p-value&lt;0.05. The resulting t-test table
is arranged by ascending p-values.</p>
<div id="function-arguments-5" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and an outcome factor with 2
levels.</li>
<li>variable: Character value that should represent a column in the df
to be used as a grouping variable. Needs to have exactly 2 levels.</li>
<li>pair_id: Character value indicating which column contains the paired
sample identifier. Only used for paired t-tests.</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">olink_ttest</span>(<span class="at">df =</span> npx_data1,</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>            <span class="at">variable =</span> <span class="st">&#39;Treatment&#39;</span>)</span></code></pre></div>
</div>
<div id="function-output-5" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink® ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink® Panel.</li>
<li>estimate <em>&lt;dbl&gt;</em>: Difference in mean NPX between
groups.</li>
<li>statistic <em>&lt;dbl&gt;</em>: Value of the t-statistic.</li>
<li>p.value <em>&lt;dbl&gt;</em>: P-value for the test.</li>
<li>parameter <em>&lt;dbl&gt;</em>: Degrees of freedom for the
t-statistic.</li>
<li>conf.low <em>&lt;dbl&gt;</em>: Low bound of the confidence interval
for the mean.</li>
<li>conf.high <em>&lt;dbl&gt;</em>: High bound of the confidence
interval for the mean.</li>
<li>method <em>&lt;chr&gt;</em>: Method that was used.</li>
<li>alternative <em>&lt;chr&gt;</em>: : Description of the alternative
hypothesis.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="mann-whitney-u-test-analysis-olink_wilcox" class="section level2">
<h2>Mann-Whitney U Test analysis (olink_wilcox)</h2>
<p>The olink_wilcox function performs a 2-sample Mann-Whitney U test or
paired Mann-Whitney U test at confidence level 0.95 for every protein
(by OlinkID) for a given grouping variable using the function
<em>wilcox.test</em> from the R library <em>stats</em> and corrects for
multiple testing using the Benjamini-Hochberg method (“fdr”) based on
the function <em>p.adjust</em> from the R library <em>stats</em>.
Adjusted p-values are logically evaluated towards adjusted
p-value&lt;0.05. The resulting Mann-Whitney U table is arranged by
ascending p-values.</p>
<div id="function-arguments-6" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and an outcome factor with 2
levels.</li>
<li>variable: Character value that should represent a column in the df
to be used as a grouping variable. Needs to have exactly 2 levels.</li>
<li>pair_id: Character value indicating which column contains the paired
sample identifier. Only used for paired Mann-Whitney U tests.</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">olink_wilcox</span>(<span class="at">df =</span> npx_data1,</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>             <span class="at">variable =</span> <span class="st">&#39;Treatment&#39;</span>)</span></code></pre></div>
</div>
<div id="function-output-6" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink® ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink® Panel.</li>
<li>statistic <em>&lt;dbl&gt;</em>: Value of the Mann-Whitney U
statistic.</li>
<li>p.value <em>&lt;dbl&gt;</em>: P-value for the test.</li>
<li>method <em>&lt;chr&gt;</em>: Method that was used.</li>
<li>alternative <em>&lt;chr&gt;</em>: : Description of the alternative
hypothesis.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="analysis-for-variance-anova-olink_anova" class="section level2">
<h2>Analysis for variance (ANOVA) (olink_anova)</h2>
<p>The olink_anova is a wrapper function that performs an ANOVA F-test
for each assay using the function <em>Anova</em> from the R library
<em>car</em> and Type III sum of squares. The function handles both
factor and numerical variables, and/or confounding factors.</p>
<p>Samples with missing variable information or factor levels are
excluded from the analysis. Character columns in the input data frame
are converted to factors. The automatic handling of the data from above
is announced by a message if the flag <em>verbose=TRUE</em>.</p>
<p>Crossed/interaction analysis, i.e. A*B formula notation, is inferred
from the variable argument in the following cases:</p>
<ul>
<li>c(‘A’,‘B’)</li>
<li>c(‘A:B’)</li>
<li>c(‘A:B’, ‘B’) or c(‘A:B’, ‘A’)</li>
</ul>
<p>Inference is specified in a message if <em>verbose=TRUE</em>.</p>
<p>For covariates, crossed analyses need to be specified explicitly,
i.e. two main effects will not be expanded with a c(‘A’,‘B’) notation.
Main effects present in the variable take precedence. The formula
notation of the final model is specified in a message if
<em>verbose=TRUE</em>.</p>
<p>Adjusted p-values are calculated using the function <em>p.adjust</em>
from the R library <em>stats</em> with the Benjamini &amp; Hochberg
(1995) method (“fdr”). The threshold is determined by logic evaluation
of Adjusted_pval &lt; 0.05. Covariates are not included in the p-value
adjustment.</p>
<div id="function-arguments-7" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and an outcome factor with at
least 3 levels.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.
Otherwise, if length &gt; 1, the included variable names will be used in
crossed analyses. It can also accept the notations ‘:’ or ’*’.</li>
<li>outcome: Name of the column from df that contains the dependent
variable. Default: NPX.</li>
<li>covariates: Single character value or character array. Default:
NULL. Confounding factors to include in the analysis. In case of single
character then that should represent a column in the df. It can also
accept the notations ‘:’ or ’*’, while crossed analysis will not be
inferred from main effects.</li>
<li>return.covariates: Logical. Default: False. Returns F-test results
for the covariates. Note: Adjusted p-values will be NA for
covariates.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co"># One-way ANOVA, no covariates</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>anova_results_oneway <span class="ot">&lt;-</span> <span class="fu">olink_anova</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>                                    <span class="at">variable =</span> <span class="st">&#39;Site&#39;</span>)</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co"># Two-way ANOVA, no covariates</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>anova_results_twoway <span class="ot">&lt;-</span> <span class="fu">olink_anova</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>                                    <span class="at">variable =</span> <span class="fu">c</span>(<span class="st">&#39;Site&#39;</span>, <span class="st">&#39;Time&#39;</span>))</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co"># One-way ANOVA, Treatment as covariates</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>anova_results_oneway <span class="ot">&lt;-</span> <span class="fu">olink_anova</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>                                    <span class="at">variable =</span> <span class="st">&#39;Site&#39;</span>,</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>                                    <span class="at">covariates =</span> <span class="st">&#39;Treatment&#39;</span>)</span></code></pre></div>
</div>
<div id="function-output-7" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink Panel.</li>
<li>term <em>&lt;chr&gt;</em>: Name of the variable that was used for
the p-value calculation. The “:” between variables indicates interaction
between variables.</li>
<li>df <em>&lt;dbl&gt;</em>: Numerator of degrees of freedom.</li>
<li>sumsq <em>&lt;dbl&gt;</em>: Sum of squares.</li>
<li>meansq <em>&lt;dbl&gt;</em>: Mean of squares.</li>
<li>statistic <em>&lt;dbl&gt;</em>: Value of F-statistic.</li>
<li>p.value <em>&lt;dbl&gt;</em>: P-value for the test.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="post-hoc-anova-analysis-olink_anova_posthoc" class="section level2">
<h2>Post-hoc ANOVA analysis (olink_anova_posthoc)</h2>
<p>olink_anova_posthoc performs a post-hoc ANOVA test using the function
<em>emmeans</em> from the R library <em>emmeans</em> with Tukey p-value
adjustment per assay (by OlinkID) at confidence level 0.95.</p>
<p>The function handles both factor and numerical variables and/or
covariates. The post-hoc test for a numerical variable compares the
difference in means of the outcome variable (default: NPX) for 1
standard deviation (SD) difference in the numerical variable, e.g. mean
NPX at mean (numerical variable) versus mean NPX at mean (numerical
variable) + 1*SD (numerical variable).</p>
<div id="function-arguments-8" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and an outcome factor with at
least 3 levels.</li>
<li>olinkid_list: Character vector of OlinkID’s on which to perform the
post-hoc analysis. If not specified, all assays in df are used.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.
Otherwise, if length &gt; 1, the included variable names will be used in
crossed analyses. It can also accept the notations ‘:’ or ’*’.</li>
<li>covariates: Single character value or character array. Default:
NULL. Confounding factors to include in the analysis. In case of single
character then that should represent a column in the df. It can also
accept the notations ‘:’ or ’*’, while crossed analysis will not be
inferred from main effects.</li>
<li>outcome: Name of the column from df that contains the dependent
variable. Default: NPX.</li>
<li>effect: Term on which to perform the post-hoc analysis. Character
vector. Must be subset of or identical to the variable and no adjustment
is performed.</li>
<li>mean_return: Logical. If true, returns the mean of each factor level
rather than the difference in means (default). Note that no p-value is
returned for mean_return = TRUE.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co"># calculate the p-value for the ANOVA</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>anova_results_oneway <span class="ot">&lt;-</span> <span class="fu">olink_anova</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>                                    <span class="at">variable =</span> <span class="st">&#39;Site&#39;</span>)</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co"># extracting the significant proteins</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>anova_results_oneway_significant <span class="ot">&lt;-</span> anova_results_oneway <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>  <span class="fu">filter</span>(Threshold <span class="sc">==</span> <span class="st">&#39;Significant&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>  <span class="fu">pull</span>(OlinkID)</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>anova_posthoc_oneway_results <span class="ot">&lt;-</span> <span class="fu">olink_anova_posthoc</span>(<span class="at">df =</span> npx_data1,</span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>                                                    <span class="at">olinkid_list =</span> anova_results_oneway_significant,</span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>                                                    <span class="at">variable =</span> <span class="st">&#39;Site&#39;</span>,</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>                                                    <span class="at">effect =</span> <span class="st">&#39;Site&#39;</span>)</span></code></pre></div>
</div>
<div id="function-output-8" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink Panel.</li>
<li>term <em>&lt;chr&gt;</em>: Name of the variable that was used for
the p-value calculation. The “:” between variables indicates interaction
between variables.</li>
<li>contrast <em>&lt;chr&gt;</em>: Variables (in term) that are
compared.<br />
</li>
<li>estimate <em>&lt;dbl&gt;</em>: Difference in mean NPX between
variables (from contrast).</li>
<li>conf.low <em>&lt;dbl&gt;</em>: Low bound of the confidence interval
for the mean.</li>
<li>conf.high <em>&lt;dbl&gt;</em>: High bound of the confidence
interval for the mean.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="one-way-non-parametric-test-olink_one_non_parametric" class="section level2">
<h2>One way non-parametric test (olink_one_non_parametric)</h2>
<p>The olink_one_non_parametric is a wrapper function that performs
either a Kruskal-Wallis test or a Friedman test for each assay using the
function <em>kruskal.test</em> from the R library <em>stats</em> or the
function <em>friedman_test</em> from the R library <em>rstatix</em>. The
posthoc test for Friedman test is performed using the function
<em>wilcox_test</em> from the R library <em>rstatix</em>, whereas for
Kruskal-Wallis test the function <em>dunnTest</em> from the R library
<em>FSA</em> is applied. The function handles both factor and numerical
variables, and/or confounding factors.</p>
<p>Samples with missing variable information or factor levels are
excluded from the analysis. Character columns in the input data frame
are converted to factors. The automatic handling of the data from above
is announced by a message if the flag <em>verbose=TRUE</em>.</p>
<p>Adjusted p-values are calculated based on the Benjamini &amp;
Hochberg (1995) method (“fdr”). The threshold is determined by logic
evaluation of Adjusted_pval &lt; 0.05.</p>
<div id="function-arguments-9" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and an outcome factor with at
least 3 levels.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.</li>
<li>dependence: Logical. Default: FALSE. When the groups are
independent, the kruskal-Wallis will run, when the groups are dependent,
the Friedman test will run.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co"># One-way Kruskal-Wallis Test</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>kruskal_results <span class="ot">&lt;-</span> <span class="fu">olink_one_non_parametric</span>(<span class="at">df =</span> npx_df, </span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>                                            <span class="at">variable =</span> <span class="st">&quot;Time&quot;</span>)</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co"># One-way Friedman Test</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>friedman_results <span class="ot">&lt;-</span> <span class="fu">olink_one_non_parametric</span>(<span class="at">df =</span> npx_df, </span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>                                             <span class="at">variable =</span> <span class="st">&quot;Time&quot;</span>, </span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>                                             <span class="at">subject =</span> <span class="st">&quot;Subject&quot;</span>,</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>                                             <span class="at">dependence =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="function-output-9" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink Panel.</li>
<li>term <em>&lt;chr&gt;</em>: Name of the variable that was used for
the p-value calculation.</li>
<li>df <em>&lt;dbl&gt;</em>: Numerator of degrees of freedom.</li>
<li>method <em>&lt;dbl&gt;</em>: Name of the performed test.</li>
<li>statistic <em>&lt;dbl&gt;</em>: Value of the test’s statistic.</li>
<li>p.value <em>&lt;dbl&gt;</em>: P-value for the test.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="post-hoc-one-way-non-parametric-analysis-olink_one_non_parametric_posthoc" class="section level2">
<h2>Post-hoc one way non-parametric analysis
(olink_one_non_parametric_posthoc)</h2>
<p>olink_one_non_parametric_posthoc performs a post-hoc Wilcoxon test
using the function <em>wilcox_test</em> from the R library
<em>rstatix</em> with Benjamini &amp; Hochberg p-value adjustment per
assay (by OlinkID) at confidence level 0.95. The function handles both
factor and numerical variables and/or covariates.</p>
<div id="function-arguments-10" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and an outcome factor with at
least 3 levels.</li>
<li>olinkid_list: Character vector of OlinkID’s on which to perform the
post-hoc analysis. If not specified, all assays in df are used.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co">#Friedman Test</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>Friedman_results <span class="ot">&lt;-</span> <span class="fu">olink_one_non_parametric</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>                                             <span class="at">variable =</span> <span class="st">&quot;Time&quot;</span>, </span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>                                             <span class="at">subject =</span> <span class="st">&quot;Subject&quot;</span>,</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>                                             <span class="at">dependence =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a><span class="co">#Filtering out significant and relevant results.</span></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>significant_assays <span class="ot">&lt;-</span> Friedman_results <span class="sc">%&gt;%</span></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>  <span class="fu">filter</span>(Threshold <span class="sc">==</span> <span class="st">&#39;Significant&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(OlinkID) <span class="sc">%&gt;%</span></span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a>  <span class="fu">pull</span>()</span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a><span class="co">#Posthoc test for the results from Friedman Test</span></span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a>friedman_posthoc_results <span class="ot">&lt;-</span> <span class="fu">olink_one_non_parametric_posthoc</span>(npx_data1, </span>
<span id="cb13-16"><a href="#cb13-16" tabindex="-1"></a>                                                             <span class="at">variable =</span> <span class="st">&quot;Time&quot;</span>, </span>
<span id="cb13-17"><a href="#cb13-17" tabindex="-1"></a>                                                             <span class="at">test =</span> <span class="st">&quot;friedman&quot;</span>,</span>
<span id="cb13-18"><a href="#cb13-18" tabindex="-1"></a>                                                             <span class="at">olinkid_list =</span> significant_assays)</span></code></pre></div>
</div>
<div id="function-output-10" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink Panel.</li>
<li>term <em>&lt;chr&gt;</em>: Name of the variable that was used for
the p-value calculation.</li>
<li>contrast <em>&lt;chr&gt;</em>: Variables (in term) that are
compared.<br />
</li>
<li>estimate <em>&lt;dbl&gt;</em>: Difference in mean NPX between
variables (from contrast).</li>
<li>conf.low <em>&lt;dbl&gt;</em>: Low bound of the confidence interval
for the location parameter.</li>
<li>conf.high <em>&lt;dbl&gt;</em>: High bound of the confidence
interval for the location parameter.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="regression-models-for-ordinal-data-olink_ordinalregression" class="section level2">
<h2>Regression models for ordinal data (olink_ordinalRegression)</h2>
<p>The olink_ordinalRegression is a wrapper function that performs an
ANOVA F-test for each assay (ordinal transformed) using the function
<em>Anova</em> from the R library <em>car</em> and Type II sum of
squares. The function handles both factor and numerical variables,
and/or confounding factors.</p>
<p>Samples with missing variable information or factor levels are
excluded from the analysis. Character columns in the input data frame
are converted to factors. The automatic handling of the data from above
is announced by a message if the flag <em>verbose=TRUE</em>.</p>
<p>Crossed/interaction analysis, i.e. A*B formula notation, is inferred
from the variable argument in the following cases:</p>
<ul>
<li>c(‘A’,‘B’)</li>
<li>c(‘A:B’)</li>
<li>c(‘A:B’, ‘B’) or c(‘A:B’, ‘A’)</li>
</ul>
<p>Inference is specified in a message if <em>verbose=TRUE</em>.</p>
<p>For covariates, crossed analyses need to be specified explicitly,
i.e. two main effects will not be expanded with a c(‘A’,‘B’) notation.
Main effects present in the variable take precedence. The formula
notation of the final model is specified in a message if
<em>verbose=TRUE</em>.</p>
<p>Adjusted p-values are calculated using the function <em>p.adjust</em>
from the R library <em>stats</em> with the Benjamini &amp; Hochberg
(1995) method (“fdr”). The threshold is determined by logic evaluation
of Adjusted_pval &lt; 0.05. Covariates are not included in the p-value
adjustment.</p>
<div id="function-arguments-11" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and an outcome factor with at
least 3 levels.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.
Otherwise, if length &gt; 1, the included variable names will be used in
crossed analyses. It can also accept the notations ‘:’ or ’*’.</li>
<li>outcome: Name of the column from df that contains the dependent
variable. Default: NPX.</li>
<li>covariates: Single character value or character array. Default:
NULL. Confounding factors to include in the analysis. In case of single
character then that should represent a column in the df. It can also
accept the notations ‘:’ or ’*’, while crossed analysis will not be
inferred from main effects.</li>
<li>return.covariates: Logical. Default: False. Returns F-test results
for the covariates. Note: Adjusted p-values will be NA for
covariates.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co"># Two-way ordinal regression, no covariates</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>ordinalRegression_results_twoway <span class="ot">&lt;-</span> <span class="fu">olink_ordinalRegression</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>                                                            <span class="at">variable =</span> <span class="fu">c</span>(<span class="st">&#39;Site&#39;</span>, <span class="st">&#39;Time&#39;</span>))</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="co"># One-way ordinal regression, Treatment as covariates</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>ordinalRegression_oneway <span class="ot">&lt;-</span> <span class="fu">olink_ordinalRegression</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>                                        <span class="at">variable =</span> <span class="st">&#39;Site&#39;</span>,</span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>                                        <span class="at">covariates =</span> <span class="st">&#39;Treatment&#39;</span>)</span></code></pre></div>
</div>
<div id="function-output-11" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink Panel.</li>
<li>term <em>&lt;chr&gt;</em>: Name of the variable that was used for
the p-value calculation. The “:” between variables indicates interaction
between variables.</li>
<li>df <em>&lt;dbl&gt;</em>: Numerator of degrees of freedom.</li>
<li>statistic <em>&lt;dbl&gt;</em>: Value of F-statistic.</li>
<li>p.value <em>&lt;dbl&gt;</em>: P-value for the test.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="post-hoc-of-regression-models-for-ordinal-data-analysis-olink_ordinalregression_posthoc" class="section level2">
<h2>Post-hoc of regression models for ordinal data analysis
(olink_ordinalRegression_posthoc)</h2>
<p>olink_ordinalRegression_posthoc performs a post-hoc ANOVA test using
the function <em>emmeans</em> from the R library <em>emmeans</em> with
Tukey p-value adjustment per assay (by OlinkID) at confidence level
0.95. The function handles both factor and numerical variables and/or
covariates.</p>
<div id="function-arguments-12" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and an outcome factor with at
least 3 levels.</li>
<li>olinkid_list: Character vector of OlinkID’s on which to perform the
post-hoc analysis. If not specified, all assays in df are used.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.
Otherwise, if length &gt; 1, the included variable names will be used in
crossed analyses. It can also accept the notations ‘:’ or ’*’.</li>
<li>covariates: Single character value or character array. Default:
NULL. Confounding factors to include in the analysis. In case of single
character then that should represent a column in the df. It can also
accept the notations ‘:’ or ’*’, while crossed analysis will not be
inferred from main effects.</li>
<li>outcome: Name of the column from df that contains the dependent
variable. Default: NPX.</li>
<li>effect: Term on which to perform the post-hoc analysis. Character
vector. Must be subset of or identical to the variable and no adjustment
is performed.</li>
<li>mean_return: Logical. If true, returns the mean of each factor level
rather than the difference in means (default). Note that no p-value is
returned for mean_return = TRUE.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co"># Two-way Ordinal Regression</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>ordinalRegression_results <span class="ot">&lt;-</span> <span class="fu">olink_ordinalRegression</span>(<span class="at">df =</span> npx_data1,</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>                             <span class="at">variable=</span><span class="st">&quot;Treatment:Time&quot;</span>)</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a><span class="co"># extracting the significant proteins</span></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>significant_assays <span class="ot">&lt;-</span> ordinalRegression_results <span class="sc">%&gt;%</span> </span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>  <span class="fu">filter</span>(Threshold <span class="sc">==</span> <span class="st">&#39;Significant&#39;</span> <span class="sc">&amp;</span> term <span class="sc">==</span> <span class="st">&#39;Treatment:Time&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>  <span class="fu">select</span>(OlinkID) <span class="sc">%&gt;%</span></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>  <span class="fu">pull</span>()</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a><span class="co"># Posthoc test for the model NPX~Treatment*Time,</span></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>ordinalRegression_posthoc_results <span class="ot">&lt;-</span> <span class="fu">olink_ordinalRegression_posthoc</span>(npx_data1, </span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a>                                                                     <span class="at">variable=</span><span class="fu">c</span>(<span class="st">&quot;Treatment:Time&quot;</span>),</span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a>                                                                     <span class="at">covariates=</span><span class="st">&quot;Site&quot;</span>,</span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a>                                                                     <span class="at">olinkid_list =</span> significant_assays,</span>
<span id="cb15-15"><a href="#cb15-15" tabindex="-1"></a>                                                                     <span class="at">effect =</span> <span class="st">&quot;Treatment:Time&quot;</span>)</span></code></pre></div>
</div>
<div id="function-output-12" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink Panel.</li>
<li>term <em>&lt;chr&gt;</em>: Name of the variable that was used for
the p-value calculation. The “:” between variables indicates interaction
between variables.</li>
<li>contrast <em>&lt;chr&gt;</em>: Variables (in term) that are
compared.<br />
</li>
<li>estimate <em>&lt;dbl&gt;</em>: Difference in mean NPX between
variables (from contrast).</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="linear-mixed-effects-model-analysis-olink_lmer" class="section level2">
<h2>Linear mixed effects model analysis (olink_lmer)</h2>
<p>The olink_lmer fits a linear mixed effects model for every protein
(by OlinkID) in every panel, using the function <em>lmer</em> from the R
library <em>lmerTest</em> and the function <em>anova</em> from the R
library <em>stats</em>. The function handles both factor and numerical
variables and/or covariates.</p>
<p>Samples with missing variable information or factor levels are
excluded from the analysis. Character columns in the input data frame
are converted to factors. The automatic handling of the data from above
is announced by a message if the flag <em>verbose=TRUE</em>.</p>
<p>Crossed/interaction analysis, i.e. A*B formula notation, is inferred
from the variable argument in the following cases:</p>
<ul>
<li>c(‘A’,‘B’)</li>
<li>c(‘A:B’)</li>
<li>c(‘A:B’, ‘B’) or c(‘A:B’, ‘A’)</li>
</ul>
<p>Inference is specified in a message if <em>verbose=TRUE</em>.</p>
<p>For covariates, crossed analyses need to be specified explicitly,
i.e. two main effects will not be expanded with a c(‘A’,‘B’) notation.
Main effects present in the variable take precedence. The formula
notation of the final model is specified in a message if
<em>verbose=TRUE</em>.</p>
<p>Adjusted p-values are calculated using the function <em>p.adjust</em>
from the R library <em>stats</em> with the Benjamini &amp; Hochberg
(1995) method (“fdr”). The threshold is determined by logic evaluation
of Adjusted_pval &lt; 0.05. Covariates are not included in the p-value
adjustment.</p>
<div id="function-arguments-13" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and 1-2 variables with at least 2
levels and subject ID.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.
Otherwise, if length &gt; 1, the included variable names will be used in
crossed analyses. It can also accept the notations ‘:’ or ’*’.</li>
<li>outcome: Name of the column from df that contains the dependent
variable. Default: NPX.</li>
<li>random: Single character value or character array with random
effects.</li>
<li>covariates: Single character value or character array. Default:
NULL. Confounding factors to include in the analysis. In case of single
character then that should represent a column in the df. It can also
accept the notations ‘:’ or ’*’, while crossed analysis will not be
inferred from main effects.</li>
<li>return.covariates: Logical. Default: False. Returns F-test results
for the covariates. Note: Adjusted p-values will be NA for
covariates.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># Linear mixed model with one variable.</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>lmer_results_oneway <span class="ot">&lt;-</span> <span class="fu">olink_lmer</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>                                  <span class="at">variable =</span> <span class="st">&#39;Site&#39;</span>,</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>                                  <span class="at">random =</span> <span class="st">&#39;Subject&#39;</span>)</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="co"># Linear mixed model with two variables.</span></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>lmer_results_twoway <span class="ot">&lt;-</span> <span class="fu">olink_lmer</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>                                  <span class="at">variable =</span> <span class="fu">c</span>(<span class="st">&#39;Site&#39;</span>, <span class="st">&#39;Treatment&#39;</span>),</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>                                  <span class="at">random =</span> <span class="st">&#39;Subject&#39;</span>)</span></code></pre></div>
</div>
<div id="function-outcome" class="section level3">
<h3>Function outcome</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink Panel.</li>
<li>term <em>&lt;chr&gt;</em>: Name of the variable that was used for
the p-value calculation. The “:” between variables indicates interaction
between variables.</li>
<li>sumsq <em>&lt;dbl&gt;</em>: Sum of squares.</li>
<li>meansq <em>&lt;dbl&gt;</em>: Mean of squares.</li>
<li>NumDF <em>&lt;dbl&gt;</em>: Numerator of degrees of freedom.</li>
<li>DenDF <em>&lt;dbl&gt;</em>: Denominator of degrees of freedom.</li>
<li>statistic <em>&lt;dbl&gt;</em>: Value of F-statistic.</li>
<li>p.value <em>&lt;dbl&gt;</em>: P-value for the test.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="post-hoc-linear-mixed-effects-model-analysis-olink_lmer_posthoc" class="section level2">
<h2>Post-hoc linear mixed effects model analysis
(olink_lmer_posthoc)</h2>
<p>The olink_lmer_posthoc function is similar to olink_lmer but performs
a post-hoc analysis based on a linear mixed model effects model using
the function <em>lmer</em> from the R library <em>lmerTest</em> and the
function <em>emmeans</em> from the R library <em>emmeans</em>. The
function handles both factor and numerical variables and/or covariates.
Differences in estimated marginal means are calculated for all pairwise
levels of a given output variable. Degrees of freedom are estimated
using Satterthwaite’s approximation. The post-hoc test for a numerical
variable compares the difference in means of the outcome variable
(default: NPX) for 1 standard deviation difference in the numerical
variable, e.g. mean NPX at mean(numerical variable) versus mean NPX at
mean(numerical variable) + 1*SD(numerical variable). The output tibble
is arranged by ascending adjusted p-values.</p>
<div id="function-arguments-14" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and 1-2 variables with at least 2
levels and subject ID.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.
Otherwise, if length &gt; 1, the included variable names will be used in
crossed analyses. It can also accept the notations ‘:’ or ’*’.</li>
<li>olinkid_list: Character vector of OlinkID’s on which to perform the
post-hoc analysis. If not specified, all assays in df are used.</li>
<li>effect: Term on which to perform the post-hoc analysis. Character
vector. Must be subset of or identical to the variable.</li>
<li>outcome: Name of the column from df that contains the dependent
variable. Default: NPX.</li>
<li>random: Single character value or character array with random
effects.</li>
<li>covariates: Single character value or character array. Default:
NULL. Confounding factors to include in the analysis. In case of single
character then that should represent a column in the df. It can also
accept the notations ‘:’ or ’*’, while crossed analysis will not be
inferred from main effects.</li>
<li>mean_return: Logical. If true, returns the mean of each factor level
rather than the difference in means (default). Note that no p-value is
returned for mean_return = TRUE and no adjustment is performed.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># Linear mixed model with two variables.</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>lmer_results_twoway <span class="ot">&lt;-</span> <span class="fu">olink_lmer</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>                                  <span class="at">variable =</span> <span class="fu">c</span>(<span class="st">&#39;Site&#39;</span>, <span class="st">&#39;Treatment&#39;</span>),</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>                                  <span class="at">random =</span> <span class="st">&#39;Subject&#39;</span>)</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="co"># extracting the significant proteins</span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>lmer_results_twoway_significant <span class="ot">&lt;-</span> lmer_results_twoway <span class="sc">%&gt;%</span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>  <span class="fu">filter</span>(Threshold <span class="sc">==</span> <span class="st">&#39;Significant&#39;</span>, term <span class="sc">==</span> <span class="st">&#39;Treatment&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>  <span class="fu">pull</span>(OlinkID)</span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a><span class="co"># performing post-hoc analysis</span></span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>lmer_posthoc_twoway_results <span class="ot">&lt;-</span> <span class="fu">olink_lmer_posthoc</span>(<span class="at">df =</span> npx_data1,</span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a>                                                  <span class="at">olinkid_list =</span> lmer_results_twoway_significant,</span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a>                                                  <span class="at">variable =</span> <span class="fu">c</span>(<span class="st">&#39;Site&#39;</span>, <span class="st">&#39;Treatment&#39;</span>),</span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a>                                                  <span class="at">random =</span> <span class="st">&#39;Subject&#39;</span>,</span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>                                                  <span class="at">effect =</span> <span class="st">&#39;Treatment&#39;</span>) </span></code></pre></div>
</div>
<div id="function-output-13" class="section level3">
<h3>Function output</h3>
<p>A tibble with the following columns:</p>
<ul>
<li>Assay <em>&lt;chr&gt;</em>: Assay name.</li>
<li>OlinkID <em>&lt;chr&gt;</em>: Unique Olink ID.</li>
<li>UniProt <em>&lt;chr&gt;</em>: UniProt ID.</li>
<li>Panel <em>&lt;chr&gt;</em>: Olink Panel.</li>
<li>term <em>&lt;chr&gt;</em>: Name of the variable that was used for
the p-value calculation. The “:” between variables indicates interaction
between variables.</li>
<li>contrast <em>&lt;chr&gt;</em>: Variables (in term) that are
compared.<br />
</li>
<li>estimate <em>&lt;dbl&gt;</em>: Difference in mean NPX between
variables (from contrast).</li>
<li>conf.low <em>&lt;dbl&gt;</em>: Low bound of the confidence interval
for the mean.</li>
<li>conf.high <em>&lt;dbl&gt;</em>: High bound of the confidence
interval for the mean.</li>
<li>Adjusted_pval <em>&lt;dbl&gt;</em>: Adjusted p-value for the test
(Benjamini &amp; Hochberg).</li>
<li>Threshold <em>&lt;chr&gt;</em>: Text indication if assay is
significant (adjusted p-value &lt; 0.05).</li>
</ul>
</div>
</div>
<div id="pathway-enrichment-olink_pathway_enrichment" class="section level2">
<h2>Pathway Enrichment (olink_pathway_enrichment)</h2>
<p>The olink_pathway_enrichment function can be used to perform Gene Set
Enrichment Analysis (GSEA) or Over-representation Analysis (ORA) using
MSigDB, Reactome, KEGG, or GO. MSigDB includes curated gene sets (C2)
and ontology gene sets (C5) which encompasses Reactome, KEGG, and GO.
This function performs enrichment using the <em>gsea</em> or
<em>enrich</em> functions from clusterProfiler from BioConductor. The
function uses the estimate from a previous statistical analysis for one
contrast for all proteins. MSigDB is subset if ontology is KEGG, GO, or
Reactome. test_results must contain estimates for all assays. Posthoc
results can be used but should be filtered for one contrast to improve
interpretability.</p>
<p>Alternative statistical results can be used as input as long as they
include the columns “OlinkID”, “Assay”, and “estimate”. A column named
“Adjusted_pal” is also needed for ORA. Any statistical results that
contains one estimate per protein will work as long as the estimates are
comparable to each other.</p>
<p>clusterProfiler is originally developed by Guangchuang Yu at the
School of Basic Medical Sciences at Southern Medical University.</p>
<p>T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L
Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal
enrichment tool for interpreting omics data. The Innovation. 2021,
2(3):100141. doi: 10.1016/j.xinn.2021.100141</p>
<div id="function-arguments-15" class="section level3">
<h3>Function Arguments</h3>
<ul>
<li>data: NPX data frame in long format with columns Assay, OlinkID,
UniProt, SampleID, QC_Warning, NPX, and LOD</li>
<li>test_results: a data frame of statistical test results including
Adjusted_pval and estimate columns</li>
<li>method: String of method name. Must be either “GSEA” (default) or
“ORA”</li>
<li>ontology: String of database to query. Must be either “MSigDb”,
“KEGG”, “GO”, and “Reactome”</li>
<li>organism: String of name of organism. Must be either “human” or
“mouse”</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>npx_df <span class="ot">&lt;-</span> npx_data1 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;control&quot;</span>, SampleID, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>))</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>ttest_results <span class="ot">&lt;-</span> <span class="fu">olink_ttest</span>(</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>  <span class="at">df =</span> npx_df,</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>  <span class="at">variable =</span> <span class="st">&quot;Treatment&quot;</span>,</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>  <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>)</span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a><span class="fu">try</span>({ <span class="co"># This expression might fail if dependencies are not installed</span></span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a>gsea_results <span class="ot">&lt;-</span> <span class="fu">olink_pathway_enrichment</span>(<span class="at">data =</span> npx_data1, <span class="at">test_results =</span> ttest_results)</span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a>ora_results <span class="ot">&lt;-</span> <span class="fu">olink_pathway_enrichment</span>(</span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a>  <span class="at">data =</span> npx_data1,</span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a>  <span class="at">test_results =</span> ttest_results, <span class="at">method =</span> <span class="st">&quot;ORA&quot;</span>)</span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a>}, <span class="at">silent =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="function-output-14" class="section level3">
<h3>Function Output</h3>
<p>A data frame of enrichment results. Columns for ORA include:</p>
<ul>
<li>ID <em>&lt;chr&gt;</em>: Pathway ID from MSigDB</li>
<li>Description <em>&lt;chr&gt;</em>: Description of Pathway from
MSigDB</li>
<li>GeneRatio <em>&lt;chr&gt;</em>: ratio of input proteins that are
annotated in a term</li>
<li>BgRatio <em>&lt;chr&gt;</em>: ratio of all genes that are annotated
in this term</li>
<li>pvalue <em>&lt;dbl&gt;</em>: p-value of enrichment</li>
<li>p.adjust <em>&lt;dbl&gt;</em>: Adjusted p-value
(Benjamini-Hochberg)</li>
<li>qvalue <em>&lt;dbl&gt;</em>: false discovery rate, the estimated
probability that the normalized enrichment score represents a false
positive finding</li>
<li>geneID: <em>&lt;chr&gt;</em> list of input proteins (Gene Symbols)
annotated in a term delimited by “/”</li>
<li>Count <em>&lt;dbl&gt;</em>: Number of input proteins that are
annotated in a term</li>
</ul>
<p>Columns for GSEA:</p>
<ul>
<li>ID <em>&lt;chr&gt;</em>: Pathway ID from MSigDB</li>
<li>Description <em>&lt;chr&gt;</em>: Description of Pathway from
MSigDB</li>
<li>setSize <em>&lt;dbl&gt;</em>: ratio of input proteins that are
annotated in a term</li>
<li>enrichmentScore <em>&lt;dbl&gt;</em>: Enrichment score, degree to
which a gene set is over-represented at the top or bottom of the ranked
list of genes</li>
<li>NES <em>&lt;dbl&gt;</em>: Normalized Enrichment Score, normalized to
account for differences in gene set size and in correlations between
gene sets and expression data sets. NES can be used to compare analysis
results across gene sets.</li>
<li>pvalue <em>&lt;dbl&gt;</em>: p-value of enrichment</li>
<li>p.adjust <em>&lt;dbl&gt;</em>: Adjusted p-value
(Benjamini-Hochberg)</li>
<li>qvalue <em>&lt;dbl&gt;</em>: false discovery rate, the estimated
probability that the normalized enrichment score represents a false
positive finding</li>
<li>rank <em>&lt;dbl&gt;</em>: the position in the ranked list where the
maximum enrichment score occurred</li>
<li>leading_edge <em>&lt;chr&gt;</em>: contains tags, list, and signal.
Tags gives an indication of the percentage of genes contributing to the
enrichment score. List gives an indication of where in the list the
enrichment score is obtained. Signal represents the enrichment signal
strength and combines the tag and list.</li>
<li>core_enrichment <em>&lt;chr&gt;</em>: list of input proteins (Gene
Symbols) annotated in a term delimited by “/”</li>
</ul>
</div>
</div>
</div>
<div id="exploratory-analysis" class="section level1">
<h1>Exploratory analysis</h1>
<div id="principal-components-analysis-pca-plot-olink_pca_plot" class="section level2">
<h2>Principal components analysis (PCA) plot (olink_pca_plot)</h2>
<p>Generates PCA projection of all samples from NPX data along two
principal components (Default PC2 vs PC1) colored by the variable
QC_Warning and including the percentage of explained variance. The
function used the functions <em>prcomp</em> and <em>ggplot</em> from the
R libraries <em>stats</em> and <em>ggplot2</em>, respectively. By
default, the values scaled and centered in the PCA and proteins with
missing NPX values removed from the corresponding assay(s). Unique
sample names are required. Imputation by median value is done for assays
with missingness &lt;10% and for multi-plate projects, and for
missingness &lt;5% for single plate projects.</p>
<p>The values are by default scaled and centered in the PCA and proteins
with missing NPX values are by default removed from the corresponding
assay. Unique sample names are required. Imputation by the median is
done for assays with missingness &lt;10% for multi-plate projects and
&lt;5% for single plate projects. The plot is printed, and a list of
ggplot objects is returned.</p>
<p>If byPanel = TRUE, the data processing (imputation of missing values
etc) and subsequent PCA is performed separately per panel. A faceted
plot is printed, while the individual ggplot objects are returned.</p>
<p>The arguments outlierDefX and outlierDefY can be used to identify
outliers in the PCA. Samples more than +/-outlierDef[X,Y] standard
deviations from the mean of the plotted PC will be labelled. Both
arguments have to be specified.</p>
<div id="function-arguments-selection" class="section level3">
<h3>Function arguments (selection)</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain SampleID,
NPX and column that will be used for grouping/coloring.</li>
<li>color_g: Character value indicating the column name that should be
used as fill color. Default QC_Warning.</li>
<li>x_val: Integer indicating which principal component to plot along
the x-axis. Default 1.</li>
<li>y_val: Integer indicating which principal component to plot along
the y-axis. Default 2.</li>
<li>label_samples: Logical. If TRUE, points are replaced with SampleID.
Default FALSE.</li>
<li>drop_assays: Logical. All assays with any missing values will be
dropped. Takes precedence over sample drop.</li>
<li>drop_samples: Logical. All samples with any missing values will be
dropped.</li>
<li>n_loadings: Integer. Plot the top n_loadings ranked by size.</li>
<li>loadings_list: Character vector indicating for which OlinkID’s to
plot loadings. Arguments n_loadings and loadings_list can be used
together.</li>
<li>byPanel: Logical. Perform the PCA per panel (default FALSE)</li>
<li>outlierDefX: (Optional) The number standard deviations along the PC
plotted on the x-axis that defines an outlier.</li>
<li>outlierDefY: (Optional) The number standard deviations along the PC
plotted on the y-axis that defines an outlier.</li>
<li>OutlierLines: Logical. Draw dashed lines at +/-outlierDef[X,Y]
standard deviations from the mean of the plotted PCs (default
FALSE)</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
<li>quiet: Logical. Default: False. If TRUE, the resulting plot is not
printed.</li>
</ul>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(SampleID, <span class="st">&#39;CONTROL_SAMPLE&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>  <span class="fu">olink_pca_plot</span>(<span class="at">df =</span> .,</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>                 <span class="at">color_g =</span> <span class="st">&quot;QC_Warning&quot;</span>, <span class="at">byPanel =</span> <span class="cn">TRUE</span>)  </span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABO1BMVEUAAAAAx+EzMzNNTU1NTW5NTY5NbqtNjshuTU1uTW5uTY5ujo5uq+Rzc3Nzc4xzc6RzjIxzjKRzjLxzpKRzpLxzpNOMc3OMc4yMc6SMjHOMjIyMjKSMjLyMpKSMpLyMpNOMvNOMvOmOTU2OTW6OyP+kc3Okc4ykc6SkjHOkjIykpHOkpKSkvNOkvOmk09Ok0+mk0/+rbk2r5P+8jHO8jIy8pHO8pIy8pKS8vIy8vKS807y80+m86em86f/Ijk3I///TpHPTpIzTvIzTvKTTvLzT06TT09PT0+nT6bzT6enT6f/T/9PT/+nT///kq27kyMjk///pvIzpvKTp06Tp07zp6bzp6dPp6enp6f/p/9Pp/+np///+HwT/yI7/06T/09P/5Kv/6bz/6dP/6en//8j//9P//+T//+n////6q0sqAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAdvUlEQVR4nO1di5/stHXOsGkeTecu3HYXAiRhKXuhJSlbhtImzSVDnyn03jAlaVN2mbCE6fj//wtqPWzrcY50ZEm2PKvvB3fH1pGs8/nT0WNszbeaiooIfGvuClQsG1VAFVGoAqqIQhVQRRSqgCqiUAVUEYUqoIooVAFVRKEKqCIKVUAVUagCqohCFVBFFKqAKqKwFAHt1+v1o6fi83HTHjy+VVK3/Gi3vuKpF3ARh+sb2rVWq4iKctzdxZawGCxDQMfN+bOmub9kAmm1xP5sOzkx7Fhys32X6ehwfRV1rRWHcfL+cq0o2IM7DvPslula1h2Q+P0rz8LrWgCWISAhkPY23sh/mKaUGMTPHZ784uX2Dt+/TLvNGGAB8du7pykIFtCO1Xf77gWrepzEi8IiBNQHFdZ0d1I4f3hmGOwff7G5kWLbtvGCa+rn6/NPr//mmh2xLuwgP3OLRz8HBLFaQQoSAjpuKJ3g3R2oIFbE4e1P327/fRIn8aKwCAH1QWV//gwe4nBlXfA/3f+7Vh2H61Zth+v2U3skBCQ+N9tWZnuoT/IJiA3GLviYjGXeA/0aIiB2+fvXv/jgaav0W1nM/WvvnP/6lX9Ytx1be4V78Yn1l39BkmoJWISA+q5jz2QAxf/2nvC7K/7wJs66NW7M/2mPhIDEZ9EPbqEuydWFnT9jhbBaPOE6kH8MY7gLa7ZXzU78d9HIYlg12MiuLZgJSHxiabt1FVBC+AXUxih2M9s/XbTad71W/48QkPi8P0fHNP5BdHslOaODJ3aIgPZthOQa3950xbDK8v9f4QISn/Y8alYBJYS/C2vjDuNd/mkb8Pr815cOAe1wAYHT+GGOxKTEO0cxyJIjKgPgNP7+9f99uxsGyWJYsfJ/8R/7tBOOkJiZH4sQ0DCIbrntBtF77c5tr3ZMWNsL9qefrY2JQCB6ATEty0GwKKP748fxg1+8fiv/lcWAAqoRKD1YwGjn7b+7vIKn8e1tFMPO/Uvv3Mh7und1YaKQXbiAWNHt0IkJgA1cxB9iIdtXWTvYvXs1FAMJqI6BMmDLFSSWn/li3HGj3/z7V3k3d3/5qB8/t2aogPBZGIxeQGwV/A0xGcNmYShEwOORUxYDCoh1b3++mJn+QgTEp70teGtnA4+1MfeRAUn+2bH7uhUzd1hAbB3o/F/JwWNqLGepaCkCkvg4Ka/03mdKhPWLc2NhAkoF8d1H5LdmubAL6RfnxpIFJBZn6F9yquBd4lIGqiVjyQKqKABVQBVRqAKqiEIVUEUUqoAqolAFVBGFKqCKKFQBVUShCqgiClVAFVGoAqqIQhVQRRSqgCqiUAVUEYUqoIooVAFVRKEKqCIK0QL67V+t1+vXPsEN7i8vzHdwnPijdrSFHg5mxYUUORsO18ZbkId31uv3Rtf8j9bbTPMjUkB8rycG/OniQAH9l75PzokJiO0J8suxNWfUFOd2pIC265f+o72hn1/ijyUzAYWUeE4TUEiZs8EUEN8oa3TtQS7mRpyA9uv+PWNUJVVAA3jFq4AG9PujHP+lDUTN5z9sI/RPGUUXu/X5J83X7fjoJ7/ru7CvP7xcr3/Mhkvbx198uH70s+b4YWvA2Pz6wy6n2H6nK6kl7T8/7sZYff6hC/u6HVO87hiAzQwuIPZO9jvMW/EqyIWo+eChwYXFjbQU1BhMDmXPhSgBGW1p1w2HjpsXL9vYJF67ebETkHwLh28g9vjv2Mf3NnL4JJMuOgH1JbUC+uvuxZ0hfy8gcarEhikgBdR5owhI8dDgwjzuLBUBqUz4hqC5ESUgPUIfrtug09xfs5srXl9fv3Hb/PZyLQW0ZQ3q+DELMNv1409YSvvv52ue9AZPuhFxeiipTTn/ZZ+py98JiF/n+E/lvt/VCeiCKeZC6cI0D00utGPVshtAqUz0Zc+EhAJq8T//9vc/XLMbKzZV5al7ecf5dnNyHMm7PmHFkqQl3/pHdvSyJNlJMlMlfycg6/qlQQqIucSrr46BNA8HLqxjxVIKSGOiL3smpOzCZGTtxidyV15GolCJiLPdDoUyt0iSiwGSpaGkTk87th1cn78TUHeqWHRjoFt1/Kz2Qp2Hio15rFsaTCp2MyHRILrhW2Gsf/TTf//v6wgBseDEu7CupFMVkOkhLiDD8sQEdH8pa84+iC3DDoqA8C7MFNAgBLFxT1fSaXRhtoBUD90CMiytLmzZAmILif/YOvDPl2u251I7nmvn1f0q8RYfROskHTePfsZmtkxxTDFDSewCn7gG0e817RWKlZFDQIOHPgENli011iB64QJSv8rg+z7xfkg6JHqmHxnTeBZSkH6eJbWz1ouhpG4arwyMJItKvnL3QnF1YWqf7e7COsudMY3v9LRkATXHz19tXXmNLSPyVb0X39t2PbNYHtQWElunf/L7xhYQTxJrjAfW2PqS2EJim/TjvjieX19IfKn8hUR7EK166B5ED5aMmi90JpcvoIoHjiqgiihUAVVEoQqoIgpVQBVRqAKqiEIVUEUUqoAqolAFVBGFKqCKKFQBVUQhSkBVfW48BH6qgDLiIfBTBZQRD4GfKqCMeAj8nLCAgB9fnhjZ+JnftR4nKyDw598nRiZ+SnCtRxVQRlQBSfzf3579ya/Ex6++d3b2VlDmebBaFUBzHn6KcK0HycePvsv+Y/jmzbeaL7/9fkjmeVAEy1VAAt/85fvNV3/KQ9CX3/msjUddCCpYQEXE+dqFCTDxMBFJsCjUNC+0GEHQZI6XwHIVkIAhoOdtFKJn1pDEdWoB03E8+RixGPmMEdDzjqt5BFRU85NY4BgxGYIF9LynJ5ygFMO/AgW0yDFiKgQOotX4M4uAypqCCKQcI0Ioy1sDgdP4r77/vnJ+hi5sAQKKGCNCKM9fDSQfv3mTDxI/+rPmozOG0SH6NMdA6caIEAp0WMXUX2UkmIMVx2eyMSKEXCE3VYkL/C6sMPkkHCNCyCOgdGUuUEDlIdkYEUKWAFQFVBSSjREh5AtASUotQkDFdUqJUOw0/rQElMwXWiETqrXcCH1SXViSuT21mEkncVVAOTN3iA+nKwU02/HXCkK5AjqBaXzvAOXW+1LJCpp2IbtkAaXCTAJSb6PrlvLznnu+qgIiIJfPRQtIpoQIKOTC2VGOgPK5PY+AjNuNxx+SNqqAvEjkNlBEEQLyWNEEVKfxONL03GAZBXRhHiOK84WuRKYX0EhHH6iAiGOgcpFaQOOJSKcfs5T5p/FOG1nhRcpnKgEFMRl99WIERMNClSORmB/wDlKlkWQIXU4X9jBQkoCSXd84eUoCKi5c5e7C+vHhVJ4XM43PgQKH2nkFRJ6jZkUVUEbkncan0U8sZScjoNmbIoCs/CQJQPGcVQFlxCQCioohVUAD3FzMoqy8/HhuPn15KIobzce9/AnmmzGZ54aLikTBqSx+nE7RPE4qoMP1Wv7++nHDf2Y7JPMk8H/5iqfEC6hAfghfMHtLSNaFHd5WOdGPvJkzwHIrwtcU46PS+HGD6vHJjoEAx2YW0BgE8JO4cnSPnTaEAhYjoCgRZNaPskNZ07/k3ATwk75+KUoklWH6eH+5Xj++JV4h4xZuVuULERDIz/BqM3tLFd2dI+8YLX2JYwR0ePK0nWyYDEW3sCAg3/DEUZLo9oD8aJsrnP0A4QfySEtK3otFF0CplaGB+9dvgREisYUlwUqBxoLqjfqFUIY64AD5Ubd3+c1nUkrWDmXm3VCOjVvl9unuLs4FMgIFdLhmqxsRLSwRVhaUJNVE/TSBjBz86BtMIRHackceyQfm1Mbhumt3HNHe+GHTD0PxccsWyFgfby5ygC3MyJwKfbWFLqQHWGcG6iwXSPz4BWREHemsKSDMnzQCIq8QBQmIUXQF2YAEJdxEUoVabTQQDYFnSgGF8cPQ86NXVp7STljjIdijO4mwmGtEcQdf4cMyQwO7brVVBamFpQLCsV9AkwyGQH7ULh7iZ2XCOqsURhKQrTsH/Nd0WHpL177KWD962uztWeosAnIIZDicUkA4P9okAxWQ2StgVacoSCbT/EauCeRTz1M57TVw3LAe/pVnrJs3OnlfC0sMWzokAWWpywAHP8oOZQ3Aj1JBvaawBEyX9ORWO51+iI5bl0fzGZcNFBCfmx4/eAoZeVpYBnTVx9eEGi8faeHiB4NHQNAJI4NpsILgrgQmIKelOKT72KgtzIa7hTkrT6kDlG9QEOCwPKH/yQwXPxiMLkx8NmwcgxFgwKu5TGw6ipEn1xgmBw0cN7yPH5cZxPg7awsITA5gMR7j+RlRQcg7Wzi0gi0BoQ07TkAjkE1A1rgTSFTIDYy6k0GbxgfBFhDkMolg3chjHk5gRgHFxIYAAdmZikH07iWKj5jLxLKA02Mrp0P5KoMtkrGHNmlP22mZQaQQkGuUaaVlFVB6fgagw2g0AEGZsLI1WgLnWH7oAtqxNQ7xrU9QZhix+oFEohcMLsF1R2MujCMDPxKQi5p0Gp0QLRNEDliSfqmItm1AE5CkZkd9ICi/gFbIWyu+WWj6YJSBHwmPgNSjodHYEUkrDFw1A5Y04ynSBHR/yQnaU4N08mm8EmkN0HLr+kkuoOT8MCAeqiqw2UDpsc8BQswmoDwtzAN1hmC5GiQhS0ApFZSNH39dATIwehBRWZdKpyBFQOyNp4umGy4GZR4P23stQeWDNOPQ85oJEcjHj/dGjhSQORZSL5VBQA3niH9bSOUnsYBMn3CaaIWCoS0GmfhJIyBXCILmrYn0M/dbGaqvpuvFCSgYo6fxegKuGaeA9ILkmeSkFCQgI64inLlL00oFLzIlIvkBVGJyYROzWgFrHzADaafxTbNtJxfsiU36Fz5ZxkBu1ihFabarZLt4Tc7PCAF1n12LQWkx+Mj5eflp+oUyJ6zmY7OmnYdK0PI2Gl1E7VEwOT8QFyuTCsM32FPlEcY4AEXoK9Fb/sDmXNN4cWRz5hLQcNoyV0/G05ebH7t+gIAaK+ZA+rGKCnwGFq8iULb2Ws9xwycYaRfKwgBx1mAtS02wSW60exBbsbz8gP5Z7aDRWkRvNfzFGgt/CvYuduDsFhBvXbsZIpAJlTLlJOz5QJjZMI3keGTlR28s9klAQEB+dLjHH4S9i+QDzqw+VH/+jDexPXmUmHH7EkBAuJk9dEaKiUFOfiw3+vOKSaPNCKBJljl01NPROX54JRWoPsoNuIAXVxBk3gNQ+Yj6PCSB7iUUUE5+MAHBVkgzMRU0BDSagLw8gTkL3d5Fh19AAz22QfbqoQjtwjxLDisd0HlLJCtXol144yJssQJykWq0sGkqREW4gNwRAjHSThgWoQJyMe2exivYzTgLY4D7IYdCbK+zSikLP1794AKyJhAwtMTwyyfwMW1mDIALntZhuR3MQiSU/ZO0rZSC+PHfQUxBDkUBZcJXgG0JWJCATB6sZFcRWaG8eKm9g+ngxxkD0DyAJoBFM9tSvxx4leULqHcV8sF0DuYLPswO5dVv/S1wlB+Ch458ih1y2y0BeS8PRHkagaaP+1FfFo5cV9AO3E47BGRlySkgiB9l8wl9H4okAjIHd5ohSJmuBbtIVFZqzagUahtMra/YFm7BXxb6ruWYUxiHqIL00+Bnm8HEQPkBBeTaPwnwEHPdYwhmogoIG6vDF0agfJXx+LbZ8tYVulTvvhaSqp92EqKnY2JSslCdDwLOT3AEAjzEXO99RiyBPPpJJ5u4t34L00fervjjCqFfFnquBSeCtxxjsdF81ixQ3mke0OHgxy8gZMBhnXFpwUEVPhrwCMg/2nJyovkoHhU//r4JjkDuayGp5mmIlq5XhpM76iDigVrQPMLh4Mc3iMbuoH3G9INw1sFtR6JtRWLMnd5hiLL98ljwWwfOa2FV9ZGid1V2sklhCjKccPDjmcZDl0dnRnb/BdxtNMFKdQvI4S6VM6Wb3onpxd7+NSPPQpn7Wkgq6hRZNA4KaVUIBM6Psn+S/ChA6uIddUSd9Plvzscxsj01ctdYgLSU410oG9X8HarShzoe7UDEA6XmmdY74RWQcVOB9N5DNToBJGBZkQqkI4QioPCFMh1Q/+W3pQpoYAebgM4sIPRu+WulCWgwh923M2ulWK3KX/+gLszxe1jhC2W+elFuZ6iAFB2ZxafQT8TvhY0XkNIowJjiFJBxHV+0G1U9Y5tf5Bf5ghfKEtQMaWTevi1fwI75xUIkQgTVS6chSEKmAe26oQJq0N8ETRyBSMQNvNjmRoLBI1h8mv4r7W+mFi4gYvUoGphaQMgUvksWrxiYuyUPVkG3JSvS7p+kmgcqyBytU66cUECxg2i4asBpJdVBUP97EUg/p1vPKaVyBKQUQpYGrdQ00/gwgDUz7v9Ai+2u8os1MIvmVG5cNeORegMuy69Q/WjZiMYekDTgXigbAecSEILBVP3JI7wVIjOyKRG9O4cjR0YBKfN/f5FZHigbccvsgLNS37Uw/B1+MQsTEKqrKZFofyAsi9HewrKlqk4GAY26ZyYlVoJj6ZEqoMmFlF9AYa2EZDleQLtughr51kEEIVavA6sDy+3Uzyhdq0jFjwH6/bcyEfmxsyWsjvJt/KOnzeGarZXFETSGEPNW2wmuMgGLDAJKxY8JtZKBi0LWgNE0QrJRq0NB76PYeeK4eXw7kYBAmSBlecp0ycWc+HqdQpCMHwt9xegVNHmBnINL83fngTTpTyQ2/MnNKbowzQhvQh4FaeuyeEeolkXzzEY6fkys9Pp7w5A+09Bdt4qFLuW+RWMFJPe+YduYTC4gWEG2fIxSIYPBCiNvJNLxY8Lrp+aKbqX/heyAS/m4CGLJfiLxcE3+OZHx03jDPaqA4EJ8dqo1wSsECfkxIGuGOaCfMa2wER5ekpumUGizMBGkj5vs78bbLpD0QwxARjHQxzHIxk9XZ9QD5USfjsQW44SbMaw6tGoLzPFmqp8m3QpxmKIfF1H5EbA/kPgXd8E/F4CDDXzOQUsoY8UICB204O3Flo3+qVmMgCQGd/WT6ilUQFZRiBDzCei4YYsc95dXYzIHAFcEZOdsc6aAGinDFZhkZQ+rd25+sArrZ4gCckR0eeSuBLnW6hOJ4nWn7m9Q5hDQ66iJAS5JmoV39qEtLT8/Q3iAz6tH1BBkHnvzRgho2/GS/dd6rIBBN6ecpwWgYAHl5scpdpKZIxdZQOO7sH6dI/82v0MdKbVFnY4SUHBLy84PrUbB9TYz0kIQvVxrJbqZYqNxfU7hs3XpB45kSppHY+Qq5+fHVSH3ihliCxVP8HrkNF4JzJPtkUgPqZaNToZpMqR5QhC9ri5+tNd1hyd+kwlIbylu0jwOB3rth75Tvfw01U712QTUT+MBaWkFBFTWwY/6nO83b45/YtM9NVI/43V3pEGLBNEYfJS/KKt8CMg8EhR/PPpxT9SMFOMgqK44P+qbBs/PfjA2AqEAnfB0VECC/jcVFB/lbznuJvzJS7qAHPP0sQIKBcqP+q7Tbz6TUop48dKE6Z4/ACENLk11dKg+HjdhO/mnaGFj1lTVs2oIwnM7iyID40d/WW70GAiHUWunE3DiNAKaNjMKLNjAZ/WRjrvpRQoIw9QCcjY7Z3NLVB8VxQnI9FRXimWode2InV14Oiqfn519N7uAwioM0DWNgHYBb31bmZMBFRBs7MyLFI6ZeQhG+dFf180ioCDMJCD20Dh9AmZkTgbbVbrzFEt8bciX18GP9rru7AKyaHA1mmgYX2WQV+n1zOmACWhcXiXJa+a5josf5c3dsgWUQUHGVxnkVXo9c0IAjpL9xvsm9TxMpo/jUvjRQVlkX6lIXoMlCCg6L8goaLMsASEVftgCcgQcgvvOb4H0Y8wKuUg5/Awg9diZFVSggBSMGwvZpZh5XYvWyxGQVxMy6cEKiBI3RhQ0zmxSfqInDUNyX17+LmzdI8/jCsHQXI5pQWkEhPODIesuto0lEXcxuQXkQqrHFSyQVv2go6QXCjajYjIBQTloHXc0AvdITPq4grcHB0JQ3BUnRu6XDoxRzvhiIhC4S2vSxxXMMY4vvQpIN/PkmIawUAGlGwOtNLcpDCxMPlm7MCA4PywBrQYox7bV+CsUgKkEhK8ojqtACHw+5npcAdTPwvViId80vpzxYeggOpGAjAC0Gi+ggC/KwguPRL5lDouv2dpe8DQ+sYD0w5HlpDRMiYzrZFR/ipiF5XhcARwEWqPAsGKSGKbE7AKawOu5HmkFBYRbOEshC21qBWVdqVfYc9icrID8X5RWAZHg8msKr8t4qB4NQGGz2RSGKVEFlDOzDsvTKiAS3DSddBemw3aV3DmRrzCmXlGoAsqZWQPAAzwuSnbFKTB3F9aUMo3PklkD2JDm+XInIQoQUHYUIqAR3/8sANO81jMvJVMLCJMAYcg8xZwiLWZ6M3VSTCsglwa88ggT0JRCUx75/ep7Z2dvdeergFJndmrAPZ3wGASUlR7Dd4XfvPlW8+W3434W3YniAvCkAlqtHAryJhUrIOVphS+/81kbj7oQlFxABfbhCxLQit7+ph0u6c9L8SgkUAWUOvO4LixcDrMK6HkbhZJucdejxGnEAgQExC3iE3uh1RsJXUDPE772JBH92FRGlDKNl4lwBoM4AouTEW098vu8H0KnEpDiS3n6KWYhEYWtHxqNE/KsvTc3xJ8qoLyZaUADUEk8DtP4r77/vnI+CT+6uyW5zbEcAVmn8l+ciuGR34/OGJLOwspzV0PpAsKfNZvg2rEYxY/lWdnuLldA5XLaY+QkY1HtpXgBgd1+2Zz2SCOgAgc+ChYgIBtLCUHh/CzFswFVQBlRBZQzcwwWwnKqLqxkVAFlRBVQzsxxWATJaabxZWOpAloEHgI/VUAZ8RD4qQLKiIfATxVQRjwEfuIEhOIFPGkhxknIDUdQjeNyxWXqfUxBlI0XTt04F8ZVYlSuNJmqgMYZ50IVEHKdUzPOhSqgioeFKqCKKFQBVUShCqgiClVAFVFILiBsqxOfsbaDudtUyxVbbmCN8yGoznYePykxF8JJSi4gbKsTj7H+U4huU+2XF2LLDaxxPgTV2c7jJyXmQjhJqQWEbnXiMdZ/CtFpqv/2S2S5gTXOh6A623n8pMRcyEFSagGhW534jPufQvSaGpeIKzewxvkQVGc7j5+UmAs5SMosILHVCc04j4C83XxQjfMhqM52nnECIo+BcJLyCui5u4ctUECeGufDkgSkkpRSQM6tTrzGRQjIVeN8COSix0wC0kjKOIj2t2b0pxCdpmGDaC9HQTXOh6A623nGDaLJAsJJyjiN17c68Rg3Pl9GT+O9HAXVOB+C6mznGTeNJ68DoSQlFxC21YnHuPH5opjKj2nKDaxxPgTV2c7jJyXmQjhJ9auMiihUAVVEoQqoIgpVQBVRqAKqiEIVUEUU5hXQds1xwQ+Om/bj41uRcv/Ks87ouLlpmt16/eipON6LTOub5nAtc99f8ozcsjk8eTq1H7kwip/hYAp+ZhYQ5+Nwzf7s11fsjKDhcH3eE7RrGdi1h/tHiuOHa8bLy+LMcXN13LTHe8H0vmN58RjHD7O5v7yZhp8SBMRd5x4zXwVZ656gwzULNVcs6WLIyRhr9tKIEbV7fHv8oOPrZkIfcmIUP/yA8TEJP0UIiJGzk83iD894Y9v3BPWfVAHdX3KS5AlJ0L5L351KCBrFj4hE7N8p+ClCQK2zWnhpFFqabe+20oVtefL21baLv5Ih+qprYI3e2S0Zo/iRAmrHiFPwU4KA2HhGxN0BPUFdwp5T0agnxdhgeyUHiX0D68L98jGKH+H9bn0zCT9FzMKuGhdBnbOi+xepShPqLNsG1s46+IFZ2GIxjh8xiF7f6JaZ+CkhAjGgIVppLYNstkon3hm0DayN5jzldAQ0jp9Wd+efdpP1zPyUIqB+YLcXLQdoYQNXGgFyrsob2JXId4ICCuKHH3QLRZn5KUZA2jTV7ONFotnqtLN7MU7gRyc2BuII4YehFdwk/BQjILFQdtx0Uwl9lsEM24lEZyqNRBI/y6cYXYg+tVkYB5kfvsbKKZiCn3IEJBbeuxPmOseWf3chhwK7LlGebeSSUDdIPLV1IAEyP8xQSGQCfsr/MlXp5LPlWDJm5qd8AfXrqWSczndhJMzLzwIEFPrVzQl9G0/CvPwsQEAVJaMKqCIKVUAVUagCqohCFVBFFKqAKqJQBVQRhf8Hyo98wv8at6cAAAAASUVORK5CYII=" style="display: block; margin: auto;" /></p>
</div>
<div id="function-output-15" class="section level3">
<h3>Function output</h3>
<p>A list of objects of class <em>ggplot</em> (silently returned). Plots
are also printed unless option <code>quiet = TRUE</code> is set. If
outlierDefX and outlierDefY are specified, a list of outliers can be
extracted from the <strong>ggplot</strong> object based on these
parameters.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>npx_data <span class="ot">&lt;-</span> npx_data1 <span class="sc">%&gt;%</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">SampleID =</span> <span class="fu">paste</span>(SampleID, <span class="st">&quot;_&quot;</span>, Index, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">olink_pca_plot</span>(<span class="at">df=</span>npx_data, <span class="at">color_g =</span> <span class="st">&quot;QC_Warning&quot;</span>,</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>                    <span class="at">outlierDefX =</span> <span class="fl">2.5</span>, <span class="at">outlierDefY =</span> <span class="dv">4</span>, <span class="at">byPanel =</span> <span class="cn">TRUE</span>, <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a><span class="fu">lapply</span>(g, <span class="cf">function</span>(x){x<span class="sc">$</span>data}) <span class="sc">%&gt;%</span></span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>  <span class="fu">filter</span>(Outlier <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a>  <span class="fu">select</span>(SampleID, Outlier, Panel)</span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a><span class="co">#&gt;   SampleID Outlier           Panel</span></span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a><span class="co">#&gt; 1  B22_103       1 Cardiometabolic</span></span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a><span class="co">#&gt; 2  B68_149       1 Cardiometabolic</span></span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a><span class="co">#&gt; 3    B9_88       1 Cardiometabolic</span></span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a><span class="co">#&gt; 4   A28_30       1    Inflammation</span></span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a><span class="co">#&gt; 5   A57_59       1    Inflammation</span></span></code></pre></div>
</div>
</div>
<div id="uniform-manifold-approximation-and-projection-for-dimension-reduction-umap-olink_umap_plot" class="section level2">
<h2>Uniform Manifold Approximation and Projection for Dimension
Reduction (UMAP) (olink_umap_plot)</h2>
<p>Computes a manifold approximation and projection using umap::umap and
plots the two specified components. Unique sample names are required and
imputation by the median is done for assays with missingness &lt;10% for
multi-plate projects and &lt;5% for single plate projects.</p>
<p>If byPanel = TRUE, the data processing (imputation of missing values
etc) and subsequent UMAP is performed separately per panel. A faceted
plot is printed, while the individual ggplot objects are returned.</p>
<p>The arguments outlierDefX and outlierDefY can be used to identify
outliers in the UMAP results. Samples more than +/-outlierDef[X,Y]
standard deviations from the mean of the plotted UMAP component will be
labelled. Both arguments have to be specified. NOTE: UMAP is a
non-linear data transformation that might not accurately preserve the
properties of the data. Distances in the UMAP plane should therefore be
interpreted with caution</p>
<div id="function-arguments-selection-1" class="section level3">
<h3>Function arguments (selection)</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain SampleID,
NPX and column that will be used for grouping/coloring.</li>
<li>color_g: Character value indicating the column name that should be
used as fill color. Default QC_Warning.</li>
<li>x_val: Integer indicating which principal component to plot along
the x-axis. Default 1.</li>
<li>y_val: Integer indicating which principal component to plot along
the y-axis. Default 2.</li>
<li>config: Object of class umap.config, specifying the parameters for
the UMAP algorithm.</li>
<li>label_samples: Logical. If TRUE, points are replaced with SampleID.
Default FALSE.</li>
<li>drop_assays: Logical. All assays with any missing values will be
dropped. Takes precedence over sample drop.</li>
<li>drop_samples: Logical. All samples with any missing values will be
dropped.</li>
<li>byPanel: Logical. Perform the UMAP per panel (default FALSE)</li>
<li>outlierDefX: (Optional) The number standard deviations along the
UMAP dimension plotted on the x-axis that defines an outlier.</li>
<li>outlierDefY: (Optional) The number standard deviations along the
UMAP dimension plotted on the y-axis that defines an outlier.</li>
<li>OutlierLines: Logical. Draw dashed lines at +/-outlierDef[X,Y]
standard deviations from the mean of the plotted UMAP dimensions
(default FALSE)</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
<li>quiet: Logical. Default: False. If TRUE, the resulting plot is not
printed.</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">requireNamespace</span>(<span class="st">&quot;umap&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>) ){</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(SampleID, <span class="st">&#39;CONTROL_SAMPLE&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>  <span class="fu">olink_umap_plot</span>(<span class="at">df =</span> .,</span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>                 <span class="at">color_g =</span> <span class="st">&quot;QC_Warning&quot;</span>, <span class="at">byPanel =</span> <span class="cn">TRUE</span>)  </span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a>  }</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABRFBMVEUAAAAAx+EzMzNNTU1NTW5NTY5NbqtNjshuTU1uTW5uTY5ujo5zc3Nzc4xzc6RzjIxzjKRzjLxzpKRzpLxzpNOMc3OMc4yMc6SMjHOMjIyMjKSMjLyMpKSMpLyMpNOMvNOMvOmOTU2OTW6OyP+kc3Okc4ykc6SkjHOkjIykpHOkpIykpKSkvLykvNOkvOmk09Ok0+mk0/+rbk2r5P+8jHO8jIy8jKS8pHO8pIy8pKS8vIy8vKS807y80+m86em86f/Ijk3I///TpHPTpIzTpKTTvIzTvKTTvLzT06TT09PT0+nT6bzT6enT6f/T/9PT/+nT///kq27kyMjk///pvIzpvKTp06Tp07zp6bzp6dPp6enp6f/p/9Pp/+np///+HwT/yI7/06T/09P/5Kv/6bz/6dP/6en//8j//9P//+T//+n////x8urzAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAeOElEQVR4nO2d+X8k5XGHPZZvZgVyJDBgm10jIA6xw4KISex4YchNsHdhgsnBLhOEV6j//9/Tb5/vUVVv1Xv0Ib3fD0gz3e9RVf10vcdoer9VFRVF6FtzG1C0bhWAiqJUACqKUgGoKEoFoKIoFYCKolQAKopSAagoSgWgoigVgIqiVAAqilIBqChKBaCiKK0FoMN2u73zoH19fVG/OXmsnd017/bbu83ZU7iJq/P7vL42mwhDGz15EtvCarQOgK4vjh9W1eWZAqRmSf3a9Tgp7dXpavem4ujq/G5UX5tG1sHLs61GsEdPGtlHd4rrznYA8csXHsptXYDWAVALSH0Z73c/FFNaDmqOXb32u+frK3z5PO8yY4IBai7vgUcQDNBe2bt781SZHof4orQKgIakom7dfQfO/z20ChxOvri438G2q/NFw9Rvt8d/OP+bc/VODWFX3eumxJ3fAkBsNhBBLUDXF5xB8MkTkCDVxNXrf3i9/vlaHOKL0ioAGpLK4fghPMVpyDptfvX/72s6rs5r2q7O61f1uxag9nW1qzE7QGOSDyA1GTtt5mSq8gEY1xCAVPeXL3/x7oOa9MddM5cvvXH80Qt/v60HtrqHy/aVGi//goXqErQKgIah46AwgPJ/fU2aq9v+am5xNaw1hZsf9bsWoPZ1Ow7uoCGJGsKOH6pGlBWvNRx0v6zC8BBW7e5W+/a/06prRpmhZnZ1wwqg9pU6t98WgBLKD1Cdo9TFrH/12erQj1rDjxag9vXhGJ3T+CfRdU/dig5e2CEAHeoM2TC+u983o4xt/n+hAah9dWiyZgEoofxDWJ13VNy7X/UNvD3+6IwAaI8DBC7jxzWSQqkZHNtJVjejsgQu4y9f/t/X+2lQ14xqtvu//U+92reOsCIzv1YB0DiJrmPbT6IPxpXb3d0rsHan6tewWgvJQKAGgBTL3SS4baP/5df1u797+XH3s2sGBKhkoPRSCaNet39+dhdexteXsZ12Hp574353TQ/UENY2spcDpJqup04KADVxaX8xG9m9qO6D/Zt3x2YggMocKIN2DUHt9nOzGXd9YV78yxebYe7y7M4wf66LoQDhqzBYA0BqF/yVdjGGrcJQtQmvyZxdMyBAanj78WpW+isBqFn21mrudjXx2Fprny4hdb/26rru2pU7DJDaBzr+V3bymFrr2SpaC0CdPkwaV/7oM6Vk4+LcWhlAqdR+9hH5qVku7SXj4txaM0Dt5gz/Q05dzZC4lonqkrVmgIoWoAJQUZQKQEVRKgAVRakAVBSlAlBRlApARVEqABVFqQBUFKUCUFGUCkBFUSoAFUWpAFQUpQJQUZQKQEVRKgAVRakAVBSlMID+9Jfb7falj/ECl2en9hdvSP3ZeLeD/iJYNSdpcjZdnVtffbx6Y7t9K9jyPztfYVqUQgBqHvCkhP9JsRCg/zQfjnPDAFIPAvl9qOUqNEt2OwSg3fa5/6gv6Gdn+N8iK4AkLR7zAJK0OZtsgJqnYwVbD8ZiQQoA6LAdvlyMUlIAGtUYXgAaNTwU5fpf6kRUffaTOkP/SoXodL89/rj6up4f/fzzYQj7+r2z7fZnarq0O/nive2dX1fX79UFVDS/fq+v2T5zp2+pDtofP+znWEP9cQj7up5TvExMwGZWA5D6IvYbytv2+x+nreWjh1YsnNh0JdvQWJEc216A5ABZ99K+nw5dXzx7Vuem9rs2z/YAdV+9aZ4advIb9fKti2761J067QEaWqoB+mX/bZ2x/gBQe2i5N2YHUO+NBpDmoRUL+31fUgNIj4RvCjqh5ACZGfrqvE461eW5urjtd9a3rzyu/nS27QDaqRvq+kOVYHbbk4/VmfrnZ9vm1CvNqfttnh5bqs8c/36o1NfvAWr6uf6n5X6pqwfoVBFzqg1hhod2LIz3esl+AqVHYmh7fsUCVOu//+3vfrJVF7Z9kmpz9tBd8eYZc908shn62lLqVFeyed5PN9B3LXWDpCqq1e8BcvpfmjqAlEuN+focyPBwjIXzXivZAWREYmh7fkUPYV1m7ecn3aN4VRBbSto82z+WsKvdnuo2A7oojS31PO3VM+CG+j1A/aHFqp8DPdbnz/oo1HuolbHfmyWtSGrl5lfMJLpqnn+x/emv/v2/ziMAUsmpGcL6lm4qQLaHOEBWyRsG0OVZZ7l60T4n7EoDCB/CbIBGENqn9fQt3YwhzAVI95AGyCrpDGHrBkhtJP5D7cA/n23Vg5bq+Vy9rh52iXf4JNoM0vXFnV+rla0iThEztqQ6+JiaRL9V1T0sFiMCoNFDH0BjyTo0ziR65QDpH2U0D3tqxqHOoXZk+qm1jFcpBRnn1al61Xo6ttQv47WJURdFrd5yH4BCDWH6mE0PYX3JvbWM73laM0DV9Wcv1q68pLYRm129Z9/a9SNzuz1obCTWTv/8fyoXoOZUu8d4pW62oSW1kVif+tnQXFPf3Eh8bvkbie4kWveQnkSPJVVovjAjuX6Aiop6FYCKolQAKopSAagoSgWgoigVgIqiVAAqilIBqChKBaCiKBWAiqJUACqKkhygghytWxafAlBq3bL4FIBS65bFpwCUWrcsPqsHCPgnlufVxPGZ2/+VAwT+I+/zatL4bDZzR2C9ADVRmzt8gGYBaL4YrBWgjaG5rdE1ZXxkQcgSp1UBpEWgAKS0EYQhU6RWBJAegeXyM9cQVgDyZlgtApI7b2qliw9rVOIGIlesFgOQIAJGwBaGT7r4sCc2KwDoqx8cHb0qqiFXKECLU5r4iDz0FG5PzDiEPb33avXld94R1CAFeoHGYDzk8IM2NqvSAsR0jyhrhi2Jcbo47n75vU+rb97uU1BkgFBOwBP6QRsgvLGplT5Dy5MskX+y3mxcd1UWqqpv18oDEHLcPGpMoanGplXiDK2UbJDOPtpz3X1UZyFRDTKhoimIVdpKQLMTlDRDd7ppAD367geiGqjRuD+Cgc1MRbMDpNRmaKWUACVqJmeIeO4+GhJ0IoDgo8y8xD49odoMnWCIH1J3Er8WApCWf3g1yDzjnNjgbq4FIGmGNuV8RJPEJrK9ZH1w3P3qh+9o76IAAk+MAMk9XQQ+8gyty/B8GoAS9sJx9/0jJdEY70lAG+jYQpJJgMQZ2pDueYY4OIvYanKAAmr0FpIbO8ahFfMjz9C6DN8zBiJTL9k+C0M2PyHL1w5QQIbWZPqeLw5rA0gJNBMbkf3+rAOwZc2BrF6crdho5QSIDzqvzDpyVBxA8A0WadLYy00FiN9amrZyKm4ZD51M47c9UMa32GiyISzWYgaNy8Ar9Z+7pAco6a04EUDxNnsBWkqGChjC6LPJJohZJurZARpfRdqMNpF+XI+SMKI+s3GA9GMs53lrMGEMMwLkTNpSpCD08CZFHynEiw9/2eXxm24F+pCEDJRxihPMGQAKvsbEp7MrA2i01G81ByCkFaAM0aLdEi+c+QCyoOFYH6Cxl6UQlBog8MaxnPUBtNnoFYgemcU0rQkg+sO1FQGkW2rbzLPfdpbgxxLdiVaOGc5phrCKO4NjNwcdXgQ/cQAxXXBuFjZA3EXf8gCyDBR3e5MAwqeqoQDR6RnAyNcy25bMy3joWAw/9EpjEQSJAXKOswnilRIB5EycfH1M/s1U1CyPrSLf6UKeApESLuOto0kB6rohCQKTOmmkqRm+2kyuxqlqvBIJGItSVHz45gmcIACKD8dSvhvfeUI4cxMAYqRH3Dx05e2ziEhANwYgOL06hVhNcAoFmskRHh9W1/QOX0ibg8/jZodzKlwRAMVeBnf1EeWNCKCMCEUCJKyMtolSAoU90CalYIBiewZzaXCj7MozAkT27DMIqYy1aSFDdBsbiywA8dYHgJOB/ggq509BIQD5LbpxAA2W991DM3vvlHhj5CD2Et8uI4IvN0JYRKleGfYgRSh+9EtCtEt261UsQJ2dur3QMUfgZeQFCanGhkJWWiwtouAQTVhENUvGZmOBkPcWMRQ5hAGwDAdpH+CAchI8ARDLgakAMroge+SZ46ls1J+MnzQAGR5AxwCNp588oTu0GgLalYYrPUFf/eiD/iUIkKdD7ZzcplUAhAfIhaV76b1G7eknjai+0wOUPMc/vec+XAGIiR8guBBppxN/Rp2E4gEEBajSY+TebdrwjKrNPz6AvENYFRCupPw8OnrGvcHMi+q/m1CzWPdhUn8k4j3eBQqQEghQHw2WV0+ecFMQ9jZMSQP+yaddhjaeDwTwE7RL5auZkR9GkyyA4ABVnrHb71Z9jgOQdxkfpKTx5s2BvJ1CKHjDmA0gVqMRc6Cuk2AD2tMUPzNl5QDB8XHmbd4rAhRiVGOgKReTyliA/CZ4TuMAZUvMGeSJTwxAE66oIEN8PecEiJOfmlHMU2AF8sWHTRCAS54ocAfTeQEyLHIO0PZxPViEcgKUY0k+9CLZnQQ1EUBwWEj7aBcWhpU/Pu5aFVMi1xibR7hBJkBUU8sFiEpRk14JoYj4TDgYcbM7foNq+FBNae7ut9vt/ebF8UPC3hCAEFN5E0r0FKdPmZ2kUsQn42BEH4amVRRB2gkuQPs7D6qr89NqQoBCK7GaSw1QzvjgYu49wketbQTgPBEk5h7o4O71xd3m58njHAEKupqe2yM4fwUpb3wQBbtpHnbLdAc4IeICdHXepOdqd/J48QBxmksNUN74IAq/T8ajSKZhTG9YRlgZqNbuNEuAgi4lvciki9MJWqzc8YFEeWB/SGKd62tudBHNeK3AzmpzoC4sV+fb2ADlXv74bjrtbboclDA+XOEeDMeBIhA6cYFgLuP37RqjvtfiAiSyNsQtJCD2weQEJYqPQETi8ALkEpTQsFEZvlgosDbMMTggztG8gUMVChCZaMDj8DTGzTxUFKKjkx4gyWULvMBEAgJSkLj5OIUBhKMCvQkACGovxVYHtJHIrgHKDxAUB5HgWlgopa1jCo9P9EzVKWK+hsoh+EC3mNc4SuZG4uWZP0KxQ5h2OhQg90ZC+k2Zf4Ljw/GREwkMIKytrgwBTHj8R1nL+P3JY3YNTHyA9Lkgw1ZPg6ybOLiX8PgkAshJJv6SvmmiACDuRuKBXGCYNdi9gUO09iY4k5qZh3kJ5N00ksdHlGa5CQjY8mGWdE5seJZ5rMsCENU5dCOFXVmDvlFsS2SSxmfoLBggOH/4DhGHrRMLAwi1w+7cNYbvCNQwG6DAbjoFA8Tt18sGCpB/OQGdEISDKpoIILwDq3PIlrArO9TqK/samRQgrbMY9+xjUBkYIbRZf2XQkmUBBJdxq1J2iAGadAjT3Q4haKhDfeOJzYCnNmfujpXSANoOEm/VU10A/CB3ElGPbhjbEWHbyJA0PlpnMQCN31mhpstBXiUGiK18ALEmiGiBkDVMDgEP4ArpWuPnCdpEFECs6t4iEA7Xfyv+sJDshJ5CV/DV5URGyz++4pGhNsSMjz3dkHWiA/QEb6J3KhKh8AIuDvuAIYy/JKTuJLBFZoMeQDae8wIFxkfajQkQYvzGUEjHvph4g2a5e6hH+DsPyC7jABoSBlAdapJqxOyZC1AUQXB8vnn7CHx6SYx0gNDbqT+KOeZ3eGiAOMsGqJ0n3qX6q+g5EPPyACUFmxngBFVAEMdCUGh83v+++q9TGoB6Y8k5kNfz/jCdY3z7RmTERnd3dXBO6yDNAxBczF9fDhDHQEh4fJ7+4h3ti3NpAaqGNVgAQP6w6OUIIwjZf1QfBpBkCMPmzJyaQP3upT9QkQAR8VHwKIjcx99EyDGWGqIQz5gAoSVEADXDe2gGigVIdmkhgFjehuNTUfEZAVLKBRBVyJdC6LbwAgwjDHd3yBjPmSRKlpLA3SXKDWZx9l6Q24ykdIXGJwSgmBEcKkTtwTFTkOTEKMtdNU90VxmpJ4lwAoKWZ1DdsH0VrxEsgfGRA5TEAZZY4SIKeK103T04+xwZJonIpiHTVafQ8FacxuSXMUV8pgOI01uUNaC7/2gGKP0Yj+35eBMRCJBWix8JRmZHZcVHnKEjug4Tay8oTBwcUq8yXIONBESEFz4dB1CCy/j0nmwjkdFzUro4/IR2qP1NNPppc+oMBNlrLKRkAA0HZJEQAkTEx1SKISxpguJ1Fg1QPT/EPsNIDBATEE8BoL0wgLhm4/ExhcdHXzjyAcq/WhjCF9K67m59k4GbQIkn0akBChvC5MXR+JjC4mP2xloqMGz020+F0yoRgpDl7g5Mz4mX8ZSt3qi1c224PXEQAnaCvH+SyAOIkn456VqcNjlw+BbARGXH3T3w9UvpJNEjbk6VtGi/yCYoPobQjVamV0Y68NRiNclKL1QZsjbg7uVZ9qdPsLYLgypOoLD4mFeINaA4teg2Oe15S+HnkIq2u2qxQWdp+ffCYgtoJTn3cFbIwuOjGc+BYuPUwsuSFrP4oTqi65vuHhjPD5ABhPbNipBekFMWKJMUp5j4CAEy3/rbxMRLQDyAoPO6uzv1cbNXSQAaD3ucc+9GVsA83YcpNj4MP9zZbCqAPFZXJh7Gl4m6BuB2tKdz+BcYVg2GMPOHoz7/pACBN7DEYlTp4oP6MWaLjXMIbYvXmdcoXdY/gMMDKOlOq9Ez0O142BsflyCsJFohCUFp4kNn3eFK2YfCnQiq7PwLShvcjAyPuAP6xU5UvhzBBWg8Y5RJCBBbRHzMuwY5DW5yTQqQ+48AzgQQ4TtywbE2tLfmb1970/PDAQgZeTwAMVcaaJ9skQDZhYGvNnueoQQHCM/IRKzwqm5BN69jPZElgxUZH8dKvABWTVqvgubkLAkSEOCu5ym2xLcyqMPANIjqxC7oySskqtIOPQqJT28BGqeKvNg+COjwyx3XvlBtGA51A7h7fUEuVkMAik4ERgOAK94O0qWikPiQRugXh9yJoZsllrpyWd9FkwHkeRCgWwNruz01lmAaD8jswO3OE+EEFuiSxsdjBH5xnDLkSXypGygYIKs9ECDhMhX3f7iv+K5wppe6ZxUHn7TLsbT/XhhxcZxSopOxPhv10auYACAfPxKAPA1ZpYzmGe3ODxAgFkDmeWv0ZkxBnVoCw7D2WiUYwrwACZYCPID6YBix5xDEsICh4CHMlD0iU/bpBYyiQ4DRStAbnmlu0HkAiSeJmG1jXLjWI4H0HB7OasMa18YAhU+iddlAeOzTzgu48KUPui+3CtAA4O5Oukz1AVSx15IxAFFLhbY0xwKGxPEBpdvZL+NZxQ0X/eQZ9SVXYTCMVpqNRKQjzyVFawgP8wCiDGWIjo/4X7WWhaWtAdSkm3ESkGAawTVtms/CMHOQpII0hB8eE66nO9k1Y+vpPemf/MoBsqoazXjLARkvkWlZASJWf+1J4DgXNeewDKCkHD06ekaagSJgjgXI27PItMwAVRTQiRMCkx94ehilTz4VD2FRJgwTFMpd55wJnc+yWQCiPMGOp7uO8wE0zIHYX/2ON2Djcxc75w27zLaEAEkTTWqAyGW8YQlxcwb2LM1AaQES9eELu/CyTACQZ5EmqBArBKCYvPTo6Eh95VIIUPy9401A8hs6zLZ0AMmDEuhhjKC9uCiAOs0GEHmvyaYU3POm5gQo1EORkE1FJwH57mdaYQBJe9FrxASJu2fJ0SRDmExRVxJsCzzh9jcVQPKeus0uKG2mkdmSpF0mQJwAsV3yFIq8lnKb+kJp+mQDxG5xGKzMWqnXr2F1eQAxd1qZ+NjW2tXSAcRrZ2KAAvJPbIJkdRBWlwVQyE4rKvdGcoGioiWfpvtrRC/jNbEB4nZm8pMBoKiWWTgMO60JnpHoWAsZj0cr6OYNNjZEgjkQz7KJElBOgEI2yvAzprWI9ZhL/XGgAt5bmKWBSg4QNgdKp5iWswBEGgQmIC5A2M0Ip7HxZ5ClYZLsA0lSUJVtfzUrQEE7rYBB+PoTzzVoyxyAeEGZCSDpsGH5GlDJUyQ8BjkykBsc833Ilbbb9qYxVrOyy8hT5k/jJTcGa79EboKhOQByy/OM0Nv2AQQixbA0XgsACIhQShMMZZwDbbQEmfA62Ts2wC4t1CVsQXp+5N/KsF97KrHJ8JVLc1Wy/EHZaH4/kU12oewENPRjtw+nJLi5BHaNCtjmEFghAGPFADnXGbq+4o61hvWtv6ELuKBlENBgmCGo8gLE2phg8TPtECat4QEo2HQ7MFSg6CEtm+QRFdmWLAEtGyBgqHHPiruWAYQYFNCrSHEASdbedGscVxNEYyqAwJP+RpCq+Hu/QYyyUYoZwpJY6GtklkVnQA1s833DfdiCVcaYNGOFzMJQe9kJmhwguwbdSOoYTPDFQuAQywujEJ7kGWnK7Jpvf5CCPmw2V6sSE6EadP5ZEUCAK1yA7FIiv9G0NAE/MX/ukgogRhcB1sHK/8VCQ4P9spWoLLJY4cUDFECQtMaNAYhXbKPNDOIByrDpAygkPmaenQAguYmYpgBIvOq2EpCQn6lSDaLQZbz7WlB5tgSUDiDUKDc8AoDMd+EATYhTASikBmGWdopNglFIik9bh2ldEn31g6OjV/s34oha7kUv4xmdCXuglBQg3+5LGEBCp6dffj2992r15XeE/6p1yG461Qq7wiIBIsAwzvAHo6DdZq2sWz8fQV9+79Pqm7f7FCR/uEKwdUEVE8chMUAoQf5ylOT7HG4GyzsNUlmoleDPXYy3dAWiHYGVGZR4CPMAlGEZidhhdDJBoB/VWYj9tScAcX9xvBnojN+IRMo+B6osdyYEKHASFaJHkmckSvOwFKBJE1PCZXxrtWx6k1xwCsrXZ/utlUfDFDrDwxWs0t7MmjnJW0r6dI5w+FP5i/CTWVr+CZkD8Us7LNmtTO9+4p3oMNOJWmGbatMC9NUP39HexX7xEi8OsIHjs06AmLbbBdBKgZEGusip94+UhMt4QfsuQEjtOfLvDAA5NxFeSz/OmVxNm3lgxT69xNGm+9sFLx1zJKA5hjCjDOmydoLf8A0DaHTJi8cM+MwBkOki6bMQoEVsrOUDyJ9hZuAn/Z9zMK8yEBO0rF3J07Sn/9xKC5DuOBugpBZ4xHI37tNmW3ACootyQzM7PikB0ifO4xEyFtP7z3E36NNmXIb7Hi4s0mK7nkKpAELTzRx5BhfH3YBPm2lxP6Zyxrr4rvMrO0DLigXX3fbT5gTPSHRE5x89VyXuOJMSxYeY7qwSoPbTZkmNeC0rV3OVHiDoZJpOEojprujT5lS61QDNsiYPEO8ZibJPm5NpBfFzlRygRO1lEu9B48JPm1NpDQF0lHIZvwL/Oe6KP21Op6WHD1DS+Czff467AZ8232LdsvhM/NXmW6BbFp8CUGrdsvgUgFLrlsUnACBQ34YPT1k2bbPhEY21bR3FwwGC9e35y+YyIYWE/a2oeAFoEq2ICGHxAtAkWhERwuK3bMpXlFoFoKIoFYCKolQAKopSAagoSokAMr634S/9ow/8hWp98/bRd3klBY1KjU0hcY98Z2RBErbNsDwNQOb3NvylmR6//331X+JGhcamkLhHvjOyIEnbZlieBiDzexsePTp6hncPPP3FO+zbhd2o0NgkkvYocEYUJGnbHMvTzYHGpwR69MmnTIdVMRWfpI22YhubTJIeJc5IgiRtuxVteTqAxu9teJUBIOHQLjA2kWQ95gNIGKjKZ3kygB4JRtbZAZIYm0bCHhcEkMfyeIDc7234ys4OEM/YNJKEp5JGqMoPkM/yRBlIdoMxXZDND/klF59/JM5IJ9HSsd5neRqAzO9t+IszXRCtUNlxERqbQPIeBRdZuowXAeS3PA1A5vc2vOK68PRejo1EobEJJO9RcJFFQRK2zbC8fJRRFKUCUFGUCkBFUSoAFUWpAFQUpQJQUZQWA9D1xWnze3fy+Ppie7d5fdiePG6Ode9320ZtweryhYcz2DmXlhqfhQLUB6b5ffn8Xx0/7M7VP6/Ou1/Htxeg5cRnmQD9+LUH9cur13/ZhGJ//NHZ/e6c+nW480DdfdvbC9CC4rNMgE53KicfTn6jIlK/7U52Abqsw3XY3j3cXoAWFJ+FAnRQodjdbSKibqi9uqmMO6z+dYsBWk58FgqQmgBevfagiYj6cXV+txrH+LbkbQZoOfFZKEDXF3UWPnmsInJ51sRGRaNbZbRLjlsN0HLis1CAqv1pnaGbW2rfRmV7f0jRnW4zQMuJz2IAqnan/S8VoMsX/vjugz5aVZeXbzNAS43PcgDaN/6qwVzF5Prir+thvo5INyNspom3GqCFxmc5ALVzPzWWNzfVbtveUn1Q1FB/qwFaaHyWA1Adl3ok7zY21EZYM6h/3m6RVU3objVAC43PggAqWqMKQEVRKgAVRakAVBSlAlBRlApARVEqABVFqQBUFKUCUFGUCkBFUfp/w9iR/qRenDsAAAAASUVORK5CYII=" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(SampleID, <span class="st">&#39;CONTROL_SAMPLE&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>  <span class="fu">olink_umap_plot</span>(<span class="at">df =</span> .,</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>                 <span class="at">color_g =</span> <span class="st">&quot;QC_Warning&quot;</span>, <span class="at">byPanel =</span> <span class="cn">TRUE</span>)  </span></code></pre></div>
</div>
<div id="function-output-16" class="section level3">
<h3>Function output</h3>
<p>A list of objects of class <em>ggplot</em> (silently returned). Plots
are also printed unless option <code>quiet = TRUE</code> is set.</p>
</div>
</div>
</div>
<div id="visualization" class="section level1">
<h1>Visualization</h1>
<div id="boxplots-for-outcomes-olink_boxplot" class="section level2">
<h2>Boxplots for outcomes (olink_boxplot)</h2>
<p>The olink_boxplot function is used to generate boxplots of NPX values
stratified on a variable for a given list of proteins. olink_boxplot
uses the functions <em>ggplot</em> and <em>geom_boxplot</em> of the R
library <em>ggplot2</em>.</p>
<div id="function-arguments-16" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt and a grouping variable.</li>
<li>variable: Single character value indicating the column name to use
as a grouping variable in the x axis.</li>
<li>olinkid_list: Character vector of OlinkID’s that should be used for
the boxplot. If not specified, all assays in df are used.</li>
<li>posthoc_results: Data frame from ANOVA posthoc analysis. This data
frame need to be generated using the olink_anova_posthoc()
function.</li>
<li>ttest_results: Data frame from ttest analysis. This data frame need
to be generated using the olink_ttest() function.</li>
<li>verbose: Logical. Default: False. Flag indicating if plots shall be
printed additionally to assigned to a list variable.</li>
<li>number_of_proteins_per_plot: Number of boxplots to include in the
facets plot. Default 6.</li>
</ul>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>plot <span class="ot">&lt;-</span> npx_data1 <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> <span class="co"># removing missing values which exists for Site</span></span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>  <span class="fu">olink_boxplot</span>(<span class="at">variable =</span> <span class="st">&quot;Site&quot;</span>, </span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>                <span class="at">olinkid_list =</span> <span class="fu">c</span>(<span class="st">&quot;OID00488&quot;</span>, <span class="st">&quot;OID01276&quot;</span>),</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>                <span class="at">number_of_proteins_per_plot  =</span> <span class="dv">2</span>)</span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>plot[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABCFBMVEUAAAAAVZ4Ax+EHcYMZGUgZGXEZSJcZcboaGhozMzNIGRlIGUhIGXFIl91NTU1NTW5NTY5NbqtNjshuTU1uTW5uTY5ubk1ubqtujo5uq+RxGRlxcXFxut1xuv9zc3Nzc4xzc6RzjLxzpNOMc3OMvOmOTU2OTW6OTY6XSBmXcRmX3f+kc3OkpIykpKSkpNOk0/+rbk2rbm6r5P+6cRm63Ze6/7q6//+8jHO86f/Ijk3I///TpHPTpKTT6bzT///dl0jdurrd///kq27kyMjk///pvIzp09Pp6bzp///+HwT/unH/xwD/yI7/06T/3Zf/5Kv/6bz//7r//8j//9P//93//+T//+n////0QMtPAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAQCUlEQVR4nO2dCZ/UthmHaRsCS8sEyFVomZAsV9m2SaANgUmZhDJtumHYZTfg7/9NasnHK9uy/UqyfEj/5we7Xls+Xs0zOmzLvpAA4MCFqQ8ALBsIBJyAQMAJCAScgEDACQgEnDAV6GDphBxbV3C+MBbIy1GMR6dAox2FJyCQfyDQwEAg3rJFAIH8A4EGBgLxli0CCOQfCDQwEIi3bBFAIP9AoIEZXKDjywV/fpj9vpoteHL5Up7gdvrHhz8117pNi9U1xbJLlYmk2IB2kfXx9+f/21vpfvJDp2k1mDd/+E5zPKzwypW32Zw3V7Jkt9nhBSGQQOZS8utDmbHbLOfeXPnTB4/KhTWBnmS5dKlcfPE7uUox4+2tS8qEnCk2oF1kf/y9sW3FZ/7rQxmHMq0E8/aWPPLa8bDCK1cWWz4ufHl7K934cfo/S20fnC/8C1R4dPEfV66WC6sCHSvfTiWH5fSvD2U2f/ConEjktz/dQLblrViUT7gdf19sb7IA5GeqTlMwaUkhjrx2PKzwypUzm7ZZkuRJuuTtravKHIcAfDCWQCJbimqnLlCeMPut5rCYkX1S6c9yIv25/fAvEwiUH1Xy40+16SKY9OC2TYF44ZUrv7kiMi9fWS4+5gTGCcAH/gWSX1I5K5vdFKgs7EXGqjksfmf5mX4Jy4k0W3//6AlVYfnHku3H5fh7YlN3UNmZGsxWqcLKxhInPFq5TFv82l78563o20AiJ7KcbAqUVwfZ166ew9m20lXLiXTDV/MNiHbm1USdcDn+XoEuaac1AlWOhxeesnKSt7DK6uxyumJW13UTnEBlLyLPRJlx7gJti3bHE5mxadpywun4hxOocjwWAhVt6C211zkVWXACpRmbF+Tby2XP1bkKSyuwbAPZJyMX5RNux98TW9GWqU83Baoej3kVti1PfWSZJ9XJm+K2wfnCexUmu5+5RkozktWIFllftDJvlxO5jB88KnZzqZxwO35uI1pUypXpukDV4+GFV64sev25P7l6x1ELJLO4CL6cduzGT1ICFXWRjEud7imBuN34QqBtaUruVba541i78eL0WPF9FV8p3YnErfZMG81oni2coA3EOZGoawMxw0sq9kmKb4uYL3oOLsH5wr9A8jx8EfyTvOKpnMxPsl6LeqKamt+dlzK2xdWEcsLl+Ptjk1cX8v3QtBpMXgvVjocVXr5yWUNXTkWyOpnBCDRjcDF1YCAQb9kigED+gUADA4F4yxYBBPIPBBoYCMRbtgiWINCFqUfvuhJybIsQyMtRzIOQY/MGBCJCjs0bEIgIOTZvQCAi5Ni8AYGIkGPzBgQiQo7NGxCICDk2b0AgIuTYvDGeQNOc5zIBAlkwmkBTnSk1AAJZAIEIfWwzP+ipgUAEBLIAbSACAlmAXhgBgSyAQAQEsgACEb4FClJFCERAIAsgEAGBLIBABASyAAIREMgCCERAIAsgEAGBLIBABASyAAIREMgCCERAIAtmJtCkV1whkAXzEmjaez4gkAUQiJiFQEuzDAIREMiCeQmENhAEWjAQyAIIREAgCyAQAYEsgEAEBLIAAhEQyAIIREAgCyAQAYEsgEAEBLJgRIFmnzUQyIKIBTr59Hn64+PV6vqrbAYEsiBegc7uXUsFen2D5kAgC6IV6OXqj6IEenaTZkEgC6IV6N+vRBX27tvHNGs5As0oK4cRaGlRS4RAZ/c/W30kHTo4OIBAFnQL1GhnhibQySePk7MHz7MZEMiCToGa7czQBBKc3c+rMQhkQZdAmnYmBHIgOoHq7cwDt3bCjKKWiOBep0Xsyefu54HmmGgc+ttA1M5sT7+0qCWyBHq5Wl3Lm0A+YjN/O9zSsrJfoEo7MySBGvgQ6DetRCOQoGwmzKOY9wQEsgACERDIgn6BKu1MCKQHAulptDMhkB4I5JY+jF4sBLIAAhHhC+Qh4yAQAYEsgEAEBLIAAhEQyAIIRAQrEPs6igUQiAhXoN/WUTdwfrRer++8SKc2X/6c7O62bGSz/ko3GwIRcQq0XwtlNl98L/86P2oR6PTwb3mSKhCIiFMgUewIcbLypVWg3Z1/HeoWQSAiSoHeP5UCSTZf/vdQ1ma7tFKr2iIE21BKAgIRUQqUVmGlF6khsgTapZXVabW82a+/lv8bQCAiXIE6e2H7tLjJmje5QJk7O9muLtikfxXVXAUIRIQrUGcvLJEdMaFQLtBe6nR6qJQ3mVObilMZEIiIVyDRFErlyAUSLSCBIlAxq9mMhkBElAIVBY0odiolkMr5kWwnqQ3uAghERClQ0bDZUQmU1VeqRkXzeddsRkMgIlyBuhrRWS9Mnk4semHirGKlxVw0fjTNaAhEBCtQ93ryUoYsbuSljOI8kKIKdek3jdoNAhGRCuQGBCIgkAXhCmSeWeELZMbpYd55115FzYFABASyAAIREMgCW4G6nhrgPWsgECfROFgL1Dg31XGePF+lZfZsnmARvkAzakQPKFB7Ho+cybMWyPxrZrBxByAQUebFcGXigAL9rpVwBWLVTnMUaKgvBwRqTc8T6H/tQCDDLTkI5FLr9RGxQPLRI+++0TziLjiB2nM/4Q3r2TTuECqIV6DsGcbPboh/1djiEog1rCe7HK+5myNegbJnGItHrxWPYYtUINawnkwg3U3R0QqUPcP4JJcoqbzqICaBeMN6CoGGu6V18QLlj6AtBUpiLYFYw3ryKkwzMAwCxSBQZy+MM6wnb0QPOCoDArF2NhOBukogQe+wnqwEOj1EG0jhBI1oomdYT35TtGZgWOQCzaIbz7py4ksg1rCeUqDmkoUIxLooUsu1vlCkQGf3/J5IZLnxSyv+BWIN6ykFGm5c2NgCtefxLy7FlDa28ATq+tJxhvXkAu25I1N/aDa26+kHFIiTyYsViFWF+Rao+zD7h/WwLmWc/72YOj0sm0qNy0UQyGxnSxDIDRLoKK/zNtTbb14umup2Dgg0e4HSjpwotU4PqezSXC6CQGY7G1AgRlYOjPGwnv36i7+qiTWXiw7QiDba2aIFYlFpRGeFkELjZC16YbR9zic6syrMA6pAO1ECVc41QqAOgRixRSVQ2pv7Sv5U+voQKCyBxu6FNS8XQSDaPgQSMM4DOXTjIVA8Aqn8UBFIvVyEE4m0fQgkmP5aWJW22RDIQSCfPX9FoJ3+Yoc+PQRakkAdO+e+rUem6rylNbsRts+gqQQyP48GgTqOQlmF97aenUj1/mnH/UDvn8obYXXv09Clh0ChCMQa1pPfYp8/L1qbaedHovDZa26b1qZHFRaIQCbDepLkP+03lEGgOAViDevRlDyNTAtAIIt6Tp8XwQnUmR2MYT3a9/TUMi0EgdrzmHWuxOz5QEsSqKsEEvQN64FAEKhToL5hPbq3rNQzTfbz160DEOvpWdXFYgUargpjqchK5Ekgo2E9mudzeH1KKwTixDatQLxhPXmTWlMSTf+Iu1qsLbODFsh7FdaV+6xhPYwTiaZVGEegWhh9cXYkgkDTDuvJb5DmPZ1joxk+1kgPgYIRyI2GQKmPXdfDIFCZCAIJ6gLtu+qvBAItWyAzbN7Ws9NVc7r0ECh8gVjUh/V0NH8q6SEQBJKoAvU0fyrpIRAEkigC9TV/KukhEASSKHck1nv+nekh0BIF8tkLw4lECGSD10sZVWYp0MnHq9X1V9XYIJABQQlkfukteX2DpoMViJ8d5sQu0LObzdjCE6hju0Zv69Esm5tALfiqwt59Wzw14sD0XRmhCGTwtp7RHzReC8NwV5V1PQl0dv+z1Uf1J4/MUyDz0rVjB8oqBm/rSWVrnCiMXKCTTx4nZw+snlQ/skCsLbF3QBswGtYz9g1ltTAMd1VZ15NAAst3ZZgXCbMUyORtPbp3HSxFIM7HBYHadtB5hKy39SxdINaWLAR6fe15cvL5Es4DsbbE3kFtA8y39UCgJi9Xtu/KCEkg7tt6IFAnUQpkNKxHM8AQAhGzHtbD2lLbDjoEMnpbz5DdeO9RdxCYQJ4SKcm7PiKjt/UMdyKxdoScVSGQQ2wWibjrDfS2HiYQqEwUikBuQCACAlkAgQjfArnEZpHIGZthPb1AoDJR8AKxYAik3rQHgWhvEEjAEEi9aQ8C0d4gkIAhkHrTXiQCcc5xQSBJv0B00157+sGibj0LydnSYAJxdrZIgSbphdFNe+IL6FegjvPYjC1BIF/rddAvUOWmPc8lEAQKUCBB842FVSDQrAViX660YF4CoQ3kR6DO7DAY1qN5+E+/QJWb9uLohXF2FoxABsN6dDBKIPWmPQhE8wMRyGRYjwZcyiCiFMjsbT1NIBCxHIFMt9ldhRkM62kyokDDAYEMt9nZC2MN62neZ58zM4HQCxu9BBKwh/U0mZdAOA80jUDsYT1NIBARpUBmb+tpEpRA5iNFKkQpkNGwHg3zEsitDWSRqIKhQG62tjFRL4w1rEfHzATiMQuBHHfmf0vKNjsVNxjWo7k7epECsYBAo2wcAhEQyAIIRIQvkBlzG9YzMhBoFCAQAYEsgEAEBLIAAhEQyIJhBJojvE/i3TeaR9yZbwcC+Uo/HbxP4tkN8S8DAlkQuUBitMDJp/UHjZtvZ2kf+3BELpCQJxty4n3QZKBAIM2Dxs23A4F8pZ8OCDQKEAgCORG5QGhEuxK5QPPrxi+N2AU6uzezE4lLI3aBVCCQBRCIgEAWQCACAlkAgQgIZAEEIiCQBRCIgEAWQCACAlkAgQgIZAEEIiCQBRCIgEAWQCDCITYI5Cv9dECgUYBABASyAAIREMgCCERAIAsYmaaOvYNAQ+0sFBiZpt60B4GG2lko9Gda5bZhCDTUzkKhP9MqAxcg0FA7CwUTgdpfeTlHINAooAQiIJAFEIiAQBagEU1AIAvQjScgkAWMTFPH3kGgoXYWCriUQUAgCyAQAYEsgEAEBLIAAhEQyAIIREAgCyAQAYEsCFcgc0KOzRsQiAg5Nm9AICLk2LwRu0AnH69W119l06HFNgqxC/T6Bk2HFtsoxC7Qs5s0HVpsoxC5QO++LW50WtTdljMicoHO7n+2+mh5N8vNiMgFOvnkcXL2oPNJ9aCTiAV6uVrJFnTPuzJAJ8YCLRxNSCTQ0nGVwYbIv3Wvrz1PTj5/NfVhLJjIBRL1GN2vC8yJXSDgCAQCTkAg4AQEAk5AIOAEBAJOxCfQ+dF6vb7zQkx8nST7r6c+noUTnUD7tVBm88X38i8pEXAgOoE2X4mf75/KXxDImdgEKsxJpDynh+v1lz8nu7RSuzvlUS2Y2ARKq7DCIFH6yBJol9Znp4cwyIroBEpEqSPa0KVA50fCnb2cB0yJT6BE1GPrtNApBNrLBvXpIVpDNkQpUMrmzotSoHUGBLIhNoGKgiYtdqolELAjNoGKXtieSiB05V2ITaC0wkr77enPu9SAFr2w8tQiMCM6gbJLGcIWWfJsivNA6ITZEZ9AYFAgEHACAgEnIBBwAgIBJyAQcAICAScgEHACAgEnIBBw4v8NA3SNkxGVpgAAAABJRU5ErkJggg==" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>anova_posthoc_results<span class="ot">&lt;-</span>npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  <span class="fu">olink_anova_posthoc</span>(<span class="at">olinkid_list =</span> <span class="fu">c</span>(<span class="st">&quot;OID00488&quot;</span>, <span class="st">&quot;OID01276&quot;</span>),</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>                      <span class="at">variable =</span> <span class="st">&#39;Site&#39;</span>,</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>                      <span class="at">effect =</span> <span class="st">&#39;Site&#39;</span>)</span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> npx_data1 <span class="sc">%&gt;%</span></span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> <span class="co"># removing missing values which exists for Site</span></span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>  <span class="fu">olink_boxplot</span>(<span class="at">variable =</span> <span class="st">&quot;Site&quot;</span>, </span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a>                <span class="at">olinkid_list =</span> <span class="fu">c</span>(<span class="st">&quot;OID00488&quot;</span>, <span class="st">&quot;OID01276&quot;</span>),</span>
<span id="cb24-11"><a href="#cb24-11" tabindex="-1"></a>                <span class="at">number_of_proteins_per_plot  =</span> <span class="dv">2</span>,</span>
<span id="cb24-12"><a href="#cb24-12" tabindex="-1"></a>                <span class="at">posthoc_results =</span> anova_posthoc_results)</span>
<span id="cb24-13"><a href="#cb24-13" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" tabindex="-1"></a>plot2[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABNVBMVEUAAAAAAGYAOpAAVZ4AZrYAx+EHcYMZGUgZGXEZSJcZcboaGhozMzM6AGY6kJA6kNtIGRlIGUhIGXFIl91NTU1NTW5NTY5NbqtNjshmAABmZmZmtv9uTU1uTW5uTY5ubk1ubqtujo5uq+RxGRlxcXFxut1xuv9zc3Nzc4xzc6RzjLxzpNOMc3OMvOmOTU2OTW6OTY6OyP+QZgCXSBmXcRmX3f+kc3OkpIykpKSkpNOk0/+rbk2rbm6r5P+2ZgC2//+6cRm63Ze6/7q6//+8jHO86f/Ijk3I///TpHPTpKTT6bzT///dl0jdurrd///kq27kyMjk///pvIzp09Pp6bzp///+HwT/tmb/unH/xwD/yI7/06T/25D/3Zf/5Kv/6bz//7r//8j//9P//93//+T//+n///+cMpL2AAAACXBIWXMAAA7DAAAOwwHHb6hkAAARHUlEQVR4nO2dj5/WNh3HUadjgIryMFA3mUN5xnaDIYI6dhtzjD3K45BH5273yB3Hjv7/f4JN+iPp0zT9JmnaJP28X9vRa/O0/ebeT5qkTXomA8CBM1OfAIgbCAScgEDACQgEnIBAwAkIBJwwFehC7KQcmy44XxgL5OUsxkMr0Ghn4QkI5B8INDAQiLYtCiCQfyDQwEAg2rYogED+gUADA4Fo26IAAvkHAg3M4AIdnK/4w93i30vFhofnz5UJbuS/vPF1+1M3xGb5k2zbucZCVu1Aucn6/Pvz/8X1/DjlqYtlOZjnv/xCcT6k8OoPr4s1zy8WyW6Qw0tCIAbPpez7uzxj10XOPb/4+9fv1Rt3BHpY5NK5evPZL/hHqhUvrp+TFvhKtgPlJvvz741tzf7m39/lcUjLUjAvrvMz3zkfUnj1h9meDypfXlzPd36Q/1+ktg/OF/4Fqjw6+/nFS/XGpkAH0rdTymG+/P1dns2v36sXMv7tz3dQ7HnNNpULbuffF9vzIgD+N5WXRTB5ScHOfOd8SOHVHy5sWhdJsof5lhfXL0lrHALwwVgCsWypLjs3dgQqExb/yjnMVhR/qfxnvZD/XL/xpwkEKs8q++fXO8tVMPnJrdsC0cKrP/z8Isu88sN88wElMEoAPvAvEP+S8lXF6rZAdWHPMlbOYfZvkZ/5l7BeyLP1F/ceiktY+WcpjuNy/j2xyQdoHEwOZi1dwurKEiU88eE6bfXP+uzfrs++DsRyosjJtkDl5aD42u3mcLGv/KP1Qr7jS+UOWD3zUiYvuJx/r0DnlMsKgRrnQwtP+nBW1rDqy9n5/IPFtU5PcgLVrYgyE3nGuQu0ruodD3nG5mnrBafzH06gxvlYCFTVodeivk65kCUnUJ6xZUG+Pl+3XJ0vYfkFrNhB8Zfhm8oFt/Pvia2qy+wutwVqno/5JWxdd30UmcfVKaviOlIUqGh+lhpJ1UhSJZplfVXLvFEvlDK+fq86zLl6we38qZVodlFuLO8K1DwfWnj1h1mrv/SnVO9g1gLxLK6Cr5cdm/GTlEDVtYjHJS/3lEDUZnwl0Lo2pfSq2N3BXJvxrHus+r6yr5SqI3Gt7GkTK9q9hRPUgSgdiao6EDG8rGEfp/q2sPWs5eASnC/8C8T74avgH5YXnkZnfla0WuSOalH91t7KWFd3E+oFl/Pvj43fXSiPI5blYMqr0M75kMIrP1xfoRtdkaRGZjICBQxupg4MBKJtiwII5B8INDAQiLYtCiCQfyDQwEAg2rYoiEGgM1OP3nUl5diiEMjLWYRByrF5AwIJUo7NGxBIkHJs3oBAgpRj8wYEEqQcmzd0mXb01mJxp/jnyjNC+thJOTZvaDLt5Nad7PDyfnZ4lZZ+cL59bfenXwaKbezTnhZNph3mxc7pgzvZ43dp6Qfm2zO//tmvfi7/9H3EQWIb/7SnpSfT8lLo9JP98hfWVTVmCfTpDz7b+emXgWIb+7SnpSfTnl55dnL72uJy5dCYl7Bvf/jH15o/PTNMbKOf9rToM+3pm19lR2/vZycffUVKrz/UuFicoOfY7HcfMNqonlYlz8ntqghKMxcKUo7NG7pMY+VPAQQCHWgyjV27cg7ZZewd9AMNThoNfU2mPV4w7mRPF4u6KIJAQ5FKQx+3MgQjx5ZGQx8CCcaNLZGGPgQSjBvbp7/5308/K39GDAQSpNzH5Q0IJEg5Nm9AIEHKsXkDAglSjs0bEEiQcmzegECClGPzBgQSpBybNyCQIOXYvAGBBCnH5g0IJEg5Nm9AIEHKsXkjLoH8jpgZP7bYHwbK4hLI94iZsWOL/2GgLC6BfI+YGT+22B8GyiITyPOImdFji/5hoCwygeRHaDw8SDN6bNE/DJSFJ9CUz8wEEZvncxic0ASakpRj8wYEEqQcmzcgkCDl2LwBgQQpx+YNCCRIOTZvQCBByrF5AwIJUo7NGxBIkHJs3iDM0nr6AJMrgC4Is7Q+vsr+608/JYM8FxFobGHTP0srm1zq6LcDTHHnD5vnIhTFa5CxhU7/LK1MnuBnKDN+LkJVvAYaW9j0z9IqBBp7ml865s9FqIrXMGMLHMIsrRGUQHbPRTSK12C/HIHTP0trgAIN9FxEo3jNAoktNvpnaY2gEm1Hq3hNKLbx6J+lNYpmvAXt4jWd2EaEMEvrya0UOxIVxWsysY3JXG9lqIrXVGIblbkKpCpeU4ltVMYTaJrXmpuQvkAe/gKjCcR6IQM3CAJZAIEE6tgCP2kjIJBXIJAFqAMJkhXoQpvB9j3XVpiKdAX68S5yUC/vL5fLm0/ypdX732SbDzt2slp+oFoNgQS+BZpMRa1A2yVTZvXel/y3l/c7BDre+0uZpAkEEsxTIFbsMHGK8qVToM3Nf+ypNkEgwSwFevWIC8RZvf+fPX412+QXtaYtTLCVSCmAQIJZCpRfwmovckN4CbTJL1bHzfJmu/yY/98CAgnSFUjbCtvmxU1RvSkFKtzZ8Hp1xSr/rbrMNYBAgnQF0rbCMt4QYwqVAm25Tsd7UnlTOLVqOFUAgQTzFYhVhXI5SoFYDYghCVStalejAxNo0t7GWQpUFTSs2GmUQDIv7/N6klzhrghLoGnvd8xSoKpisxElUHG9kjWqqs+bdjUaAgmCEMhH+NpKdNEK492JVSuM9So2asxV5UdRjYZAgmQF0u+c38rgxQ2/lVH1A0mqiCb9qnV1C0sg1IEmEMiNwASaFAhkAQQSpC+QGcd7ZeNdeRe1BAIJIJAFEEgAgSyAQAIIZAEEEqQvUNyV6OC/WxDIAggkgEAWQCBBPALZntPoAvGJK9h0lFeekdJrgUDjJpKSTzWs5+QWm3jg8Kq0yiGTIdC4iaTk/93FdFjPqvWEUIV2hrLF71gJ9Pjd3vQQiEKYApGG9RS34xVPc2gF+hefQvD0k31pHQSKJpGUXCcQaVhPIZDqoej+OtDJ7WuLy9yh7ml+IVDXeg1mezJPJCXXCEQb1lMJZPpIKxOIzeV18pF+EiYI1LX+R50EIhBtWE95CVMMDCO0wrKsd5rf6RuoQ5CuQM7DespKtPGoDAiUhkDaVlhGGNZTlEDHezZ1oEM2nfI7+n4gCNS1Pg6Beof1lA9FKwaGEUqgp4veaX7DboRQmaVApGE9tUDtLcPcyoBAXeuDF4g0rKcWyNe4MAjUtT4MgVyH9ZQCbakjU//ermxr00OgsAXSf65/WA/pVsbLv1ZLx3uK1lorff+JTZzInJkK5IYQ6H55zVupWvvt9A0gUNf6+QiUN+RYqXW8p55LsZ1eBgJ1rQ9LIDOMh/Vsl+/9WZs4g0AdJCkQiUamFYUQPX0NBOpaPyuBNqwE0lWAMgjUAQTirbkP+M+umaZ30stAoK71YQmEVphBInMgkAXoBxJAIAvQEy2AQBbgXpggWYFGGtazUd/s6EwvnyHlUCEmapCuQNqD097Ww1NpH2ktHoTtMwgCGSWKQCDa23o2LNWrR5rngV494g/Cqt6noUzfPMOeCIJN1GCeApGG9ZSP2JfzRctIzXhW+Gx7+hHr9LrxKmZZM9yeLBI1mKVAJsN6suzf3Q+UmQrUmrtaM49++RFqdJNkcha4QOZfM91ZNC5h/cN6FCVPhVeBSIVLgAINVyYOKNBPOukXSBsGYViP8j09BX4Faj2Lq3gqdzKB+IiB0wdixIAQaKjSNRSB+g7eN6wHAikoph55fJX914xtfgJlPcN6VG9ZKZEEWtYQbmVEL1Ax9QgbMVmNnpynQEbDehTzc9j2REcvUDH1yFEpUSO2WQlEG9ZTVqkVJdFsBSpnjqgFuiDNPJKcQLpKNGlYD6EjcW6XsF2BsoRLIP3OKcN6ygekabNzrHSPlEEgo4PFIJAbLYFyH3X3wxITyHslmtKjREsUiUDbnoeiJxJIl8e2lnFxfDfjSW5818nAAplh87aeTfMy1+pr8yEQJZM1efydk0Ant6bvSAxVIBK7w3oa1Z92X1tKAnXnBQSiIwu0W/1R9LVNdS8MApEuhhMgCdSq/ij62iCQ2cHmJNBGMSq11dd2YapKdHgCkf6igV3CfLbClB2JriVQ0gIRYpuTQEogEATqgTJTvX0lGgIFIZDPahNhllY046MXSJtnBm/rUWwjCCT3tXkQaCdWahZAoIEEMnhbj/lE422mEsi8FQuBNAeQPmLwtp5cttZ9UgjUjm1WAhkN6wnygbKdWDtWd+cxLmGulzDy23pU7zqAQO3Y0hNIW2qT3tYzC4EsrnPK2IidzBEJ1PelI76tJ3mBuvOY1NnmQyDSnkiJfAqUEd/WM6RAlKijFWiwSxgltmkFMhrWoxhg6FcgUtY0Y+1YnbRAhNi8CWT0tp6xm/E7YVD2D4F8CKT7+hq9rWfkjsSdMCj7h0AeBNKfwUBv6yECgepEqQjkBgRqxwaBDIBA7djSFcgMm2E9vUCgOlHyApGAQO3YIJABEKgdGwQyICmBzPstlbFBIAMgUDu2dAWaQSusA1zCIBAEchPIrXTVn4EDEKgd24AjU5tBmAc3TFa6ONdHLAJR/lwQqOsA2uwwGNajmOMuEoFIe0r6EkbaU9cBdNlhMKxHBQQSzFMgk2E9CiCQIOgnEkl76jqAJjvM3tbTxusTiTthGB6qm8QE8pRISq6/hBkM62ljK9DOGVI+CoEcYrNIJCXXfsdJw3raz9mXQCBBugL1ZQd5WE8bCCTwLZBLbBaJpOT92UEc1tOGINDRW4vFlWfa9BAoWoHM3tbThiDQ4VXpFwhUJ0pDIKNhPQoIAj1+tzf9YFHTOtm9C0RpYSYikMmwHhX9Ap1+si/95lcgTScAYU+DCUQ5WFQCaVthBsN6FE9H9wt0cvva4vJ+dR4QqF4fj0DOn9PQL9DR2/vZyUet90k0gEAQSEt7mt8midWBKAeLUiAzBhzWM5pANCCQwwEGpl+gwze/yo7emVU/EOVgEKiAUAI9XSim+W0CgcYVKCBGvJVBAXUgCOQCWmEQyAkIBIGccBTI/CG3BoYCuR2sCwjkhFMdyCJRg5AbCMESmEADAoFGAQIJIJAFEEgAgSyAQAIIZMHcBTp9ILrZIZAFwwgUIrS/hPw+TwhkwcwFarxRGAJZMHOBGu80D0Kg2IBApUDeH9dNFAiEEsgJCASBnJi5QOFVomNj5gKF14yPjbkLdHIrsI7E2Ji7QDIQyAIIJIBAFkAgAQSyAAIJIJAFEEgAgSyAQAIIZAEEEkAgCyCQwCE2COQr/XRAoFGAQAIIZAEh0+THhiHQUAdLBcosrdL9Rgg01MFSgTLJpvTEAwQa6mCpQJhkU37qs+OxzxCBQKNgIhAtfShAoFGAQAIIZAEEEkAgC1CJFkAgC9CMF0AgCwiZJj82DIGGOlgq4FaGAAJZAIEEEMgCCCSAQBakK5A5KcfmDQgkSDk2b0AgQcqxeQMCCVKOzRsQSJBybN6AQIKUY/MGBBKkHJs3IJAg5di8AYEEKcfmDQgkSDk2bxgLFDkpxzbJFwDfOuAEBAJOQCDgBAQCTkAg4AQEAk5AIOAEBAJOzE+gl/eXy+XNJ2zh4yzbfjz1+UTO7ATaLpkyq/e+5L9xiYADsxNo9QH7+eoR/wcCOTM3gSpzMi7P8d5y+f432Sa/qH045VlFzNwEyi9hlUGs9OEl0Ca/nh3vwSArZidQxkodVoeuBXp5n7mz5euAKfMTKGPXsWVe6FQCbXmF+ngPtSEbZilQzurmk1qgZQEEsmFuAlUFTV7sNEsgYMfcBKpaYVtRAqEp78LcBMovWHm7Pf/5oahAs1ZY3bUIzJidQMWtDGYLL3lWVT8QGmF2zE8gMCgQCDgBgYATEAg4AYGAExAIOAGBgBMQCDgBgYATEAg48X+UGWHwzHVRvwAAAABJRU5ErkJggg==" style="display: block; margin: auto;" /></p>
</div>
<div id="function-output-17" class="section level3">
<h3>Function output</h3>
<p>A list of objects of class <em>ggplot</em>.</p>
<p><strong>Note:</strong> Please note that plots will not appear in the
<em>plots</em> panel of <em>Rstudio</em> if not assigned to a variable
and printing it (see sample code above).</p>
</div>
</div>
<div id="boxplots-for-qc-olink_dist_plot" class="section level2">
<h2>Boxplots for QC (olink_dist_plot)</h2>
<p>The olink_dist_plot function generates boxplots of NPX values for
each sample, faceted by Olink panel. This is used as an initial QC step
to identify potential outliers. olink_dist_plot uses the functions
<em>ggplot</em> and <em>geom_boxplot</em> of the R library
<em>ggplot2</em>.</p>
<div id="function-arguments-17" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain SampleID,
NPX and Panel.</li>
<li>color_g: Character value indicating the column name that should be
used as fill color. Default: QC_Warning.</li>
</ul>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>  <span class="fu">filter</span>(Panel <span class="sc">==</span> <span class="st">&#39;Olink Cardiometabolic&#39;</span>) <span class="sc">%&gt;%</span> <span class="co"># For this example only plotting one panel.</span></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  <span class="fu">olink_dist_plot</span>() <span class="sc">+</span></span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="co"># Due to the number of samples one can remove the text or rotate it</span></span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABNVBMVEUAAAAAx+EZGUgZGXEZSJcZcZcZcboaGhozMzNIGRlIl5dIl91NTU1NTW5NTY5NbqtNjshuTU1uTW5uTY5ujo5xGRlxurpxuv9zc3Nzc4xzc6RzjKRzjLxzpKRzpLxzpNOMc3OMc4yMc6SMjIyMjKSMjLyMvNOMvOmOTU2OTW6OyP+XSBmXcRmXl0iX3f+kc3Okc4ykc6SkjHOkpIykpKSkpNOkvLyk09Ok0/+rbk2r5P+6cRm6//+8jHO8jIy8vIy8vKS807y86f/Ijk3I///TpHPTpIzTpKTT06TT6bzT/9PT///dl0jdunHdurrd///kq27kyMjk///pvIzp6bzp/9Pp/+np///+HwT/unH/yI7/06T/09P/3Zf/5Kv/6bz//7r//8j//9P//93//+T//+n////x0ZZ9AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAY/UlEQVR4nO2dC4PctnHHdY2ro2q16UvR6qrmzk7cc622Z6duc46iNFaybRypjeptIytb6aSzD9//I5QAX4PB4EGCJEDu/G3t8TEAhpgfByB3l3tLsFgRupXaAdayxQCxosQAsaLEALGixACxosQAsaLUF6CClZkmwSJcvQGaxAvWYKUOCAO0cKUOCAO0cKUOCAO0cKUOCAO0cKUOCAO0cKUOCAO0cKUOCAO0cKUOyOIBenvv+Pj4vd9Z9j4u9zy27m31/H1Y4fto7+s790MqSaPUAVk6QM+Py3B/++iP/oneHRZ3jRkaoGyVOiALB+j1HRXtt/csoDBAU2vhAD3+3s/V3/+SnDyuB7PHf/zouExJ5ep7/9AMYc/Lfe9LFn5QDnn3Xx0fq5xVbX195/i4rKcu/vben9xTtqDQ/bYSW6pLptQBWTZAWuaRMf720W0Jkgy/ZOv5cQ3Q4zLwMlu9vqNIee933z4qtz6vt6qk0xQvJ1X3q8zWFaoAkubPa2KzUeqALB2g2+1yNdDIAKu0VK3Wk+hqpCsBUAuvSkKkYb31ez+XALXFqzq73bKQAkjtMAe4xEodkPUAJORkuhqL1HBTjVEVQK/USgmCwkStlYS0W2sqquJv792vtmqFFIU5ToVSB2TZAKmBqNFjGf7nLUBqrHECJKc4UhVATXEGqJeWDVA7iX4uJ8b1gGTNQOXQRWQgUY1LbfEKoHKfVogBsmjhANWX8TITqXh/+6gByDYHAgBVW8s1BVBTvBrPbHMgNUvPSakDsnCAwI3E13duy9UGIHUN9Uq/CrstNIDUVolFNYmui7+9V198aYWaq7BXuV3Hpw7I0gFSN3GqtzJeyYX/lDeAqomRnNT8o3EfCAKktt4WFThN8bf3fsD3gcK1eIAOXakDwgAtXKkDwgAtXKkDwgAtXKkDwgAtXKkDwgAtXKkD0hegW6m/yctCmgSLHkBMbM9auRggVpQYIFaUGCBWlBggVpQYIFaUGCBWlBggVpQYoCyU/obgUDFAOSiHW8oDxQDlIAaIFSUGiBWnxfLDALHixACxosQAsaLEALGixACxosQAsaLEALGidLAALffOS17KG6Dporzge795aQqARgvNhFFmgEbSBACNFxsGKH8dKkBJ5kBrZHYGgCK6DRadv/fHbnGVWW/6OdBI3RZRzcCCo8ebARpgnwFAQ0syQCFigCZo0VHjqPXloBnuA43UbTEj2OAxbFiLh6S8bySOIwZhQs0DUA4hzMGHFWoWgNLNHrt25/bhUHhdN0Cg4Zl9WOUVF6WRAaJ7jQFar4Y8I9G+19ZtM3cmSc2Yb/EGGc2AUAaQDnrIpnVveKdRdhO8heatsn+bgcc4CkGeCnJIc4kAogwjukPPMz0qGtBmxDH2ZcpnvlyArH73SEBjAVTAkkWrwMJzAuRwbdjcMXeA3vzF3bufCPHdF3f/7Netfc9T3KJJAeoxLg1pMxhQOgFRReHmgt4c1EICOQB69/En4g9/+jPxqz+X/0N754GFZSfKalh3EAD1KT2I2R5JCBcjSuqbR/okzFxyAPSH739dZp9P3v39z8Sbv/o1tHd1oCtLD+4OZx4XeA40WDEDcIidkx8KoBgnZ5JnDlRmIQmPhAjaGwehZQB/mu4rH0BBhv5WeiSWvgC5Uzasz3/ZmBVBHoBefP/rDiDpOm3fHFXLT0qA7C74Wgku1Rs0b84Gq+7mlgXQi3L2TGUgcnQXboBicm8fgJw+uFsZPTYwAYXOGgtjm56g1IrakAVJToBelFNoGiB81mCAmq0jeUnNG8h9AwEq8FQqWqqisAzkLmkULprL4DxykQugF+rq3ZhEt0cATLWO0pYH+kWdldpeetIZB5DfkT41TgRQuyV7gN78ZZV20GU85TxYxQA1/RHkjS3l9wFoyBzIDtDQKAUAZGvRsF8qQL+6K/VJeSUGbySS9+o8ANkP1VZNDEADFARQYLi6A/DMgVwA+ebNeu0p1f/jHGRfwIBqneYGyJrIPAC55kBBFOF4BABEH4SNC2zd9YrNB8fmLEixaMDngcL6Ec6BPABR5ynqfkcfasyEAVSEANT5bjiHD8D0x3HIziW84lEGZE34iUR79sdJyjtKFNZ4TAaQ0SLpQwBA5G7bkmnvkqNXZtOIAJHzPGovIgZbUlXPDRDlk/IVtiO6QzGG1Go7LIJKRgLkzuwzaiqA6mPzAdSuwghYRoohAIEJjFbEARBgGuMiDIDa2Sw27gLsBKjGjzZwiQFCqyAAuKQYPgcCCUKfk2s4ae0VQE2VFEDmAWgmYQDRxxqiAvCXVHMCROWYQgCAiiGdGgsQag/hMw1AzUttMBigDDQ2QAgSGiBUqI1YJEAF9mEwQBhxwxECIH1HC5Ce7JqTpFkZBlCBHEipKQAqisIIiwsgIbSuxOjhFVuzBEDUHEjrfRPYAhfxA6TXDgHCvmOARHfaBB2rcQjpKZoAIDgIDAGI6BoPQIUZGb99iwb2FDgErNBeWyP9MlBXBDXhPvqVAaR1/VQANdFEV1PCDhBhKqA9RAMD1CUmq5ULoNYRfeg03GxyrQUgYIgbWRFAMBgF6Hp9IPIC1CVzEqCGRyJbWAAypj2aIz6AwLw4ACA8hAUC1B7NQQGkn9gUQNQbVX6A9Lq8AKFuTgoQxNoCENzHABlB1zrCFTdTNEBGI2kA6vYVhlUoQJqnEQBBXtYGkNYRTc+DIczsJFhcr0sIgTN/04iAtXoBslyJBwDUgSpg/VqzNEDAC70zGKDOPgigAp2VEQB1OUkr5AcILJEAQZcMTzF9IKBG4Q7rkQDSqhFaEcKlxJoIoG6bGAUgWJ3QQoBjivdp9m6A0D4B9tkB6kytACEQMF6GI4cNEESIBggV1+tSf+AExAcQHjzMsz8tQNhUe0GZKwCgwnApsQYDBI4JnUntGCZ6AKT1sBsg6yAwOkBmtHoChNwl2w4GyGKVWEPuA00DEOwVAyC9uh4AIV9JUwtA1Ok+OkAuNA4VIOO9MEuHC9HUJYjORQABax0gW+S74cPFBwMUqykAUptxArAD5AhsDEB6cVc7RZCBGzNB1yXcLQYA5GJ6mQAZw8IwgIraKCCmToDsNQhYnAGaSBMAVAOCRhACIGf4GaAQn9cBkKAAavYJuj+FAZ4HoA6kUID6BsMbqAEAeVpkgOg4iNaA6rGRACrWB1B/nxcLkAEEBigkWgxQtMFhA9RrEs0ArREgy0GL0K7xdZIeOhKg2aLlBmhA5Bmg4Qc9FkD+GhigibU0gLABA2TTfrPZPHhaLd88KVcevgR7t2pttzlXe0/pKq4vLgOA6KdxAeqFmUgLkMsqP4Bunpw8E+LqTAJSsiT/bBucpHZyt9h+Ljm6vjjvSYEGRF/7+QFyGeQB0GSHNxigCpCSoMv6RTIFcpDadv3plx+UUF198NRSSxAQfe3z6mEGiFSbVLbl6LSrwfm/Z8hg//D3Ty5r2LblIKeY+uXm5LcXP72Qa3IIu66XlcWDXz7ArC0coLmjlaFLpNqksj95Rk9xFFmn6k/zb1fScX1R0nZ9US6VaxVA1bLYlpjtNwxQJo1NDNC+CfReYkBNcfYPX96U6af+c/3p02pYU8bqpVyrAKqWq3FwywBl0lhygMocJakp/zTZat+MWu1LBVC1vFeTqj0DlEljEwPkH8LKvFNmn+ZPeUG/OfnqzAHQjgHKqrGJAeom0SUdzSR6v4G3dbbnOwnW9lT+aa/WOAMxQEoyYZTX7f9zdk5fxpcs/FBegYn93/zkUiWqCjArQFUlOwYok8amBkhdNN08qW4/qxuJN0/04F/9SA1zV2cP2vlzaWYFiK/C8mpscoDUWxml1MhTsoHeymgTUv1nJ9/32FZX7jRA8j7QyW9OnqFmGKClu+TRv0fcZTa1Z4AyaWw2gMZS9d6HeUeAAVq6S/0oqMY1cyrjlxoSzXfnSYD+A+cpaJ9VDzNAqdUCdP0vzdLVmTHQQfusepgBSq0OoIs6rW03Dn4YoOxcSqxuCLt5slG3JDeWj6c19ln1MAOUWnAOtN88+Gff9IoBys2lxNIm0VUS8thn1cMMUGpBgHYyA2kTIPVrzd99of9mak49vD6A8NfElwPQ9YVMP+oNkUbvPpbk4F9tXkQcXFaWbyPm4XhxtFCAiKuwF3f/usxA+HfjE/Rw+6SPkQA6mhUgu/fU5lCAwNNP3JqYMdd9oP/+WoLzpoaocjoNQEeBADmf9oJj1AV21C+4mi5Zvac2BwPU1OrTXABBtXeidYAq+8wA0s5vS26xANTWWi011Vnb6eF4ARuLBIh2KWuAWi0BILgvHKDCBpCtBrXDM5YGAOSdIzftAIDaLe1LEQBQYWg8aoAAQDvzzbLxAMJdnwigyo8qphpAjX8egNxjaQhAbf0WGJt2WlLgliPCe5uKI6SpAZKfVqw//NjqTY9JtDvH464fFyAUfjI2WnhMgI5MgMzRsSLNEvkCpA4DIBNQdFCaQdXOEQVQoSqsZ1autOIAqHpTfsA78i6Abp7Iy/ed/rE1BU7gZXzTNaMBBCa5BkDdSNCxUPc5ZW9gQAPU1YCabXbooSOOvw16h5TeGBiPmi21NIOjyqUWoAKw075UtQ4B6EP6A/JRAFVPYkCfOFMAvfvYdiOROEWLrlf0eB/Vw7DzkgdWCAJFAHREAHTUtAJjYwOodQmWPCrgFERzAGUGi0s6QHpjOocdQBCIzkAD6AizA2odDNDNE/+zN2IBouxhvPHojkcCkJeP9PRAIQhD1gKk5wpMK4opDAbEjAJIP5F1gFpPOy+IdrQxCw07BgtG/SRAXTshAH0TDZD8lNip+rCYTEf7AeNaPEAYEg0g1EkQIDO3QGZcAOETGbEDotsCVBTdFnIksAFkGsDGUCrsUqBWF3IO5TCTMM3UcDcYIEMIoDLWMujyu6ufPpXfca7/zA4Q6ggtQxgANdsoNLpoUQC1vWCcyCRA+oBiDycBEEBdCywBUJMmtKyBYSSdowCysDZ+BoKT6BKb+kFSIc+TGhMgeCaBDkRDhd657VUtmLd0AtMFAiCYw7o4ewDqzmJ7OBFA3blvTR0WBPsC5Ekw0wH0YRNliZJ6Jkf1bJcL6lPPwQBtWrk/0loU+NTROtAEyIipDhCZFHQ8IUCFUaEHIITI0GhNAZCjsVkAkl+hVw/maFOHN4NYAQq2h5DQHRgCUIcG2TVFYQOIgCQYoKhoOQCihrVFACRx2T5Qz+coF+s/8wCkRRKfgV0fUp3bmdgBgnFwJSsnQONGywWQ1cqa8/IASD588++qi7ERrsKChzAznOYZGNQ1PQByZZnAmKYAaFqXhl6FjSkiA203jod23nKFs3fXBAHktGKAfJqGGwAEUpmJXBPxWwcbrQxdyhGgvfNbYRKg6DgsNFoZuhQE0MRCAO3wQ0AM+/g4LDRaGbqUHUDlpNz3zHIGKCOXcgPIM/2p7OPjsNBoZejSN8VEl1Y9BADyTX8q+/g4LDRaublUa0I2ggQ+kej9Vqqyj4/DEqOViUuGMroKC72RGB+HpUQruUtuHeV5Ge+1j49DntGa36UBQtUwQCsHKFLexjJ9K8NtzwCNpXiX3AB9g8QAjQbQXJq6l4YCJH/bqfkZQ+LKqfu8WQgQ4aaVfQ4AJdYUTM8KkPry1vbz0+bLXDHKCqD5CMguKc4KkEwx15/99rNnzccRIzQ7QOMAkCBaKwJIfv796qPf/6L6Ekb11Z6rH//k5KsP/21TDmwlX1fVkvzM9A/dXx+bBKDhYGQcrQxdGnwVtj0Xu+r/U1F/tUd+q13+yLP8aGsJULUk9+3c72/FATQckuVFK0OXBl+F7U/FVv3c5baCoxzJ5Aei1b8PFUDV0l79gOpkADkgWWG0MnRpMEBXH/3vZ800qP5qj5wX1f+q/+XSrvpNw7EBcpOy1mhl6NJggG5+8eVHL+vX+qs9JECTZCBjNDqQaGXo0vAbidsfqWexfH7efbWHAmiSOdChRitDl4YDVD3aRf2Iav3VHhIgObz9rftKP6v7QFlHK0OXZnkvzHOraN0AFfm5NBtAjWLeAgv4ruqqAcLP3UjpUhHdS4VhEPSR1qj3UHfe76oyQHMBJAirfgAZ37P/Jv0nWrMGqFgCQAVtEAaQr1kTIK1PQhLQ1BoVoLBTNMxUezJU32qKXgAVgY4DGYxqIab8ogAynzPQ8VGY+/RH7zUPB0yscQGCHVjYrHD/GOewBhBVDepm7ZyE0bIDpDMgupPbhZL2+EwvQAU8vKJ5jJnJB3jKVsfHUfMQRXgA7cM/tQdLJtY4ABW4A70Pb8PPHutqaAUfdqoTiR6USDwOs3m6HU1FAZ5T1T0tq34UJygH/DIAKvQWteMv2rrgQyT1Jzy2hIEHLhUwwYimcFsx/WjbxIp+M1U700XXNfbHR2p9B3uR6CTQr9qzT7sOBxwWyBFQTUtFSwxqEUVZRwkQqT2mum1RzzKN4+gRkNrDIDVajWeIai41h7kegMBZDZ+DWBjdDFI/StNaIgbnPAUQ0ftwtUntOhV6XWRhDUYiMnqhpkITM/ywUIIFXKFel8WKLLIOgGAC0A8fr8ARR4ugNVp+gIyuROG0JDKqsLbqBshqZQxTYUcW1Akh1STWoDdTgw/aTCvWE3lwB/paDI6pdZ8bMyN/MEBu+/GDnonVWI0xQG77rIKeI0Azu5RYDNDSXUosBmjpLiUWA7R0lxKLAVq6S4nFAC3dpcRaLkCCAVoIQN99YfvJy6Q9LJYFkJjKpcQKACjwR3ebTgo8aAboUAAK/dnvppMCD3otAFlwYYBaNb8d39inAIiyFz2bHStaDJCmPgDJ9w0nAUj4rCgD0bPZYQYCr3oMGCBDYRlIjAiQMKyErcUQgERMtGYGCPcEAxQNUKNxARLUkg8g0hFBm04AEFXDEgAKm0SLVQFkoYJypAdAphc9ALJ2QmKNdRkvMgfI2iJYgtZegIRhYAJkFHYBJHRvVgTQu48DbiQKR8cfDEDYpQCAqCJrAwjZuwESEwMkjBbNxqhVa4tgyQ1QpykAgrVjgAx4GSBBFB4KkICNWXmytmhW2BcgcykMIGFUY7rpAEg7/nQaByAlCiBhmJoACWwlUA9DA0F3oBYjK0DWaoYCpDlH7R0OEPaB7qX0igEIdW4/gIzC6QAygz46QOiQgwES5D7c4ymVGiABC6G+FoSBMHoY1eUFSBjV0EjgugTcZ8S2L0BaKwcJkNKsACFTvBeF0wSoq2Y+gIRxUGEAIR/Qoa8KIGEcjhGUHgAJsAHW6TTFdaG2MwQItdIDIK2x5QOk1A8gzY4BElTthg9oHwOkGQhghWulADJwMbd4ADJa9AJENkbHz+wOBkiznwYgrYZuOQOAhLUxd/wmBMiymkSTAoRWTYCQXAA5TIUgXIJeJAYIbWaALIdhAajb2gsg2xZ6RzxAxl67p9MCRPlq2cIACcPGu4XeMTpAns3DAaK8GQpQDpoNIDEuQKYdDRCuhQbI0pzH03iArFv6GaZU5gCFKgwEkSlAnhoGFJ1PQwBC6gsQLjeGRgLIUuu4AAVrpQAZsgBEGaYHCHnDAMVqBIAaJQCoaJ71Pw1A9vIBADkbj2k2Ly0aoPp5mmZdYQAF+8AA2XW4APXwISqSDFCovffo5gPIaLj/HrcVA9RoToDG7wDbHCi4WQYoVssGKLDihQI0bV0jaSUADW126nlwbOEJ6xpJ8wKUSKsBKEOlAqidvcyhpACtXIkA6q6f5lBsSwyQXQzQDOXXrIMAKFbL8XR+HcQcKFYLcnV2LeoqLBV12QKUwWm4JICSjXvJo2RRDhMBBiik4RSNBogB6icGCIkB6imeAyGl52dZAKXSkCPLILaziAEK0IAjy2F0mUUMUIAYILsYoAAxQHaNCNB6xXMgu0YB6FA6i2VqDIAOJl2zTDFArCgxQKwo8RyIFSW+CmNFiQFiRYkBYkWJAWJFiQFiRYkBYkWJAWJFqTdAt4ryP9eL1yBXq2W6NAkVfYDoXYL6+Sb44jXI1WqhLiUWA7R0lxKLAVq6S4mVfAxlLVsMECtKDBArSgwQK0oMECtKawbo+mKz2Zw8C7C7nMGblWrFAO03kovtg6c+QwYoQisGaHsqX2+enPoMGaAIrRcgjZxtOZiVlFyd/bQc1s6vzuTa1Qdfln/Pa4B2G7Us5L4NAxWs9QJUDmEtQTIZ7cqx7OqsfNnJedHu5FmJyqUa5yRA1d7z8t9lM/ixQrRigFQyUXPo60+fCoWGRKR5uVR/xe7hyxKg6wu5vD95tg+YdLOA1gyQkOPYpp5E79WodXYp2hf1V+wfPC0B2iurcsv1BRPUSysHqNRWDlhlKvrqDAP0gYSmBmhT6VIix3OgHlovQFWCUYS03NgB0i72Ay79WbXWC1BzFdZMbPbmECbnPdt6DgSTDl/Xh2u9AJXEPHwpX6tLq+uLegEAVCYaeBUmM49KRXvOQMFaMUDVWxmKhZ38uwVjWQXQv6r7Qd19oCpRbTbMT7jWDJBHzSSJFSMGiBUlBogVpQMGiDWGGCBWlBggVpQYIFaUGCBWlBggVpQYIFaU/h/qmqHkCDTfngAAAABJRU5ErkJggg==" style="display: block; margin: auto;" /></p>
</div>
<div id="function-output-18" class="section level3">
<h3>Function output</h3>
<p>A <em>ggplot</em> object.</p>
</div>
</div>
<div id="point-range-plot-for-lmer-olink_lmer_plot" class="section level2">
<h2>Point-range plot for LMER (olink_lmer_plot)</h2>
<p>The function olink_lmer_plot generates a point-range plot for a given
list of proteins based on linear mixed effect model. The points
illustrate the mean NPX level for each group and the error bars
illustrate 95% confidence intervals. Facets are labeled by the protein
name and corresponding OlinkID for the protein.</p>
<div id="function-arguments-18" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain protein
name (Assay), OlinkID, UniProt, Panel and 1-2 variables with at least 2
levels and subject ID.</li>
<li>variable: Single character value or character array. In case of
single character then that should represent a column in the df.
Otherwise, if length &gt; 1, the included variable names will be used in
crossed analyses. It can also accept the notations ‘:’ or ’*’.</li>
<li>outcome: Name of the column from df that contains the dependent
variable. Default: NPX.</li>
<li>random: Single character value or character array with random
effects.</li>
<li>covariates: Single character value or character array. Default:
NULL. Confounding factors to include in the analysis. In case of single
character then that should represent a column in the df. It can also
accept the notations ‘:’ or ’*’, while crossed analysis will not be
inferred from main effects.</li>
<li>x_axis_variable: Character. Which main effect to use as x-axis in
the plot.</li>
<li>col_variable: Character. If provided, the interaction effect
col_variable:x_axis_variable will be plotted with x_axis_variable on the
x-axis and col_variable as color.</li>
<li>number_of_proteins_per_plot: Number plots to include in the list of
point-range plots. Defaults to 6 plots per figure.</li>
<li>verbose: Logical. Default: True. If information about removed
samples, factor conversion and final model formula is to be printed to
the console.</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">olink_lmer_plot</span>(<span class="at">df =</span> npx_data1, </span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>                        <span class="at">olinkid_list =</span> <span class="fu">c</span>(<span class="st">&quot;OID01216&quot;</span>, <span class="st">&quot;OID01217&quot;</span>), </span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>                        <span class="at">variable =</span> <span class="fu">c</span>(<span class="st">&#39;Site&#39;</span>, <span class="st">&#39;Treatment&#39;</span>), </span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>                        <span class="at">x_axis_variable =</span>  <span class="st">&#39;Site&#39;</span>,</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>                        <span class="at">col_variable =</span> <span class="st">&#39;Treatment&#39;</span>,</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>                        <span class="at">random =</span> <span class="st">&#39;Subject&#39;</span>)</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>plot[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAwAAAAEgCAMAAAA0bOSjAAABJlBMVEUAAAAAx+EZGUgZGXEZSJcZcXEZcboaGhozMzNIGRlIGUhIGXFISBlIl91NTU1NTW5NTY5NbqtNjshuTU1uTW5uTY5ubk1ubqtujo5uq8huq+RxGRlxGUhxuv9zc3Nzc4xzc6RzjLxzpKRzpNOMc3OMc4yMc6SMjKSMvLyMvNOMvOmOTU2OTW6OTY6Ojo6OyP+XSBmX3f+kc3Okc4ykjHOkpIykpNOk09Ok0/+rbk2rbm6r5P+6cRm6/7q6//+8jHO8vIy8vKS86f/Ijk3I5KvI/8jI///TpHPTpKTT6bzT///dl0jdurrd///kq27kyMjk///pvIzp6bzp///+HwT/unH/yI7/06T/3Zf/5Kv/6bz//7r//8j//9P//93//+T//+n///9slevoAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAVQ0lEQVR4nO2dC3vctpWGg6ZrO9vN2FHitmuvx06jtlvJrV11L3Ll1q3VSq3VTK1NpLFkyTL//59YguAFnCEJ4OBCcPC9T6KhODwDzgFf4kKK/iQDIGE+GXsHABgTCACSBgKApIEAIGkgAEgaCACSxkKAO6AX5MsM+lFoi40A7vZi0+gWIPReTAcIsGFAADMgwIYBAcyAABsGBDADAmwYEMAMCLBhQAAzIMCGAQHMgAAbBgQwAwII3j+8ffv2D7/Nsg/P+U/x8rJYFLz70R/bEWd5wO1HYil/eXm74PP6vVutuJO1NdI2LgkgwIfnP/gtf83T8+F587XLDJQ5c5yvd5+J7R85/SYZBCg54TVRVGyPAO8fftqu0Je8Ls6KShIVWrz/7rNqxfuHt6Q4/vli42rNWf7ZYmu3BBFAfJNCgCJFJ00G3j8s1jjPV9Z8tlMgAOfdZ8WZiCe4W4D8FNWu0DPpZCZVaLH84XlRq/w0WcaJ2j3hi/Waz6s1bgkiwL/8B28CXq5kS3yZE+l7N9jmq+ClaHmcAgE4ZW1k//i2uwXIBWkfq+VW4lWuUL5C6MR/VnHvPuPVzw+Nas2Zh7osCCLArRP+9bsF4Nlwn68sq09TboEAWbtp7RsDtCu0bLBFPcoVyl9F9YlzvBxXnSL5y8mnf3g43THArfcPH8kCFAmUWwDn+ZJ+dwsEyKTqyYoObj2Y6xegPhnx6l6tUNHclx/axJ2UIz4hwO1ivDHRMcCtYgjTOQaoxjWO85W1a8kdECBbFUCnBSBU6Fl1vi8F4B/goSMUSIAPzz+XZ4H4Fy7ndeRvWWOdr6xuWhwDAbKmhyotO+8CVeezSoCiNs8CTeu5FyA7+/SPVQtQ9iDbPRTH+cqyVm24AwJwXtZZf6QnQO+gjtd0Nah7JMW9rOszkwbBExYge/l53QUS3Z4hAazz5asHBAEKyhaaV4+eALrTelXciXSkl/McRRdootOg/Pu9+9d/6xoDVDjOVy2IayBAgfJC2NqU/UnnhZ1mRXXCkg731ifxn7wj7ZpgAuQnaWkWKP82gwJY58vXxDEEEBSX2tduhZAu169ds+IRok7kS/viLHW2Mhg8EW+2JgjP6hsBnBJOAOmyYf5lfvjtsADW+fIzBoYAmwZuhjMDAmwYEMAMCGBAdUvibV/3MTggJgEmm68wTE+AKRCTAFMAAmwYEMAMCLBhQAAzpinAJ2M/Ty9ikC8z6EehLTYCuNuLJEC+YgQCBAP5ihEIEAzkK0YgQDCQrxiBAMFAvmIEAgQD+YoRCBAM5CtGIEAwkK8YgQDBQL5iJFkBWPASp50vImzsHVABAYIx7XwRYWPvgAoIEIxp54sIG3sHVECAYEw7X0TY2DugYgMEYAGjbIglX0FhY++ACggQjFjyFRQ29g6ogADBiCVfQWFj74AKCBCMWPIVFDb2DqiAAMGIJV9BYWPvgAoIEIxY8hUUNvYOqIAAwfCZr+89frYVbOwdUAEBggEBYgQCBAMCxAgECAYEiBEIEAwIECMQIBgQIEYgQDAgQIxAgGBAgBiBAMGAADESqwAGNcpIBdCibIAAMQIBggEBYgQCBAMCxAgEIGN60EGAGIEAZCCABmzsHVABAchAAA3Y2DugAgKQjx4IoAEbewdUQAAI4BM29g6ogAAQwCds7B1QAQEggE/Y2DugAgJAAJ+wsXdABQSAAD5hY++ACggAAXzCxt4BFRAAAviEjb0DKiAABPAJG3sHVEAACOCvMMbxtyMugAAQwFthjMVvAASAAL4KY2wCBkAAnwJcfDn74m31S2oCMDYFA4Zr5eL+UVGLs2fl7/liXaMQQMXV7oNssVX9BgFiZLBWrnbvHeU/nmXndw+KFedb0rsQQAU/f1w9PSh/gwAxMlQri9lP8xo8z0/5Ny9EE3D4QDPUGgjgeF/CFzZ9Af75tugCZVnRCuTc/P5AehsCqOBpWxTdxzs5ECBGNMYAOQvR8b96+uOZ6Az5rtDNEICPmf69bDwHU83633K2L+5IaBaoFGBxT3hw8dVBdvXrI61QS8zmmgkFSDFep0G1ukCs/y2X++KIhK4DCAEWd6WeT5g+relZxrwAKcSrAFUj6kEAKSpOAaZ/Jbiover8L4hLAHorK0X4GwPkuTrUmQZl/W8NIEVFKsDk7wXiAvB+T8l5rsLFTyK6DkAeZ7VC/LUA5zO9C2Gs/60BpCgIQEUtwOGM86yYAl3MZk1zMGEB2jHj3wrB+t8aQIqCAFSmfSsEUYCVIAhgBgSwDlXjU4DVKAhgBgQwCfV6dEGAEghAZdoC0GaBEheAVpgEBDAJ9Xx0WQyBIQARCGAS6vvoos6BQgAyEMAk1P/RxYw/vHX8f88x/ogMAoQpzDdJCpCtHv8kAyBAiMJ8k6YA2erxTzEAAoQozDeJClBFff891QDzGH8CmO0LrTAJCGASSutem9Qoo5RgKwAhyJsAhvtCK0wCApiEknrXRjXKCCXYCkCJ8iWA6b7QCpOAACahlM61WY0y8xIySwFIYZ4EMN4XWmESEMAk1FwA0xplxiU0URCAAARQhrJmMW4BiLNAEEAXi8Ku9+aCJ296tljuqz5DvUXqAtCuA0AAXSwLu/z56/43r/dUh7d6CwhAuhKcoABSFARQhrJmMXoBQozTC6Y9CyRFBRfg+nd/5v2g07w7tMPXHucL+9nl9nz+9f9t/2/eU9rhy/xYLze53uNr94stvhsuIUIBQs4CifII4YSe07SvA0hR4QXY44fx6ePX+UGfH97H34hf+Pn9cjtfOC304P+Xm1zv8bWPX0+zBQh5HUAUR4k37zlN+0qwFDWCADvlj2z55M317/J1l9v7pQA7WfVjv9lkr1oxTQHCXQkWpZE+ILF7geqoELbVVALkx/HyMV/kB37Okvd4hAD7Wf2j3qQIyH9MVoBg9wKZliWTpgBB+ls1sgDlpOg+7+g/+ft2lwDVJhDAKAoC6EeFGXHXrLUAWXO497UAVQAE0I2CANpRoS86SALUB/OSXxhbdnWB6k0ggFEUBNCOGlGAYoonO378Wpz85zvFsLglQL1JI8COqgQIAAEMoiwEIGVZFqCY5Odn//z18ev8MM+O+XUAWYBqk7rJOJ7kdQDDMKbcQhEFAbSjQgvgHwgAAQyiEhHgb32336lDBaxZhAAVmyCAxSxQ7AJc/0+1dLnde/9pd+g6rFmEABUbIQD9KRrRC7BXTqIe999/3RO6DmsWIUCFYwFaTwSL9kpwQ+wCZB9fzb/hp3/+0zB0DdYsQoAKtwK0nwkZUABqvqIXgF9cePzf8+pimlnoCqxZhAAVTgWw+CcOzAtbidpYAcpGgBTahlUL1PYSAgxj83hT48JWozZWgFPeAmgOADKtCqXfOZWsAJp/BAQBHNHUyvUeP/0X15hNQ9dg4sXizikIMAgEcIXPWSCbG0doAniPkoEARlgJwJRbUPF5HWATBSCGcSAAHabcgorPK8EQoIXTWbPEZoGYcgsqPu8FggAt3E4bp3UdgCm3oCLVyulcPFqCELoKK35CgBaOr5uMdSV4BAEI/wyWNk2tiEdKGBgQ4SxQSgKMdC+QYWG2UUXRzKMBda18fMWnP09Vfz/QFboOK19DXweAAGZhtKjAAqwMd0r4Q6/mc40bF4qnqPQjTYPyk/9S/zqYhyvBrFmEAN72hSm3cFiYbdT6hFfD5S86D9aV1VYCXNw/yrKbF7N7R+J3adHHvUDEMFIUU26hCoMARkxRgKtdfrgfbvH/CqRFCKAu7HzGeVD+BgFoBQ8LcPnL/+JPgpuLR0XM59/ky8VjE6s1P6MLsJj9NG8Brp4eiJYgkxczCKBX2Pk9b/miRW2aANv7+fh1n49e+SG8fPyar5bWnA4PE4YE+OdbfrRflBJkmbR4JwcCaHC1+6xahAC0ghUC/Px1sVD2dPIX/lu5Zvn1dwZdoHlNbUGvAK3Qjp1uFhMXYPHF22oRAtDoO/5LAfjhvi3mg/iraAHKNacGAnQCAQbCdAq7eSEaAC8tJimKOqVOKkzG/XWARgAx7OWNgdQC5Ji0AJ1AgIEwncKaEVMkLQD5ohKlsBburwTXAvAef/7K++/8yXG5B+UaszFATxcIg+CeMJ3CFlvNcgwC9HYnfBTWxv29QLUAmZj5+fhqPv/Vq/38Zb9cw7tC/2ncAhw3fxJTHO2YBu0O0yns8EGzHIEA/QNKD4WtEP/NcIJcneZ+oEKAq93i6hevy3KxJ7SBNYtJC1ANAQoiFMD/iNs2ar10x6zWylL/j4KnLQARZrR1M2DKIAApar10x6zUyqnyabq9oTKsWUxagBYQIEZWH4ui+xfxq6FtWLMIASogQIzItdLq/puFrsCaRQhQEYEAq7NAEECuFYPu/2roKqxZhAAVMQiwch0AAsh/EWbyVLh26BqsWfR/kV4CAqiD5DnQyQjgr2KHL4Tpha7DmkUIUBGHAIELs41yETxIVP9CDFNuoWCaAgQ+KdPCiIXZRrkIHgQCmMHooX35Ct0tp4URC7ONchE8SEQCUO9TlJiiAMEnZmhhTLmFAou6IT9jXIN4BCDfp9gU5TFPFYwe2p2v8FPztDCm3EIBvWbI/yqTDtEIQL9PsS7JZ54EVjvoUwATpLBJCNDzdClxy3/PH8b3sf7HAbEIYHGfYlWQ4VO4CNjtYNIC0Gum7/mCqwJ0q6B8RMTGCGD8HEZzLPcwZQEsagYC6OFfANtdTFgAm6pRCnD5iz/N5zvFX8AQnpECAbTxI8Bqu7KJAljVjVqA7R3+NBPxt/DGz0iBANp4EmBlZBGnADZjH+8CiAejFAKYPyPFhwCtbIWaBZquAHK+jPafthsWU3S08mzrpidW6gK9kV5Mn5HiQQDiGc0yyf5ngbwJ0ByTZt+Aths2zTOtQNuTU3eoOJbz0/qqAGbPSHEvALlPa3f8B7gOYHkcuP/3FGiwZlGvoNG7p92Rx/yA3tupjvqq+2P4jBTnAthkyzyihfdjx+486P5f1KHBmsVAAtib3Rl4nPdsdupHY/GnoRCekbJJAvi/F8iqkUpYAOvWeUI3w220AH7/HmCDBbBtnSGAHhBAA9YshhMgnduhIUAfkxXAdvRvVFZovM8CmUAIaTNtAWKdBXJxq3pCAlhkixLTYuIChLije4wrwWZlhcX7lWATSEEyUxcg9N/0TOLvAfzi/V4gE4hhDZMXIPBXgAAQwBBGD4UAMQIBzGD0UAgQIxDADEYPhQAxAgHMYPRQCBAjEMAMRg+NTgCjKSem3EIBBFBDDGuAABpUhQX544MGCKCGGNYAATQoCzO87Mxsi4UAaohhDRBAA1GY6Y1HzLZYCKCGGNYAATQQhUEAAQQwg9FDIUCMQAAzGD0UAsQIBDCD0UMhQIxAADMYPTQuATALJIAAZjB6aGQC4DpAgVqA8xnnQbF88eVs9sVbdSij7QwxrAECaFAXhivBmW4LcH7vSLxu6YUy2s4QwxoggAa0wphyCwVTFuBq95lYOHygF8poO0MMa4AAGtAKY8otFExZgEXZ7bn5/UG55k4OBDADAsSIjgA3L8oG4Orpj2d3Kwc8tADWxCYAYcwEAcKiI8DF/aNy4auD7OrXR+pQZrlXVGITgDBmggBh0RFgIVfj1dOqCYhQgNggjJkgQFh0BGhVIwTQpxkzcSBAjGgIUA8BitnQi5/4uw6waTRjJv1JAwgQFg0BynM+bwcWs9m9aggAAZRQxkxBjxMIENetEJuIaZcRAoQFAngGApiXFRII4BHKmAkChAUC+IQwZoIAYYEAwYAAMQIBggEBYgQCBAMCxAgECAYEiBEIEAwIECMQIBiJCxApECAYECBGIEAwIswXBIAA4YgwXxAAAoQjwnxBAAgQjgjzBQEgQDgizBcEgADhiDBfEAAChCPCfEEACBCOCPMFASBAOCLMFwSAAOGIMF8QAAKEI8J8QQAIEI4I8wUBIEA4pp0vNvYOeAICBGPa+WJj74AnIEAwpp0vNvYOeAICBGPa+WJj74AnIEAwkK8YgQDBQL5iBAIEA/mKEQgQDOQrRiBAMJCvGPEjAOgA+YoRCBAM5CtGIEAwkK8YgQDBQL5iBAIEA/mKEQgQDOQrRiBAMJCvGIEAwUC+YgQCBAP5ihEbAVTcUW4xdpSvwkj5iusrBC2MfhBa47PsO9FHhS3Mz6cmmy83QIBwhfn51GTz5QYIEK4wP5+abL7cgJEZSBoIAJIGAoCkgQAgaSAASBpnAty8mM1mW1l2cf+o8/3zuwemURdf5m/OOuIUZfG4LVphD0hRs3vd33kAWrrSzZc3nAlwuMW/e5HHrq9+8+I3XTkejBJrztezNVwWDyjfNS3sane9RjV2kQAtXenmyxuuBLh6yk884vt1fcuL+3/pWDscJdaIbfSjrnaf8fVfrZ0IdQpbP3p0osyhpSvdfPnDlQD1KeTi/l93eTN8tdtq6BZb2eH62WI4qszx2rlpOKrjDKhf2PrRoxNlDi1d6ebLH866QPn3LDqS+Vfk3/LmxbNs8cXb+t08UefNr3pRZYdx/UAYjOoqRqewomr4OsIudnZXBqGlK918ecPlLFD+1beq783/l04PPMuirTWIKtvR9cZ5MKq/QjUK66hQjSgatHSlmy8/uJ0GzZNffe8v5QmJYmagV/y+qLKV7TkQ+qJ6m3StwtaadJ0oKrR0pZsvH7gSQKQxPyFI4jf0nZt0otZzPBwlDoCOw0CnsPWjQSfKHFq60s2XP9wNgnlLmnf2+PfOa463jc23PRQns7Vx3XCUWFqsNdHDUdni7kG5iXFhHdN6OrtoDi1d6ebLH+66QId5U5d/9/wr5i34M94PrBu+6qTUNUXdH1WOmLpa6KGovJi+7oO6sK7egzqqO24YWrrSzZcvcCsESBoIAJLGrwBli9d9X0sUUWEL8/OpyebLBWgBQNJAAJA0EAAkDQQASQMB3HO9N5/Pn7zhC/tZttwfe3/AABDAOcs5P+SPH78ufiskANECAZxz/A3/+fFV8QIBIgcCuKY68rPi4L/cns+//i47zTtFO2PuFegBAjhnOa8M4Gf/ogU4zftDl9swIEIggHv4WZ+PgWsBrvf4sb8s1oG4gABe+Phqnp/0KwGWxYD4chujgfiAAL44fvKmFmAugADxAQFcU53o89N+uwUAMQIBXFPNAi2bFgBTofECAZyz5POe+c+dZgDMZ4HqS2MgJiCAe4pbIfjRXpz5j6vrAJgEihEIAJIGAoCkgQAgaSAASBoIAJIGAoCkgQAgaSAASBoIAJIGAoCk+X88nj4qZoPBpwAAAABJRU5ErkJggg==" style="display: block; margin: auto;" /></p>
</div>
<div id="function-output-19" class="section level3">
<h3>Function output</h3>
<p>A list of objects of class <em>ggplot</em>.</p>
<p><strong>Note:</strong> Please note that plots will not appear in the
<em>plots</em> panel of <em>Rstudio</em> if not assigned to a variable
and printing it (see sample code above).</p>
</div>
</div>
<div id="heatmap-for-visualizing-pathway-enrichment-olink_pathway_heatmap" class="section level2">
<h2>Heatmap for visualizing pathway enrichment
(olink_pathway_heatmap)</h2>
<p>The olink_pathway_heatmap function generates a heatmap of proteins
related to pathways using the enrichment results from the
olink_pathway_enrichment function. Either the top terms can be
visualized or terms containing a certain keyword. For each term, the
proteins in the test_result data frame that are related to that term
will be visualized by their estimate. This visualization can be used to
determining how many proteins of interest are involved in a particular
pathway and in which direction their estimates are.</p>
<div id="function-arguments-19" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>enrich_results: data frame of enrichment results from
olink_pathway_enrichment()</li>
<li>test_results: filtered results from statistical test with Assay,
OlinkID, and estimate columns</li>
<li>method: method used in olink_pathway_enrichment (“GSEA” (default) or
“ORA”)</li>
<li>keyword: (optional) keyword to filter enrichment results on, if not
specified, displays top terms</li>
<li>number_of_terms: number of terms to display, default is 20</li>
</ul>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="co"># GSEA Heatmap from t-test results</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="fu">try</span>({ <span class="co"># This expression might fail if dependencies are not installed</span></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a><span class="fu">olink_pathway_heatmap</span>(<span class="at">enrich_results =</span> gsea_results, <span class="at">test_results =</span> ttest_results)</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>})</span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a><span class="co">#&gt; Error in eval(expr, envir, enclos) : object &#39;gsea_results&#39; not found</span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="co"># ORA Heatmap from t-test results with cell keyword</span></span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a><span class="fu">try</span>({ <span class="co"># This expression might fail if dependencies are not installed</span></span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="fu">olink_pathway_heatmap</span>(<span class="at">enrich_results =</span> ora_results, <span class="at">test_results =</span> ttest_results,</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">&quot;ORA&quot;</span>, <span class="at">keyword =</span> <span class="st">&quot;cell&quot;</span>)</span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>})</span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a><span class="co">#&gt; Error in eval(expr, envir, enclos) : object &#39;ora_results&#39; not found</span></span></code></pre></div>
</div>
<div id="function-output-20" class="section level3">
<h3>Function output</h3>
<p>A heatmap as a ggplot object</p>
</div>
</div>
<div id="bargraph-for-visualizing-pathway-enrichment-olink_pathway_visualization" class="section level2">
<h2>Bargraph for visualizing pathway enrichment
(olink_pathway_visualization)</h2>
<p>The olink_pathway_visualization function generates a bar graph of the
top terms or terms related to a certain keyword for results from the
olink_pathway_enrichment function. The bar represents either the
normalized enrichment score (NES) for GSEA results or counts (number of
proteins) for ORA results colored by adjusted p-value. The ORA
visualization also contains the number of proteins out of the total
proteins in that pathway as a ratio after the bar.</p>
<div id="function-arguments-20" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>enrich_results: data frame of enrichment results from
olink_pathway_enrichment()</li>
<li>method: method used in olink_pathway_enrichment (“GSEA” (default) or
“ORA”)</li>
<li>keyword: (optional) keyword to filter enrichment results on, if not
specified, displays top terms</li>
<li>number_of_terms: number of terms to display, default is 20</li>
</ul>
</div>
<div id="function-output-21" class="section level3">
<h3>Function output</h3>
<p>A bar graph as a <strong>ggplot</strong> object.</p>
</div>
</div>
<div id="scatterplot-for-qc-olink_qc_plot" class="section level2">
<h2>Scatterplot for QC (olink_qc_plot)</h2>
<p>The olink_qc_plot function generates a facet plot per Panel using
ggplot and ggplot2::geom_point and stats::IQR plotting IQR vs. median
for all samples. This is a good first check to find out if any samples
have a tendency to be classified as outliers. Horizontal dashed lines
indicate +/-3 standard deviations from the mean IQR. Vertical dashed
lines indicate +/-3 standard deviations from the mean sample median.</p>
<div id="function-arguments-21" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format should minimally contain SampleID,
Index, NPX and Panel.</li>
<li>color_g: Character value indicating the column name that should be
used as fill color. Default QC_Warning.</li>
<li>plot_index: Logical. Default FALSE. If FALSE, a point will be
plotted for a sample. If TRUE, a sample’s unique index number is
displayed.</li>
<li>label_outliers: Logical. If TRUE, an outlier sample will be labeled
by its SampleID.</li>
<li>IQR_outlierDef: The number of standard deviations from the mean IQR
that defines an outlier (default 3)</li>
<li>median_outlierDef: The number of standard deviations from the mean
sample median that defines an outlier. (default 3)</li>
<li>outlierLines: Logical. Draw dashed lines at +/-IQR_outlierDef and
+/-median_outlierDef standard deviations from the mean IQR and sample
median respectively (default TRUE)</li>
<li>facetNrow: Integer. The number of rows that the panels are arranged
on.</li>
<li>facetNcol: Integer. The number of columns that the panels are
arranged on.</li>
</ul>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(SampleID, <span class="st">&#39;CONTROL_SAMPLE&#39;</span>),</span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>         Panel <span class="sc">==</span> <span class="st">&#39;Olink Inflammation&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>  <span class="fu">olink_qc_plot</span>(<span class="at">color_g =</span> <span class="st">&quot;QC_Warning&quot;</span>)   </span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABklBMVEUAAAAAADoAOpAAZrYAx+EZGUgZGXEZSJcZcboaGhozMzM6AAA6OpA6ZpA6kNtIGRlIcXFIl91NTU1NTW5NTY5Nbm5NbqtNjshmAABmAGZmZjpmtv9uTU1uTW5uTY5ubk1ubqtuq+RxGRlxurpxuv9zc3Nzc4xzc6RzjKRzjLxzpKRzpLxzpNOMc3OMc4yMc6SMjIyMjKSMjLyMvNOMvOmOTU2OTW6OTY6ObquOyP+QOgCQOjqQOmaQ2/+XSBmX3f+kc3Okc4ykc6SkjHOkpHOkpKSkvLyk09Ok0/+rbk2rbm6rbo6r5P+2ZgC2//+6cRm6//+8jHO8jIy8pHO8vIy8vKS807y86f++vr7Ijk3Iq27I5KvI///TpHPTpIzT06TT6bzT/9PT/+nT///bkDrb///dl0jdunHd3Zfd///kq27k///pvIzpvLzp06Tp6bzp/9Pp/+np///+HwT/tmb/unH/yI7/06T/09P/25D/3Zf/5Kv/6bz//7b//7r//8j//9P//9v//93//+T//+n///+M7uoKAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAWQUlEQVR4nO2diX/cxBXHU1gOQwtlaZO2nKFs3ICdUGqgKQUMFHBoOEwLcSjbQg20JNuDBLwFG2KI9X93Dkk7mluakebY9/uApZVmRrPvfTPzZjTSHitAIAcdC10BUNoCgEBOAoBATgKAQE4CgEBOAoBATrIG6A5QnOqTDgvZA9RnLUCdFdovAFDiCu0XAChxhfYLAJS4QvsFAEpcof0CACWu0H4BgBJXaL8AQIkrtF9yAejLlUfZj2+trPz47kdViaX69B6U7bbPvNZqAIX2S54AfY3g+bodQN8+co/nKg2j0H7JE6Avb3kBABpGOQH0/fP3oK4L9ULo78pt/8QAkb3PEEy/fGRl5dEvV1YIWcyHKsnXd6+s3Po26cI+RQcQTHVpcSu0X/ICCP359pHbFy0QBuL7529H+7e+jUi57bPvn8esMB/qJKQFwh/eInnvYUqLWqH9khdAuL3AEJQA0V7s01vfxkDQXk78UCapACJni09veWFRWtQK7Ze8AMLNBQNQgTsi3DWRfXywZqb6UCepACIncOZFaVErtF/yBugtjAbPTBOgKgkA1E1ZA/TlCumOdADVSZoAoeIAICvlDRCG4fvntQBVSSQxEABkoawB+vru2/GoXAdQnYQbhd1eAEBWyhogdHRl5bZ/3PKCJgaqkhCKmvNAAJCFcgFoaRXaLwBQ4grtFwAocYX2CwCUuEL7BQBKXKH9AgAlrtB+sQboWOhHeEFy9UmHDRfeE4KWSi4A7fisCMhCEVocAEpJEVocAEpJEVocAEpJEVo82yB6NApdg+VQpgCNiELXYhkEAIGclCdAoxEQNJDyDKJzBShCi+cJECboWH78xGhxACglRWjxTAFCCCUVtFkqQovnGURjJVfhNAUAgZwEAIGcBACBnJRtEJ0lQBFaHABKSRFaPCuAjqkVump+FJ3F8wJIg0kmBMVm8SKrIFoLSSYExaecAHI4C+oqAAjkJAAI5CR7gHZ2SAi3E+3WAFDw+nnY7ojHQyujUVhVwzfuotuP0fbjY8dubp5NWpFZHCs/gL760e/fJNs77yq++uEn3/3uPvZs2orM4lj5AfTGfR9jZL579hcUoGefY8/61PAr1iKzOFZ2QfQ3P38TYYM7MNyFfXXnD55jz3pUnquuWys7gDA2b9yH2x6098VN/XVhABCRzq7794/vvWCTMA6RGn7zM3zn66ZPPsYb0gh9cfPirEfl+uBHW2nsev2pc8Xln3xgThiJSA0JLjTuoS1QNSgDgPqRxq77D35QXH/6gjphZCEdruF3dARGMCKdWX/D+BD8RGZxLI1db5w/V+wdv4L28JvU0gCo69kOAoCIdHa9cX5M+FEljOzrDH4rA4bxhSEGOqmPgSL7OlpE8ljOEZnFsVxioNikgySB6qepnADKf0lrhNIG0YYuDATKaSIRFEK+b6YmM7OWTEVZpRVEGxOKXyeZydlkKtoUABSLkqloU7kDlMz9oWQqyikzgAQl45dkKhq//K4HSsYtyVQ0egFAICf5XpGYjFeSqWjkymhR/RIoQosDQCkpQosDQCkpQosDQCkpQotn9FgPKIQAIJCTACCQkwAgkJMgiE5JEVocAPKjYSa2I7Q4AORDQ91ai9DiAJAPAUAeE8alQboWZnnR0t2jzQkg0XkDtQw1QEu4SiQfgGTOG8qh1XUAIA8JQ0nivMFWrjYaoOUiKJsgWua8AR1aBkA9Xy8qi1NlDZBjl9I+Z+N6PYAUlcWpsgHIewzUJS+Tp5e2KC6LE+UNkEsz0I0Apv0BgLolDCc+AHIszAmBpYmnNVzsjbFOmhPGKGf3AUB2MnCxl+r7gdzdx5bQoaTg/Mwnk8mpi3T/aBt9OH2VOTsln2aTDXJ2TV7E4eaW8TJ6LvCroq0SxiamAWC82MqhCwI6sRAYoKPt1d2iOFjHgCCW8GZa4YQ1w6eL6SuYo8PNje4X0nNxOdK3tBo9UwNkNy6S38ZatD/dwmn/BNlbnAKCCNoq/2CmmDaIHDt88b0zCKqDMxcVpVhICxB+UzSW1/dEuxjVvkWo0tgANFpIVtGO4UxQgOpGZYp6p1kJzv92uQTz059vb5WwTVEnR5h6d7L60eZrm/gT7sIOy32S4tS7p3jWtADh12xqEnYByMWsEihMaRtdmSIfw89IKECX0aICngmytnjdqMxXd+UhDiFrjWyq/2eIjsNNRNvhJtpDnyhAdL+YIszmk3YAXT6hTRgIIEvX8NTw2RrtkkhQ2gDNK0fPMQayEGd++uoRan7KzeGLF2m3RhKTP+gTBYju035w2g6gSycX+36CaIlZrW3MudouG5O0kUvgSsBLSV6r6rbM5EtmgFAbhalBm6q1mle9Vv2HAkT35ySomrcCqAqBjAntpfmHbp+3jWv0AI0MDZDDKCwsQOYuDLU7qPWpNmhAP1n9cF0D0KwDQIu3jBsSthBv1i4wtHONdBg/Ukhasa4oBBzFL4JoREcVRM8n7LTOdGOGwZqu4U09WvPaAnVLqJfmH3qLvM6uaQVQigtVcYOBxu3/Wd+QD+MRC4/jEVgx/9XLW6ShooApAaKFzLwC5GEY3wogn47kwZFUI0JqWlh8Sgii089kIvFou+n8gydIN3ewfqqOn1EyJUCdRmGGhD4mEtsB5EeLbrBkpwgcsVirjcXxrQwk0vMgNrhbGXWDVG5m+L7HlI7c5QDheaDVv63ucpeJBKAB//E3I6mO9zkCqYPF/+owyyxqHh1Asjao39agLj2NRqehgOuB6L0PcUYg/HogQ/za5+WMQXJifBl1sE77NTGUMYt0ieLd+QgBahlYt7qUmtdRnYI74b8aWSk8QIqpIersHq6jwFXSqckBAqQaknLxLz5SUiX0IlUM5LsBMAAkzFBKW0KvlcqBRYaLWTnmo2uRNAkreQvp5E4y+spgf2mpfMO24Ee4S8uixdfNhzoUFfWiejxLjSe+UbAku/k26IIyebDCp9HaX+F6PofYAEnud5gKMX2Vjl9APBYzQOSWG75ppojQB1+R6Bsg5ZTT4qgKoHYBvnxAKeTRFyU/FzNAZIh/tP3ERLHAevglrTb8KM9LTusBYvZ1pRj5kTEozZYfQHTZomp99fCL6n0DZOrFrHhpB1BdoPq6nb5dROIB4m6YSBIOKFtPyYb8elTkaUdCAimF+iqzWUacjDU01TQ+8QApOrAIH+upLSz3j9QBOn64W6yLcx1qxTVBKh6M7WurS4dRugDV9pcBpJiGbAuQ1o3yEwJ/TMEtvtlSANTXPFDbzE2362eQpIeZ9E2CtAXJT4koj9o3KCqA4g6iJ7WGnUhsbVxZdrENaTVXwzc6jVGZRUujrGCn+ihL72Dxvpux8Ms52gHEJRaY4dsjSwvK42aN622p6MaPN4Dk5dCb8h3uyEsUHqB2Bm6kHeklpG9ZG13NLGst8mxfEclBXwCdlS+Q7yKWC/IOBwTm0bZkMmjA9UDWiat9NT8tAZIVbxUDmYbj7QHyIsWlKUBH2+Z3b5jVuJm6QTe2MZAntTGvBI7GUJ6LOdo6j0ts5EdDqWaIOJyMAOFVYmtksRhujuYd+rXmzVSimfRuRhwAGePdmqNGii546mtmAEhEOMyQXH7psgtb3cVDb/zs6osX8TPO5abdFZo3U7HQcExWSJ/zQK0HuAUbPjfOjAqhBekOkHmqT5Gtebb/oZBSCoDYIBphU75IyuZ9UoK4eaACvwzk30OvB2ol7h+2EFE3kroCZEqryNZXq+NpGE9bILK3TkA6pG9wKTetJAIke3ajGBogk+0r/0gGZGIyvSslLZZd5TToRgSQTDVA+BF68mKO2ulS3+skdmGK6eghAeI6AFWCeua54FHhoySrnkj4ZJODr02LgjrIN0AYl+kp8n4OtFtu2pUlCaLlbzwLApDUWQXrsnI7EsUWpXalEiBlHkmBKqZsv6+1fAOEJ25+Qwdj7qOwxTBe2g0OeDNVwIM/IyKmBMjgSqHxME4fSZob3TA+e4kTiYsh2I3zi9dEBwFI4l4OICGpSJDWlWxakVRTDvZop2+agXRcXDp+Zf+BCxYJfUvAo3G8anrYIzqArC4lnWDSEOT4DfORhovG+6WCACSfGR5Jm4pu/JgAkhQDADWk4WLxjlavr/m1kiIUEdlgDnDtULtLsblMJIbDJ631QHvH3x+PdW+qH+LrSAfMqhR1k9HexSwvnZqyQRQzQKL2UAS9d68mBgrxdTQOdXM3y0vdf8VGUGIAHb/CvKg1EoAEtZsF1JXDRj4KgELDFIXFmzLEQFqAIlAjBLIjSJGi2WdJu7DY2qMopB3Gn4z+554kka+QQpFBXpQkDLLIu8TSv2hcH0SHF9fpiP4VjtkBxOxILgZiFMELphzUGiAtBGIDpDgNWij8onon8fxIT0umIvXFKRqq8Px0sPi1a/6rwSodgKSu47sc6WmxCVJbVXczNUGArhHxR8l6nfJnDCVLdxbrzSyUCkCC8xa3HZoRr+E+uQkgoXzzwc7J2ssTQOTHM6avrOH75w6/dkmUIED8Oo+RIGm2UtcwPguj9uDrHhuqtha/dk1KEG5iDl/66KXdajmigxIJohdoiKDoABIJadi0F1+H7+lqKQDCy5cPnvz8HfoQBn205+DXL69+ePbP+P2GiK8DuofXTD+uf3wsaYBGfAcmIUhwJWvSPnwtoziY5F1YMd0oZvS/taJ8tAe/iR7/yDNe2ooAonv4nGKBYaVEABKWCOklyVYfERogz75OAaD5WjElP3c5pXCgngwvZCb/nyUA0b05+QHV5QOIz7Y4SnY4fFr52iJ1RPwUimH8wZP/fakKg8pHe3BcVP5P/8N7M/qbhrryUwmiuQG2FJqR5i3iLEA26w7ll7fK0GML5MviR++89+TV8m/5aI8UoJ5boMEAko2rWDKUN8x5xlT7FjUoL2SVoTeEvFl8+gQevs9e2Vg82iMDqOcYaCCAeG/wywbroxKXKQHimyarOljniB4g+gwX+RHV8tEeKUC4e3tMP9JPEKDG4ZHpXnzT7eynFgC1RK4Fme0U4uaRYaoo/iBa5Q2u/VH5rOl4Sftj42jr9E2cbb9itLJ4VjVdgNi7GRrn8ifZD/aObvLHXJuth+FebIqaGZ9VjR8gkzeMrUMDlWZCa0ezaHD5mP38ADIrN4AMiboCJB3d8fvNkq1KTV8JBNF6bxgbIO7uK5euk6NFBhu9Y4cSLRXBCixeSQCkkwU+jfM+fMwW2LFB66YoLN5U8gBJbs/LzjZC3ebW7ZrSGKgvxWHxhvICyDBbJJ5wv2azoJ5Dnzgs3lACQbRRI/OaVs8A8b1Y52LSVw4AYXVrgFwYcs2fiTIHyKJpcuzHOmbORrkApOpIlF4GgPwo/SDaJIWPPTRBwwc/EVo8f4BK9REGDa4ILb4kABlibLo/eK1aK0KLZw6Q5I5n83RT/ivg98niCC2uA2j//vH4+BWLhNGIQ6CmQg1IvwApHorISTou9k5YJgwuxa0oC4D6JWjJAbp0kvkQMUCV73kGGCrUePQJkOLB0Kyk4eLG66HeE91Spe8FCKwA8jKeV2jJAbr+9ENj+pLW4d8TLZPKv+pImP1sXGuWQgyUVhCNf+bg+jOadyR6+zotViWrTiyaIJs8ssv3MwZbZoCwFj93EPp3440ASZMEn9xZ6mF8ERFAyk6m0f0Mx0sYMtMCCL/id//h3ueB7OIPI0A91U5Xm0EvGat0XFwej+vXRIcGyNSF9VGzDpVZPkWwnMNLDNRDvUx1AYKwkgFIOxD3Wh+zAKCForiZ6rQ22Vcl2l01DD9pBdHGhOG/TihHAkC1wgPUxhF82nA9CQzjS4UGqA0CQlrN2N5D1SJUZgD5UB8AQYQ7oAID1GY8I6RVZQaABlSGAMEge0hl0IXJkwFAwyiDINqpzLSUWRAdfhgvzwwADajwAPWhPPGJ0uJ5ApSrIrR46CAalLgAIJCTACCQkwAgkJMgiE5JEVocAEpJEVocAEpJEVocAEpJEVocgmiQkwAgkJMAIJCTACCQkyCITkkRWhwASkkRWhwASkkRWhwASkkRWlwP0P6D/b/eBZS09FxcGuD9QKCkpeVi76fQAoH00nFx449/AYBAemlfcXeyjIHieE80KEaL6140/tsr+iA6wq+TuSK0uAagS+cMo7AIv07mitDiaoCuPzXGOlcl3Nkh1d+Bbbjtjng8tGAeCOQkAAjkJFjOAXISAARyEtxMTUkRWhwASkkRWhwASkkRWhwASkkRWtweILnuUBzvICiqS1F9wmHFhWP+O7zUAooKV5SjAKBlL8pRANCyF+Wo4H0oKG0BQCAnAUAgJwFAICcBQCAndQVobzy+9wLd3b9/PD5+xaUS9bKjG+eZX6p3K8q1Vkx+11oxRbnWClVlfLLedbSVF3UEaP+BC8VeWf+9E66VqJ9fvHQC/+elKNdaMflda8UU5VqrS8evYNP7qJUnOXRh158uv8lJxzrUzy/iEplFkC5FOddqkd+5VkxVHGtVmdxHrTzJAaDLtC2+8foFU0q9Fs8vYoMwNnIpyrVWTH7XWjFFudaKIcbZVp7UGaD9+8sY6PrTD9XhUCfVzy+6G2VRlGutmPyutWKKcq3V3vH3q+dkkgeoqGIg3Cdff6Z7W8o8v+jsKqYox1ox+V1rxRblWKs9FEHvUQIzAIjrkLuKeX7R1SjNRyGd7Vvm9+EqT7baQ1HDjfPnfNXKh4IDxD6/6BgYco9C+vK6j3DVE0C4GiVAiQfRuPvaf/gKt9tVtSHch6ZVUa61YvM71sqnrdAorpo9SXwYf3lMprHwsLTcdRDxOi4KtSG+inKtFfMFXWvl0VZ4IvGcL1t5EdzKADkJAAI5CQACOQkAAjkJAAI5CQACOQkAAjkpdYAONyeTyequRbqtxuej7ckG2ZlPTl/l0s5PXeSTg1RKHKD5BDt6euqiKaEEIArOVAqQtwpmr8QBmq7hv0fba6aEIkCPvYgxOXzpDwCQi9IGqEHOFHVmiJKD9ddQt7ZxsI4/HZx5D203SoBmk0nZceGcU7w7P/0nDFB9Bu2cerfqwsoSDzdxidCnSZU2QKgLqwnCjdEMtR0H6+jPDMdFs9VdhNEW6ecwEfQsJQgBNMfkTDemaFOfmaJs80kJUFXi4SYuEZolqRIHqMANDYmhD3GPdLC+RUEo/2xRXmanryIiDjdJk0MjbgTQwdldnGuKT5ZncH4SUmGA6hLJaXoKxCt1gAoSD5etw5z0WutbRf2Hup0Oq2hoU4KAADraRo3T6asIoPoMpYsZhZESyQcYl8mVAUAF7XlQ9LL64ToP0BmMRgnQhKoCqJitoR6swABVZ2YNgKoSASCN0gao6leQy2tu1ACxUQwG6ODs39+5WNQtUFH1byVAdUEAkEZpA1SNwpDjie/nYheGo5tpGQNtNXMebb+KwiB6kh6lQM4oQHWJAJBGaQNUziPP0RCcNhXlDgMQalzYUVg16UjQm+IxXDUKI2dmJPmiBSIlAkAaJQ4QvZVBmMAzOBenTF9GAXp1vZzLKeeBytseBCAyjz2t5oHIGXYeqCoRANIodYAMgsF33wKAQE4CgEBOyhwgUN8CgEBOAoBATgKAQE4CgEBOAoBATgKAQE76P8sur/WYbIrVAAAAAElFTkSuQmCC" style="display: block; margin: auto;" /></p>
</div>
<div id="function-output-22" class="section level3">
<h3>Function output</h3>
<p>An object of class <em>ggplot</em>. A list of outliers can be
extracted from the <strong>ggplot</strong> object.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>qc <span class="ot">&lt;-</span> <span class="fu">olink_qc_plot</span>(npx_data1, <span class="at">color_g =</span> <span class="st">&quot;QC_Warning&quot;</span>, <span class="at">IQR_outlierDef =</span> <span class="dv">3</span>, <span class="at">median_outlierDef =</span> <span class="dv">3</span>)</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>qc<span class="sc">$</span>data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Outlier <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(SampleID, Panel, IQR, sample_median, Outlier)</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 5</span></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a><span class="co">#&gt;   SampleID Panel          IQR sample_median Outlier</span></span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;    &lt;chr&gt;        &lt;dbl&gt;         &lt;dbl&gt;   &lt;dbl&gt;</span></span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a><span class="co">#&gt; 1 A48      Inflammation  8.64          4.53       1</span></span></code></pre></div>
</div>
</div>
<div id="heatmap-olink_heatmap_plot" class="section level2">
<h2>Heatmap (olink_heatmap_plot)</h2>
<p>The olink_heatmap_plot function generates a heatmap for all samples
and proteins using pheatmap::pheatmap. By default the heatmap center and
scaled NPX across all proteins and cluster samples and proteins using a
dendrogram. Unique sample names are required.</p>
<p>Group variable can be annotated and colored in the left side of the
heatmap.</p>
<div id="function-arguments-22" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>df: NPX data frame in long format which should minimally contain
SampleID, NPX, OlinkID, Assay. Optionally columns of choice for
annotations.</li>
<li>variable_row_list: Columns in df to be annotated for rows in the
heatmap.</li>
<li>variable_col_list: Columns in df to be annotated for columns in the
heatmap.</li>
<li>center_scale: Logical. Default: True. If data should be centered and
scaled across assays.</li>
<li>cluster_rows: Logical. Default: True. Determining if rows should be
clustered.</li>
<li>cluster_cols: Logical. Default: True. Determining if columns should
be clustered.</li>
<li>show_rownames: Logical. Default: True. Determining if row names are
shown.</li>
<li>show_colnames: Logical. Default: True. Determining if column names
are shown.</li>
<li>annotation_legend: Logical. Default: True. Determining if legend for
annotations should be shown.</li>
<li>fontsize. Default: 10. Fontsize for all text.</li>
<li>na_col. Default: Black. Color of the cells with NA.</li>
</ul>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>first10 <span class="ot">&lt;-</span> npx_data1 <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>  <span class="fu">pull</span>(OlinkID) <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>first15samples <span class="ot">&lt;-</span> npx_data1<span class="sc">$</span>SampleID <span class="sc">%&gt;%</span> </span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">15</span>)</span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a>npx_data_small <span class="ot">&lt;-</span> npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(SampleID, <span class="st">&#39;CONT&#39;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a>  <span class="fu">filter</span>(OlinkID <span class="sc">%in%</span> first10) <span class="sc">%&gt;%</span> </span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a>  <span class="fu">filter</span>(SampleID <span class="sc">%in%</span> first15samples)</span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a></span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a><span class="fu">try</span>({ <span class="co">#This statement might fail if dependencies are not installed</span></span>
<span id="cb31-16"><a href="#cb31-16" tabindex="-1"></a>  <span class="fu">olink_heatmap_plot</span>(npx_data_small, <span class="at">variable_row_list =</span>  <span class="st">&#39;Treatment&#39;</span>)</span>
<span id="cb31-17"><a href="#cb31-17" tabindex="-1"></a>  })</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAGACAMAAAByRC0tAAACplBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrYAx+E6AAA6ADo6AGY6OgA6Ojo6OpA6ZpA6ZrY6kNtFdbRJebZOfbhSgrtXhr1ci79gj8JllMRmAABmADpmAGZmOgBmOjpmOmZmOpBmZjpmZmZmZpBmkJBmkNtmtv9pmMZunclzc3Nzc4xzc6RzjIxzjKRzjLxzoctzpKRzpLxzpNN3ps58qtCAr9KFs9WKuNeMc3OMc4yMc6SMjHOMjKSMjLyMpKSMpLyMvLyMvNOMvOmOvNmQOgCQOjqQOmaQZjqQZmaQZpCQkLaQtraQtv+Q27aQ29uQ2/+TwNuYw92ZmZmcxt+hyuGkc3Okc4ykc6SkjHOkjLykpHOkpKSkvNOk09Ok0+mk0/+mzeKr0OSw0+a01ui2ZgC2kDq2kJC2tra2tv+225C229u22/+2/7a2//+52em8jHO8jIy8jKS8pIy8pKS8vIy86dO86f++3OvD4O3I4+/M5vDR6fLTpHPTpIzTvIzT07zT6bzT6f/T///W7PTXMCfZNSrbOy3bkDrbkGbbtpDbtrbb25Db27bb29vb2//b7/bb/7bb/9vb///dQDDfRjPg8/fh8/TiTDbj9PHkUTnl9e3mVzzn9eroXT/pvIzp07zp09Pp6dPp9ubp///rYkLr9+PtaEXt+N/vbUjv+Nzw+djxc0vy+tX0eU70+9L2flH2+874hFT4/Mv6ilf6/cf8j1r8lF38mWH8nmT8o2f8qGv8rW78snH8/cT9t3X9vHj9wXv9xn/9y4L90IX91Yn92oz+34/+4ZL+45X+5Zj+55v+6Z7+66H+7aP+76b+8Kn+8qz+9K/+9rL++LX++rf+/Lr+/r3+/sD/tmb/06T/25D/29v/6bz/6dP//7b//9P//9v//+n///+rkBe8AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAcmUlEQVR4nO2di58lR1XHZxOzjqIyvh1WJUFXZxdWXUXGxBeTBcEHq1cHo6KrMqIicRF1IWbQZEFHEXw/EHGMrG8hRjCKokYwgBARlIczy2Rmtv8Tu/pxb1d31amqU7e6q/r+vh/YyXZP31t763urq87pqlrKAPBgaegCgLSBQMALCAS8gEDACwgEvIBAwAsIBLyAQMALCAS8iEGgtcgY+vNIiigEGroAMpEVJ3IgUIfIihM5EKhDZMWJHAjUIbLiRA4E6hBZcSJnvAJh+NYLIxYoyKvG9Y4RAIFSfscIgEApv2MEQKCU3zECIFDK7xgBECjld4wACJTyO0YABHJ7UQSJWkCgnl4UAoUDAiVM0gL1fz+BQG3SFiidCyFQQBLywOHCBelSQ6BQF5K/CoHmSdQesC+EQL0RtQfsCyFQb8TtAbcrA4F6I26BuCchUG9AoISBQD6/CoEgkNevQqCFEaj/vjAECgg7SMu/kHsSAtEMJBDzXKAai6w4SQGB4itOUkCg+IqTFBAovuIkBQSKrzhJAYHiK05SQKD4ipMUECi+4iQFBIqvOEkBgeIrTlJAoPiKkxQQKL7iJAUEiq84SQGB4itOUkCg+IqTFBAovuIkBQSKrzhJAYHiK05SQKD4ipMUECi+4iQFBIqvOEkBgeIrTlJAoPiKkxQQKL7iJAUEiq84SQGB4itOUkCg+IqTFBAovuIkBQSKrzhJAYHiK05SQKD4ipMUECi+4iQFBIqvOEkBgeIrTlJAoPiKkxQQKL7iJAUEiq84SQGB4itOUkCg+IqTFBAovuIkRRiBJhKdA6CFxUf6aokgtcYikECf2KAQ6I0Ek5cTTB7RM3mMYPJRPZN7CSYP6Zn8M8Hk7wgmH9QDgdpAII5A+2eulP9xcPvVxo+SX5Qw1sB2ufvD+fnWqwIIFI1ANy5/X1nfR5tnr85+VDgKJF7u3DXduT9+AX2t6XwDCBSNQAe331c0ObtrXy9+Vj9qfknCog4IgQ7W6ZbJdL4JBIpGoN3z2fZG/vOBa8W9q/pRwxToaPOpd5y9+vhL19a+If/bwV1ra0+7cuPy2trZ+9a/4mVrZ77/rrUzl7KsOn+0+bT82KXi/FXjG5RwBKL33hJAIHeBjp5/JdsvGw1VH+i3JSxqqRYo71gdbZ67//HNc9eONs/n/z93TbQwB+tn7z9YP/MDR3edvTo7f+ZKfd4WlkDG34BA7gIJeY42L4n/nKtA+Z/7axt5Aye66A/89B1541IKdL44eePy2av1+frvEChBgcR9ox42qQT6HQmLWmoItFveFy4d3XXmBX+y2RVoeh4CpStQacvB11zJ5i9Q0cLk7K9dyma3sFYLlGUZBEpYoO2yzoputEqg35ewqKWGQKKPU4pyPm+NRJ9HFqg+P/s7BEpNINGFFuyL4c+8BcoeF6Ov+7PslfmPl+V9oVeunf2FhkD1+enf8/P3W8pACcTa3bagk+kZJsWUDBYV9QcSlrXbA6RAjsdndFugjxBM7iGYPKqHaisemtypZ/Imgsl79FDtoaFB/IAeCNQGAnEEknNhB+v5uGj2kf6hhEXN9gQEikYgORcmQkJTozIIBIGMAsm5MBFXvHF51gT9kYRt9YYHAkUjUCsXlmV1YLrAUSB+b90RlkCMXBgEMgnUzYVlu418+p9KGOtVqgGpKuYMRyAzEMhdoG4ubLeZEodAEIgUqJsL2210obPszyXcagACLYBAnVzYrvxIDgSCQKRA7VxYqdKMv5Rwq4EEBWp3/YMnA9JGkQsrn4qfjcJUArVijVINJC5QpwX6OQL6m/sveqgE5Qcn79NDJVo/SpV18laCyccIJm/TY1OvfyNRHGrHGrU1IFVFda9sJmrN1H53gUDpCtSONWprgBRIrVLrKARKQaCqNRHjsY3OR/oWienhZqxRWwMQaCEEqnNh2+eutXvQmVagXc3cHQuBDm7/mdzUo821M1cOvrZ4tL6Yi7EvggnF0Wx25MshUPwCVbkwzZf9byXqo7u66Tc2Aq1viD57MeRbv1TcDHfFzIyiYyWONo7sarpatgKZkxemVAYEMgpU5cI03VulQPp6tREob+aKhqiMGYiflbz5j+JoeaQIkXu2QK5tEQRyF6jOhe2f+3nV4Fx+QLY8pm1/LG9hVxs/ipiAuG+Jn2ULVB3ZhUBJCFTnwsQUCcXgXCGQoqekrgGpKkoX8ndrC1Q9iJ2/aKMFqgsGgWIXaJoL0wzO5SuKQ+1Yo7YGpKoQnfTc1I3amvr2Jd4z/ykimdt5C5QfrY7MpQ9kpY2m+BDIRqBpLqyuzBYKgaxrQBaoEG8jqwXKzb1UvHk58hIiP/OymB9/qToibmXfiBYodoFmubDtjTKhIfOwhFsNtASaK4EEamV6kAujkXJhog3o3pcSEUgz/8u/BXoDAZ2qp2YkPp2Aag/oND71bMD/EtDN7IN6bOr1HyTcaqBPgdR/gUB9CFSNvfIuhyK8nK5AdnNRZSAQQ6AqlSEGSLvduen/KOFWAwML1D5sAQRiCFSlMhoR4SauAuk6W/MGAkUjUCOVoRDo7RKO9REQR4HYuTAIZBKoTmUUcbvuMGwsAtm9KgRyF2g6rSfvRD+zG0j8JwmnOg4KBIpEoOa0HtUtDAJBIFKg5rQe1SMd8lpW1vUbHAgUiUDTVIZofba7w3gIBIEogRqpjP01VSDxHRIONRwYS4G62Q2STvAheDYpbSw+0rQFUp/X022BiObgY5PfJSBndL6ZYPJveqjm4EFydc6HCSa/R+CbC5OfQ7Crhj6AQPEIJC9xlw/Lmg91QCAIZBCotd1T3pFu9qXlf4BDDQcGAkUjkLzEnehVN0fzEAgCGQSSl7hrp8TkhUDcKjkkrgJxc2EQyCRQa4m7kQpkCQRyF6i1xF1boH+XYNRJICBQJAK1l7iDQBDISaD2EnftTvQ7JRh1EggIFIlAne2eWsP4BROoHagfJkOQDKrtnqq1VWreJcGok0D01QK9k2DyrwTkN/e/CaiFNOkG8cN6Jq8nmLyQgNyFzOIjTVIgl/ypBASau0D/IVEfdVzqMACUQNpDRiAQRyA5F9ZCKVCZ9BgUCBSNQK1cWAt5hezyWJn0GBYIFI1Aci6sjUqg2b4+w8EWyDGVAYGMAnW3e2rybon6aMoCUa8KgdwFUmz31AQCQSBaoO52TxLynh31UQgEgSqButs9yUAgCEQK1NnuqcV7JeqjEAgCVQJ1cmEtkhZIOdKiXrWd6UEujEaVC2vxnxLM2g6AnUCKCx1bIGLFy/dN/o+AnFj4dgJqPUy6rfh7PfTEwj8jIBtEi4qCQBDIJNBsiTvFIpvyR2ZVt70AgaIRaLrEnXIbOQgEgQwCVakMzUr175ewrd7wQKBoBKpTGZlyG7nxCeSWC4NAJoEaqQzVNnLy+v3e9T432AKRQCB3gRqpDNU2ThAIApECNVIZyq1x/kvCtULCAYEiEWiWylBvIweBIBAp0DSVodlGTp7Q5Foh4YBAcQg0S2V8b3Er64zCEhaIMTGjnelBLozG4iOVM8WuFRIOs0DaowTdFohYuPIeenFKKuH+LoLJI3om7yYg58IS7cib6HbtQ3ogUBsIxBGInNYjb8foWiHhgEDRCDSd1pNtK8ZhEAgCGQSqp/Vk+09VtEDyFa4VEg5ngZgrlEEgo0B1LuzGT903aoFsXhUCuQs0zYXtbqj6QP8jYVW3vQCBYhGozoUdfY9yYiEEgkCkQNNc2PYl5ShMjlnaVW4fQKBIBKpzYT+7qY5EQyAIRApkmtYjTy+yqYR+sBPIdjpPTSdQP0yGIBkspvUkLZDbb2SqFuivCOjFKal8KZmifVQP3QJRpSHzpUT69q2TZ+ixSWXIu6jZVEI/QCAI5AUEikegxryw7jPR8nLDNpXQDxAoGoHqXNh+d8PUTC1Qa0uxQYBA0QhU58LquT0y8gIQ5bHWWuSD4C4QLxcGgYwCVbmwGy9VPtKqEKi9G8IgOAtkBQRyF6jOhR09/441xbQM+UHf4lB7P5ZBgECxCFTnwsRD9UfffbX9kUIgCEQK1FziTqWFvOxWcQgCQaCZQM0l7iAQBHIWaJoLE8mMg6/rxIHk6UXFoTF3oluZHuTCaKRc2O6aKrqjECjNYbwV3RbobQR0e0BtRkfmxqk0/lsIyBaIbEmJ/P8jkzv12KQy5PasPNbaUmwQIFA8AsnTevbXpNG8SqAYgEDRCCTv1iO60/uN9kVeGHg+1TQPIFA0AnV362kOsSAQBDII1N2tp7lQmbzE+nyqaR64CGSVBtPkwiCQSaDObj0H680+0CgEsn5VCOQukGK3nmYfSN7T3LomggOBIhFItVtPsw8EgSAQKZBqt56mQH8tYV0TwYFAkQjUntbTzmhAIAhECdSd1tPKaMhvZV0TwVELpBlcWb9qO9ODXBiNxUeamEDK3w3WAj1MMHmDHmpruA9Tk2HpKa1/oWfyQwR0cR7TYyOQXA7rmggOBIpHoPYSd9KjGhAIAhkEknNhWWulO/nRI+uaCA4Eikagdi5MXuluFAKFSmVAIEErF9Za6U4eYs5XAh9cBLIHArkL1M6FtVa6g0AQiBaolQtrr3Qnb/LgWT9zBAJFIlA7F9Ze6Q4CQSBSoFYu7Ki90p38L/esnzkCgSIRSLHEndQCLZhA7UD9MBmCZFAvcScJ9HoJz/qZI321QGTa870EZDKV3L3rPXqodTQ/MHm1HqodeWzyIMHkI3psUhmpCOSaNlUDgeYukPxUgGf9zJEl5VEINIRAVS6sNSGsAgJBIINAVS6sPSGsQv6Xe9bPHIFA0Qg03e5JuTpHsgLZJ8DIXBgEMgpU58IyeUJYxa9LNM8Mu0CHWSDOq0Igd4FmuTB5QliFVqD64Y+BgECxCNTIhan6QK+SmB2fTYQeBggUiUCmJe50Aj2g3F2sPyBQJAKZlri7V6J5BgJBoMy8xN2CCdTK9CAXRmOxxJ28i1pxaLe8441SoE4L9EaCyZsJyBaI/MpTu9i9ioBsK4gU/z3Usp6PkvvfWXykCoEqIBAEKj+1MpWxX3SmN9ofqfxWzTNRCsSZjtoAAjEEmk7rUQ7j0xKoAQTqS6BZKqMOBzV5hQSnUsIAgaIRaJbKUGQyxiUQIxcGgUwCzVIZNy53G6Ds5RKetT5HeAKZLoJA7gLNUhnKTg0EgkCkQI1Uxq5q+4KflDDWa29AoEgEaqQylHteQiAIRAo0m9aj7AJlL5QwVUB/QKA4BGqkMtRbgKUrkArTRe1MD3JhNBYV9cMSFhf0hFEgFYwW6CGCyTsIJq/QQy1ceSe5Vua9BJMP6aEby/cT+E7rgUAQyCRQNa3nYF01rUeeRc2ptTBAoGgEqnJhIha0282FQSAIZBCoyoVpttL9QQlOrYWBJ5B7KgMCGQWqcmFiFL/fTYaNSiAjEMhdoGku7MblNUUy9bskgtQaCwgUi0B1Luxoc0PVB4JAEIgUaJoL0/SBvlMiSK2xgECRCDTNhUGgDAIxBJrmwm5cVt7CvkMiSK2xCCRQO1A/TIYgGaRcmDqQuFgCgbnz7RJDl2YGBEoECAS8kDt4Q5dmRliBjp/7W0Ff38zh6omL5t+6/sl3hy+KH4sp0OGpTsVsLeWc1Py+9uTh6lLFTd2qJk9mO0tLp03l1AgUpjins+vLS7e4frOeIeF4cUB6Fuj68or4saNsFsiTeXN24TTRVuyc1J0UFaoztn5n1YVEcQ6fZG7Wji+oxN3JpTq+sJJtGUrUYaEE0n4HCwcEO4pvIHky8xDo1N15Zapek2wrqOJYCLSjlLb8N3zSRfeb5ldLuF0bkjACfcGnlDyhVTHTD171+ZEnMy+B9C/KK45RoOvL6ma0uFAYOTeBDtabW7L0T7+3sOm3WvX5kScz3xaI7Eu7Fsck0NbSivpEcaG4fV3/VMdO0NMlpodF+nVfEXfsjZ77QFWH4nBV9QGTJ7Pym0v2HfR9IEOfVXkhURxaoD3i7fLyH66eLrpBbmgEEjl89Tygnuh7GH99+bReEfJkVoyKnEcvYhTm2mG1LI4GehSW/xtWcn+ci/RVEtIp1WIevTH2QGJem8ZBPOtVCUeCQAikWsyjN0KOwrofrl1YjwEZXdIygAdsvlKieUaRuu+RcKOw5aVPcA3r6YNsdFVTERu2I+wrDRfyXFcJVC6yuTtkFzroLUw90CDCekSQjey1+gWQSPbc+1xVUVY0wYHV4h+35dzkfZnE7Piw7U9YgRzDelSQjRTIL4BEohHI3DrpBNqquuQ7rm2QRqBqZfLhCCTQx1902zerv7v6sB4VZLNrgVwFmnmga2h0ApkCicUwnfp3OPv8pRLTw9vFk9SjG4XtLeeVckLZ2dGH9aggG11jhgCSsQXa0vbLmALlZqrjmnMXaHCCCHR84ZYfP/Ut36T88ImwHhFkM9SYIWJDRiDJbj1PoK2lk+JWvKW4mH0L+xIJx4sDEmYUduru8n/dU2RYTx9kO3xyVRfHz+Z0ZbQRyD19cah+Dt0lK2IVe+r+EbsTvVAC5W3I4akX/4gii80N6+V3vfKDZ4Si9RyuMuM/tEA/Rl7LHMZ/noTz5cEI0wfaueklqxbPcWnQPJ2zdHpHP+gRteKac8h88hx9s1gCZa/5zec85Se0ZzXxmhpNB3NPqyR9X6C/8nnbpr6MvIXxn0jk8rkS83tdX8IIdPwiQp8sMzyPpxQor5YVXQtUv5zqZS06Heobq3Gorgv1CHQDP75ciyVQ/qU/8W2r3Vphf3NnfSCFI8fPIsbG1LCHjAOFEYg9jP8cCbdrQxIomfrFr/38FUUn2g7VB0+NwkStFDUz1xozCaSLFVYnIZAHopb3TiqH8Ta4friitoqaUURtggmkjRVOizRfgT5bwu3akAQahd18260nbuX2IZ0/3LxTLmqmCCe2sbuFqe6opEBErLAukqpLNs26OGdpF0qgqmLan99OgEe7CvIe0tKJi4eKTpe5E83KtJKxwhJN7HKvDDbsOMc4PkvC8eKAhBHouT+a8+L2YZNAVQxaNcj3GRqbhvG0QMqQgyFWSCFk5zyZu1AClRx+YfuJeoNA9d1GP8hXT9bzwpxpjSPS+JkSQ5dmRhiBfnm5aC1aOXeDQNOBlH7qjroh8YrcWTzr4RhIrAs7z6TLggmUdxHErMKlm58ofYaeAukm61UvrnkAx/jIj3GykK6oe4b5gcSLMuT6DAnXq8MRKhs//bOBzS1MTMlTP+qgnaxXvTgRlcmoR344k4VIgWyaJ/f74kIJlL3mdYJfax01jsJEAPt56twUNVlPQIb1bBbocMKqBVJeyL7bfrqE27Uh6Xdaj+8ral+UCOsRj/wQ0LKzBeKzUAIdP+e2nOVPU7UZukG1V5BIH9azeeRH1SMxFEfcgrbyLpnzg4WjI+Qw/smqeI4urOchEB3Ws3jkR9EjMRRHRHNu+dVlxTsGi5ZGShiBPi66QL/ilBs33DPIpIMhrMd55IftAXXhGOUK2QdySm36dDqIEPasRG6VZ2yBTopnJBXjQgg0B4o+0K2KKgskEBHCNseB1Jj6QCvFH4on3CDQ3FBNbWbfwsjAizGE7Y5FcUR53L4IEMgWbR+I24lmCkS+qE+3HQLV9NsHytizWshxszaE7SmQvqw7xS3stOOtGAJZUsaBnuKeTNTXmH7cXF2nDGF7CUQ+SSTmETk/SASBwsKetanFSyD2LGR2M5si/a7OYVg+UF9j9bx37SOkanwE0o8YvZquscnV++ocesiFmVermIt66pdOy2EEoh/DnncrOzC9r86hh17Z+3nl4+jaUZhyiBbmFsY2j3rRVOl7dQ4K6hb2pIvlXcxNIC+0jYWHQPxpPbHS++ocRB+IaN7FB18kLJS3MFog3v5AxEmPWxgEskO3Oofp89Pvr1RcuKOOLRmeDyQ6HbweiU8nGrcwOzSrcwT6AlJRRrLGeNXpE4FEJ9oLpkDmCWXaKCN/f6BQq5BjGG9G+0grObOXORo3FMVngymKkXnAJtxK9U9QrVTPm9nrMxoPcAvL6DtRShso+BPycY4nKg6yZvZ6xQPn34nO+OaNT66QAnGXd+ngl1RnDuOtVmkdzWicTcjNVlTtzE5ZJ6o5gmEyEmwWKx7IJmAn+uZv7ZyYrgGtjQcKensujISdkRjfbYoiomF8kDE+fz1Vwxif6ETTSeGRydXn8i5+AgWZkUhgWEyM6Dwt1L2vz+VdsumtS7mdgSFI1HsEl7uYGATyRrO8i6DqRCu/ulSNGcbNQcJ6zMXEqC8CBLJCs7yLEX2N0R88lcYPsqY8fSHxReAvshkrYW5hr5v3C9ICcZ+fMLQHRMiBhmi62ItsxsogD9Xr5iATNUaOm7mdDvIkGXLgw11kM1aGmZWhrBM6SMQdN/Oz8fzuCrvpSpCe40DswAvZTyaaJ7LTQY/7uOEcuukam1wRzQvz2JWAaJ7ITgd5kgg5sJuuQPfFAYlIIDJIZLpU3zyRnQ7ypD7kwBYIw/igEDUWVwrAMChkNl1J0ncfiDXB1Auy08HskRjCOUS01KOVjZPeO9H6CaY+r6ptnshOh02PRB1y4IdzqFB8ivSdjScmmFbMd4sA/5G6Wq6xhXPY9P44h36CaY3b9HfzO5b/0fusjDnvlREr/T8PpJ9gGuQZUrLTwe+RWHSe5rtXRqwM8UCZZoKp+cKM01aQnQ5mj8QQNDc3XaMJA8U1jNcTbHpXgOKQF8YVjpgDiQzj2UkHPlRZxxfOYRPRMJ60ixo3W1Qn2elQnyTLOrpwDpuIhvF0kIgYN9s8pUV2OpQn6ZAD0XkaIlo6HBEN422CRGqI5slrVoY55KCC+iKMT66YhvFkjZHj5hBhPbKs9IX6L0KYUPyQRDSMp2psgMcgLEIOms4T8UXgt7KxEtEwnqqxUE8ve6LpPOm/COz7YrT0LJB5YyY17KeXyU4Hu0di8WSl/ovAui9GS88CERsz0dVJjZt9Fpg29khC5HYZofhoGaAPpNuYia5OatzMX2DaokcymrYiDDEJxO5g0o8AkgtMEydDZOMxjPdEuzFT5tXBpJ9eJheY5o3UbTzg3jTTou9RmG5jpsyqOt07JOYFplkPB/C27sAwPig21RnJF5ftAYbxA2HTIXHvdBjuRHvaN+SnOTCMD4h+L207nDsdhnFfrpXYFnVZvWUDM82BYXw49Htp07A7HeZNpIp9Ubu7T9t5MJrbFEVEAhFbIXu9KHcYXwq0wsnGl14p45oYxgeDEoifdOAP40WDKH5D0bcyFSe/Eat35MUwPiT6vbT5H7zPMH6rWOVxR/0gkak4WzfdjWx83+j30jY8ytj3fcHGg8PVL1IuroBh/BCYkg7c+8LxBdZac3YeKJabwTB+IAxdGWN7oAthH19Y4bRbGMZPSUggqitjbA80NZ0LlPd4nSNP4/OATSICkZCPANIhbCFQbkJ3RXRgyVgE4uZEuU9sjG2pQzZpCTT3RQnKFohx2diWOmSTlkCxVBimNk9JRCB2Nj5QcSBQTSICWdFj84S58TVjEGiINVPGttQhmzEIBAYkGYG28vblJaujWZdpNKQikHi2Ym/pdLaHm0ZcJCJQMewRO9hpNpwTCQn0SIYgJYF0MxKLhJY4rtkPEwQkEYHIW1jx2KA4jqhM/6QikOhE3/Ib6k708bMulmJBoP5JRiCK3B4hUflMLOiVUQhUTQFEbnwAxiEQGAwIBLyAQMALCAS8gEDACwgEvIBAwAsIBLyAQMALCAS8gEDACwgEvIBAwAsIBLyAQMALCAS8gEDACwgEvIBAwAsIBLyAQMCL/weIJzulfSnKvAAAAABJRU5ErkJggg==" style="display: block; margin: auto;" /></p>
</div>
<div id="function-output-23" class="section level3">
<h3>Function output</h3>
<p>An object of class <em>ggplot</em>.</p>
</div>
</div>
<div id="plot-results-of-t-test-olink_volcano_plot" class="section level2">
<h2>Plot results of t-test (olink_volcano_plot)</h2>
<p>The olink_volcano_plot function generates a volcano plot using
results from the olink_ttest function using the function <em>ggplot</em>
and <em>geom_point</em> of the R library <em>ggplot2</em>. The estimated
difference is shown in the x-axis and -log<sub>10</sub>(p-value) in the
y-axis. A horizontal dotted line indicates p-value = 0.05. Dots are
colored based on Benjamini-Hochberg adjusted p-value cutoff 0.05 and can
optionally be annotated by OlinkID.</p>
<div id="function-arguments-23" class="section level3">
<h3>Function arguments</h3>
<ul>
<li>p.val_tbl: a data frame of results generated by
<em>olink_ttest</em>.</li>
<li>x_lab: Optional. Character value to use as the x-axis label.</li>
<li>olinkid_list: Optional. Character vector of proteins (OlinkID) to
label in the plot. If not provided, by default the function will label
all significant proteins.</li>
</ul>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="co"># perform t-test</span></span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>ttest_results <span class="ot">&lt;-</span> <span class="fu">olink_ttest</span>(<span class="at">df =</span> npx_data1,</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>                             <span class="at">variable =</span> <span class="st">&#39;Treatment&#39;</span>)</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a><span class="co"># select names of proteins to show</span></span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>top_10_name <span class="ot">&lt;-</span> ttest_results <span class="sc">%&gt;%</span></span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a>  <span class="fu">pull</span>(OlinkID)</span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a><span class="co"># volcano plot</span></span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a><span class="fu">olink_volcano_plot</span>(<span class="at">p.val_tbl =</span> ttest_results,</span>
<span id="cb32-10"><a href="#cb32-10" tabindex="-1"></a>                   <span class="at">x_lab =</span> <span class="st">&#39;Treatment&#39;</span>,</span>
<span id="cb32-11"><a href="#cb32-11" tabindex="-1"></a>                   <span class="at">olinkid_list =</span> top_10_name)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABJlBMVEUAAAAAx+EzMzNNTU1NTW5NTY5Nbm5NbqtNjshuTU1uTW5uTY5ubqtujo5uq+Rzc3Nzc4xzc6RzjLxzpKRzpLxzpNOMc3OMc4yMc6SMpNOMvNOMvOmOTU2OTW6ObquOyP+kc3Okc4ykc6SkvLyk09Ok0/+rbk2rbo6r5P+8jHO8jIy8jKS8vIy8vKS86f/Ijk3Iq27I5KvI///TpHPT6bzT/9PT/+nT///kq27kyMjk///pvIzp6bzp///+HwT+Hzv+H2f+Szv+S5H+c2f+c7f+mZH+mdv+vLf+vNv+vP//SwT/Szv/S2f/cwT/czv/mTv/mWf/vGf/vJH/yI7/06T/3pH/3rf/3tv/3v//5Kv/6bz//7f//8j//9P//9v//+T//+n////CaCeDAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAY7ElEQVR4nO2djX/ctnnHc1ayLMlWyd6sZJM7KcuU2Wsjb75u9qz5dJVvauX2ern4rrV0lmP+//9E8f5CAiRIACRBPr+P5eORIIgDvnzwAASITzIQyEOfdJ0AUNoCgEBeAoBAXioD6Ofv9/Z+0VpKQEmqDKCnn7+8+fJxa0kBpagSgN7/EuABVakEoJuvnrkFBI1YJVy8/fw/9/aO8dY9JAAIZFIZQMiDfvvp4+qAoBGr1AK9RA2x4+qAoBGrwgcCgEDlKm3Goyrss2cOAUHjVVVH4rFLQNB45cwFAAQyqTFAs9nM4Sy3UKB01RSg2cyFDbdQoIQFAIG85AVQJRuzmVs4ULLy84Eq4WDHAaPByr8V5o4GmKMBKkwzvhYWwNGQFKwfqDYTgNEgFLQjsQERYI4SV/Ce6GY0AEepKsajjOYoAEbJKdazMEcQZmEV4AruPy9sOlNVzIep1TkVOCsN0dW9glt4z3QPiaDIT+MrbrfQGWkAyD+KpoEint8jtTCcowQiACh5tTQeyMIQAJS82htQZjJEAFDyandEYh4iACh5tT+kVWWIbq3Qrot3WfbhDW3kzsXWldiJjm8v3u1mC3zCek5Pu8LbNOS1Ep92tUzGO1so8YoIULQf3lzlklT1GzIt4Wty+R2JTv4YEecOf1c+ASBfCUNE/l9hHHAZyDxnW1u5EwUiAJGiIgB9ePN6ziNcXvKIC5din1tSeEq8IoLGAH14g6+6wnGt8NZyoe5T4lwycJYAUEARiKh9INYDMVAACJcJ29zNrglA/48LiAC0nm+Z3cnWvGRmhWciRoBIWfMIGgPEqMXwYONDEiX3iTh31CKJT+eLpKFOZ2XMKEB5bMoAul5eZbysPrxZsNALHqGMmWFUAhCLoClAO44v1npO7gNln4hzvaDX5p/OF0lD3U7rIRm55j5JASClXluyKux6O6cA4Zt+Reuwta1gFFukAUR8Fh5BU4AkDiR9v13o+5Q4+V4AKC/vB0KKM7rIFNeZb12LnbSk8R2+orUF/mN12HIhElRMIvvkAEmXm0fQGKC5umdNHXK5DwByCOiVD2RMtfi2mxV8oC0pE7mTAYTaMrjsqcu6yLR6oxogEa+IIIgF2l38hL+CBaoV0DMbNID0cqRba2qBcgBlqwUCaMdb/LxRb0mRqQpjNRiLwNcHIo1zRCMGUtkHAFUHDAMQy1YDQNjxMQD04Q8/zpn7sxVtaFuKjD4QjldE0LgVxq67UutTuQ8Aqg4YyAItab/bouhEb2cLA0DIgsxZw4t8SBfIFaAt7lLkEQToB2JdiHNLP5ABoOEQ1AOAiA+tdDrPZImuZlcGgJAF4V1Aq6rSNzfjV7PXIoKfcBSyf7pOT/SS9jpTaogbJ/ZlSpwmCzQYgvoAUDil8ywMAMpYNm6JI0tuP2VTPsHiYg+C0A16KU8fL0CDISgAQISMLQFIbMonWFz0QRDp41+I00cM0FAICgDQa+JBvr5UN+UDCBqUPQgi33mze5Y1nANkk2VQfb0o3ADySrf6JDl5BQDod5ekaX2pbuYBYg+CtoIifrpfZ3axb9uQSv8ovGM1X2QQBAUA6Pd/RP9fri/VTaXDhYmwQ4yPBtCYNYgMCADQ9eo6W18RgMSmfILFpQJ0JU8ftYaQAyEAWi+y5TUFiG+qnTNUYIEMGkAWhABod7mbZxQgvikBwr2Eojet6AONXOnnQQiAsj/9uGAA8U2LBcq3wkDJZ0IQgPDodAYQ27QARAeuLpTTx67kMyEIQBgOBhDbtAGEe6IL409HrdRzodtnYaDkswEA6lxp54MXQN5DWkFYSWdEt4PqQVhJ50Rrb+fwOn3gSvl2am1eWLI51IJSNsgtvt7FN4LhCgByUrJ5FF0AkKNSzaToSpeftl8wFSISUJ/U8ssV0r3TQGa1/nYOIGhY6uAVd6EiAvVBHbwfCKqxIUnjYrNPdVoV0FNA0HAkubg72X9Itz4+2b//3B4wgMAIDUaCi7t/VZnRv2kBwwgIGoi6e0tr4PhA3ai7l2xCNTYI6VwgP+jBq+lRdcAgAoIGIL0VdnB2/uDV3YmJoCjviY4QJ6hdqVx8fHKUIYCy82IbLNZ7ohN+DA0iUrm4OzklAG0EQDdfPTMFDCcgKHWZLNAU/VE9/QwAApXK4AOdi57ot38DFqiWBvVj3FRohe3vH5yxbz//+39RgO4hxQFoWFk+sNvBSWVcvPhFdB9oWAKANL3/l5cAUC2NHSBSgWHRVtjT4/itsCFpNhtWhewmAxesFf/+0R7WsT0gSNH40KEycTF9yLfAArlqpPiYuWixI3EgCm1+EqoKTVy09yijl6r9aojgpZ2SM27g4u7kYXHneAByLLqZYStcEtIEiLfCxJMMW8BBy7HkxOvmI5R0qgCFCZi6agEUpZxT4gcAKqgOQFGKuUaksufuv09MU2lkuFPLjo14cNVQfQKoH/edO0CRzE/NE26/xggUENHUBkACZtETbQkYTT2x3K4AxUlr/Uj7AlCwgI3VM4DwGg3K4qf07dZywcU4+DSIlQP03SGeEXr37Q/4/j9HZgAPTL49JPNE+VE6d/SIAYQCHfzQLUBuXSamvhNbXDWTH0E0Cfnll+n35RV52b4MFuPSNcUBQiScH5zdneA2NNpA7Byhv1PMzCk/irdJNw0GaIpA2+wHBYhXY65VWM0fbA9O0ekFPyyVDJOVtnaeXDAmTu9Po9M4QEfY3pySTzotYnP/OX+mwI/iMaek2kIAEbiyaVCApg9enT+k2JYHpKr9g61LCtaNKKZIYsRSi8pS4WQVGT1Y8Os2kOIDof/IJ3VsCE6UIH6UFi05cErhCuoDYU43eFaGqScxEkA9oyfjK4DI9aY5QIie1WItFtEL/vSr6ZkmgMRLMj4+YT6QChANdh4DoNPs9pvn5K80IJMvQP3wefKiAMm1FgVAc+RCLzJuhsKm3CM2qwXimpIaqw0LhGvIu2/P2gGol/BglVgg4hix2i1o6n0iMwCkt9qVvUUf6DyoD4St2vSohSqst/RkOR9od/GOA7RcYDcoBkB+mWEAiGKBLA+xLxtpgbRWGG2VhW3GT3HUxkaYFSClm4RtoptXbmGJ25l2/5fS0z1a9Por1gqbq60wuuRr6FaYZ0wmgEg/EC7GDZ1lI+1Sr/qBGECim4Rltcz01ZyGkAA5dQa5JimOSvqByOYqrA/U9c/1VSCAyA0qN5WdKMd/KwCqzKzeAEQ6ndWeaPKD8HbYnujE8cm3wkwjyQwBmVSAsH9gBmg716uwUvUIoDDBKiJJnp9cR6I6L7UsIJUGkFwodTlXttDmdQ2AeuMDhQoWPY6uleeC9EE5DmnVAZpzYz/PlC1smeoA1H2utgZQ57dKEBm4+PjE8VmYxQIpO2mPXFoAORWsu6EM8qC5xwpmgZYLI0Armj3X8oQq9SArA5a3a8hkCdJ7opv7QEorLFMBwkrMAgWVO2kxUxFRgVphaj9Q5gdQsllp1KgAwqJPSxwCZnpPNKUIAMoLALIFzIKNB/KOts8CgGwBs2gAJZuXJgFAtoBZ/ZaDc4ukXrS9llrNz0iX6p/Z45ErvaZP9Uf7AFSTIPeukzqx9lxaVwf2BolDuFygZuvVIAGqGdC1m6xWV1mqeWmSESAC0epygACRaRnGdyu0OrU51cw0KA/QT6zFOkyA6ABZ8yhHAKiRzFVYlu95jfOb8dSwzDxCmYfIH8M7puaJXWbpPdF0KO3UcUhrNA0PINZXJgBCey7jO9G3hxiEMoBMwsPibREWo9J6otmZm67fUDY8gIoWiA5fCAfQZDIp7Lv95tdH9QEqCV8BELdAroPq42k4BNkBkuPuZMDmmkwMBN1+8z+/orO0bg/xQ04EFJ0yn5Gn5gdn/Ng/Pjllx26/+c3h/v53dPc+G0T9kJ2JXOSCe5NbtZkM6Y++anOlZvXad80bfNFlBIgOzm8DoOcbXPjPyYSM+8+xT8SqF1zbbB68YsdQkbNjaAf9xyYJkjMPzuTR/DWMLxo3TY9vGSA/CLSTOwWqABB1nq9zA6iiAfTxP84oEBgZPIGDMcBmb2DEUH2Dp4nRY+LfRlZD8swKgErV7hvKPMs8N4OxQ4IKAJHprWxwvjpeP44PhGqif2I0fMTVFKnQznGlhK0FsTHn2jHxj/sxuCY7OJNH89cAgOKq02dhpLyn30kLpM9bp3USt0A5gJgFwqZHnlkGUMvrxldULAkCZL5K9wDhtjz3gSRAGAy6h/tAOYCID/T1GXaZptICfV1o4UsukE1j48k+PjF0JAUGqKJYAwLEp8JGxsjyg7oHiExY560waYG0Vtg/qDaGWRraCsOjVP9ZVHDoW76BlWuF8deCFNUuQJ75WTh7Ft0Q2eJ3fW95py3Hkr7DKnXlAzkAxAdBzBb0dWErNgF/wQ/M6V5lMOSavcqwGHP05r01/r53PLDKrOnpXQJUepx+0G430m/y4c1r+naVK7X1i/bSg3jXumTsbPRi6lX3Ux2d22dSOEjhgkzKsLwh0ROgYuZWZbYG0BZPC9pd/Bl/WcsXrmCt5+Qg2bWe586uSAMogCQX57KT2zSmzAeg4t1fWZYaQBSPS/5yDBWgJdq7oCEkP+boAaAYElwoQxGDPwvzBYi8UmV5RV7wtNQG8uFxfSsyl/pqLd+LaYseCAov5U31ou0V/Gl8AaDqktQBIk+ur/E8afokYMZ7cLHV2dJdl0v5Wjpb/EBQcAW1QFY3ozY/OYB2F+/wJHtEz3auDaNZkuprQSY20ndClV8ACAot1Qfiyyc09YGcGzq1AUIVF3k/If2QAO14e57s2s6u9LObXRpURwoXpevNhQSoRhgO0GqBVxlgH8qrG4jfvOUDI1bcDSrr43a4OMhdIfuBIgK0pa/Jox8CINIAY32L4o1WVVfojCDT8/L0FbQjMRw/eYB29NWE1M3hTvTFT8zirC7+wt8R4vAGzI4IMo7YSV/t90Q7PhuKeI1uCAKAAql7gLohqAuAyMOFhyFm71jV+oqFjoUXcEhr80SEVQcGaPoQQ2R+61O92TtWSS7Mr0Y0BPRTednJTHZ7iN304Xaf22LNQDP9aMqIBYd6s3es0h6mlrygLBRAlfy0dJv298FYszww3jaiRP1n71ilcrExjiQzBGyuimJr00/oniDzbw0IEPFLMEP+s3esatmJ7hFAnRNk+bEhAcrYSGX/2TtWtQtQZZm16mh2TJCNlHA+EBWiwn/2jlU9A6hddZuc+OaWDqug2HjO3rHKMjO16AyFAChegTUsis4JinuBj0/IGuDaDOaGs3esys3KICsi7tPFo7ObL/b2jo0Bm6lxcVXmdOXNbDvcM5MYXFP6cNx/9o5V+ts5aKsPIYurxvePjrO3nz42BGyopqVVbeurQtiPD50gTR6zd6zKr9qMxVasf/v5y+zn748NAZvJxwBFA2g0BHnO3rEqv2ozFrZA7FLYCmXZPaReA1RVyU1aJqiPj039Zu9YZfaBRBfmC2SFDAGbyKOg/MujjMHwBA30wbtRhre0YoecGaAXnz0zB6yvzhs89iINnrTxApTTC+FCpw5QuREL/WAMAKJS7I83QD33VcMTFDbC/krnAncKCFfr5svH1oC11dkoQMfC7Dng/ZXBiebze57uYYVpxnc4itSxOgGCmsnSjC8PWFvNS8erMqgDUHkaR1Qn1ZSlI7E8YG3F7AKqOts5hpJEBvKKh4hhKxbIywB5mqAapWZPZhiAnGNJCbQSH6gkYE11BlBNWdPZLkBJ9QKUtMLKAtaSj3/ablaWEeQf+wgAChGwoJTaN3HT6l6DAUCKUgKoH4lNiJ8WAOpFkbgrseR2rvgzU1Mrkf7OGOulolugBIsjwSR3J69lvx2UZGEkmeiOBACZlGaqO1FkgFItiVTT3b4AILOSTXjbigtQ28VQuwPF7yUy0GCL3ArrgJ9aBPkTAASNGyD/S46eoJgAtZ65NQb/EMUHKKWHEs00KIDqEhQdoKQeizZTLYAKf8iJaPZnic/7bzL5BJVYnLiNf7PS4xOcnjDX6qt8LFBzFzRa8wUsUNvyAqj5ZaPVbk4FxgPhVMgFWPnWtdy1peuy4hfgs51i9SAWFfhAHgH7BFCdkhJ2gQLEV25RluDgu7YztuSmWCaRLu+ydAVo+BoIQLXqijoAvSarAL0WAOEF73bcEGUAEABUAdDvLtH3P/yoAbRebEsBGn69pcgbIGX9ydx67tidYAFogazEgnDdApTzgUoB+v0f0f+XawEQXVCqFKAReM6KAgBEMpYuXCoA+fAGZ/QaE7NckKVxxRLv8tSQalJmE9WJvuJbV8ou9KtWyOJcrYUTTX8BACQUCiDMiQIQW4J7dUnzGq+6zJd4l6cGV72Cm0wcLND1epEtr9f6WuMNAVJ2DwexOABpuZ0xgNgS7/LU0Kp567sBtLvczbNqgNiVyYelP1zZOyAjFQqgtVaF7ZQ13PlXvsS7PDW0eLE4T79yASj704+LaoDYpZUUFNMAAOWlOdHXqjtBMnjL+t3wmu6ZXOJdnhpauVKsDu8EEHb9SwCaKJeeAEC1AkoLtGVect4C8YzG8PAl3uWpwVVSekapTvTFOwUgvgsDhH9DGUCTLHMDCHygvJQqbE0tUP5mlmvmXosl3uWp4VXPAgV5FmZ2foYDSJVC+UC4f0S5SVeiuUvaYwggscS7PDWcJqodyJVeSVkGBEhcazzoUIXrB9Ka8bTDbU2eISFk1pdyiXd5qr90cIy2p6xIgwNkuBr/PlSwgjXjka8pfYcMw6P0RF8K07OilVoYgPLgOAPEdvmPKpnN9MgLVyv3igagpJ+FFUrH1vaZGPbhz9qzMAoyR1ySxKFpEACV1g95giYTvqdeoTqELmMVACru6hwgMzg2I6B88wPIepYJ6MqTEpcPQM0xiDkjr1CK5QC5l685PtOlB2tuDPIDKJDrEFTFUjTWYJNMJahOkZeElegCQI0Ddq0KJlSAmtmM6rAAkE/AzlVqBuhecaRRpVMddDz8DBEgLiMUk9zeGj4QyKRxAuSGC0DloLEApDk9bgS5V2tj8nnyGjBAmc6P4hlpTnTZ6abKzhJwrAQNGSBFghodIHXDfE7um5ETAChowF5qokmv0LKiay0OmmIQ33OHYqW93xobQJlKgCh4EwLFGkwNpR0dMT8jAShvPjKdn2qAJH26KQONDKBM72cUCNgAmijbWkT6CSNmaXQAKRzlAbKcpB82ATRmazQOgBQYTFBYCMhTp5xfMEAAULCAPZWl8lGoKDklK2wVA8ZMfI81YoB022KAIn+OEUAReKQaL0DFL4XgZoJGbXAKGg1AuknJ+zJmgGzWBwiSGg9ARLL+MTFh6iayYwTCGhlA0niUYlGNDwDENFqAhDEyYGGERf0K/AiNDaBi2RdhsZgbYMek0QFUkEt1Nek6kf0VAFRGkDjWdRr7qzIufv5+77NnLgHTlgNAXSexxyrj4unf4n8OARNXFUBdp6/XKuHi/S8fZzdfPasOOABZXR/gp0IlXGB4MERZdg9p2ACZCeo6USnICaCKgMMRwFNbAJAuwKemACCQl8CJBnkJmvEgL5Vx8f7RKDoSQT6CRxkgLwFAIC8BQCAvAUAgL7kDVK57FcddFCCO4SYjJgQ+CpWwe/2IA5LRtgCgGHH0JBltCACKEUdPktGGelu3gtIQAATyEgAE8hIABPISAATyUiCA3u7tffq4Olipbr7Y2zv2TYgcwNRM2lSmrhIRKC9aURiAbr58nL31zPj3j46zt74Uvn/kmQptDFRXiQiTF+0oXBUmR78209vPX6L73++2e7H39343vz4Ks6NEhMmLlhQOoBfoR/sK33k++t+XnqWvjwPvKBFUvnnRkkIBdPNFCIvrD2EPAArhA2VhbsgWFM4C+fpASC8691/7A1CAvGhF/gC92NsjXqdPvtM4XvgYMZaMoQDklRdtqj9OdJh7zrPsQjjRIQBKxf6EAghXXzd/51dn464Af/mWXYhmvD9AYfKiFQWyQKgC8b1nnu5h+bY8fMtOm8rUVSIC5UUrgkcZIC8BQCAvAUAgLwFAIC8BQCAvAUAgLwFAIC8NAqC7k32q+88tITanVXFUhwCZNAiAsG6/PrMfvDupwqM6BMgoAMg1BMiogQF09+0PuB47R9XZEd47RRun2e3h/v6D/zv8N1TTHeFtzAoLcneC956SEK+6/QVpamgAnWAMzg/OEDQIj+lD+gXbl9tDtHFO8MJ/LMjdCd57cAYWqKEGB9AR+y/b3H9+9y3ad3t4ygA6yvh/pzLICd8BADXS4ABCHGwO8CYGB2mDaywK0Gkm/hNByAnoPwCooQYJEGvUn2JH5/5vDk0A8SAAkKcGCdABa5AJZiwWiJ8AADXXEAESMGxwx+LGVIWJIACQp4YIEGliZdODM2p89o+IW60BJIJIgI66/gVpapAAkU4ebH3Q58EZwiSb4n4gFSAeRJisKfQDNdJgAAJ1IwAI5CUACOQlAAjkJQAI5CUACOQlAAjkJQAI5CUACOQlAAjkpb8CAZYCh6GS5qcAAAAASUVORK5CYII=" style="display: block; margin: auto;" /></p>
</div>
<div id="function-output-24" class="section level3">
<h3>Function output</h3>
<p>An object of class <em>ggplot</em>.</p>
</div>
</div>
<div id="theming-function-set_plot_theme" class="section level2">
<h2>Theming function (set_plot_theme)</h2>
<p>This function sets a coherent plot theme for plots by adding it to a
ggplot object. It is mainly used for aesthetic reasons.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>  <span class="fu">filter</span>(OlinkID <span class="sc">==</span> <span class="st">&#39;OID01216&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Treatment, <span class="at">y =</span> NPX, <span class="at">fill =</span> Treatment)) <span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>  <span class="fu">set_plot_theme</span>()</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABC1BMVEUAAAAAv8QzMzNNTU1NTW5NTY5NbqtNjo5NjshuTU1uTW5uTY5ubk1ubo5ubqtujo5ujshuq6tuq8huq+Rzc3Nzc4xzc6RzjLxzpKRzpNN/f3+Mc3OMc4yMc6SMjKSMpNOMvLyMvNOMvOmOTU2OTW6OTY6Obk2Ojo6Oq6uOyP+kc3Okc4ykjHOkpIykpKSkpNOkvLyk09Ok0/+rbk2rbm6rbo6rq26r5P+8jHO8jIy8jKS8vIy8vKS86f/Ijk3I5KvI/8jI/+TI///TpHPTpKTT6bzT/9PT/+nT///kq27kyMjk///pvIzp6bzp///4dm3/yI7/06T/5Kv/6bz//8j//9P//+T//+n////uQLMRAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMOElEQVR4nO2dCXvbSBmAl7gp6cFyxE4JbjmcFojLtU4hhV1nId5CTL3FZ6r//0uY0UiyfEga5dNhSe/7tD6mY31T+/U3I1mj+cIBEPBF2Q2AaoNAIAKBQAQCgQgEAhEIBCLSCoRwsAECgQgEAhEIBCIQCEQgEIhAIBCBQCCiuQIdHR2V3YQ60FiBjo4wKAsQCETECTF/0mpdmrvHHy3qVwoEyoYYIZYXl8700bUzPbGrXzHwJxNihJiqtHN/dencnNrVhyaSIITKQvfvrr0nOukjEGyQIMT48cfl6+etR75DZCDYJF6I8fGtM3967Szf3FrVh8YRK8TYzzzL134KQiDYIE4InX8MCAQRxAih+y7FVHdjX9buOBBkQ4wQNy3NpTNutYJUhECwSWN/yoBsQCAQgUAgAoFABAKBCAQCEQgEIhAIRCAQiEAgEIFAIAKBQAQCgQgEAhEIBCIQCEQgEIhAIBCBQCACgUAEAoEIBAIRCAQiEAhEIBCIQCAQgUAgAoFABAKBCAQCEQgEIhAIRCAQiEAgEIFAIMJisZX7K66RCFFYLLZyc6L/JNeHJpK82Iq+RvT8WR5Xqme1nBqQvNiKliefC40jUA1IXmxlLVDWq/UgUA2wWGylrhkIf7MgebEVBIIYkhdbqe0gGoGyIHmxlfx24xGoBlgstrK8yOlAIgLVgDJ/ykCgGoBAIAKBQAQCgQgEAhEIBCIQCEQgEIhAoNqy6rcNZ3cRNWaDpG0k10CgWrN4OYz+x1U/SY/kGghUbxAoR5oj0OrtN7ofm6jurKdLR+rBwFl02+0X33e/Uj1dTz/WrnhVVn1dOnBrfIqPgEC1xhOorzWYdIZKGqXH6Nw80fll0VUPJq5e+q9XZdXXpZ0hGeiAwxeCL1DPu3FmZ3ert6ps0R14AvUc/2awrtL3CxDoYMMXgi+Q8mDW0Q+1OIqZ7rGMQAMnuAmquC9QNwh0wOELISyQt1M/0AOdsw/dfQL5VRCoAuELYScDOWtdojKQ/wIEOuzwhRASKJBhpg8szvZ1YUEVBKpA+EIICeTuYjmjztAkn3bPHVZvCBRUWQvUS4qAQLUmLJB7kEdnH3XfGSpNnJE+DhQWyK8SpKwRx4EONnxNQCAQgUAgAoFABAKBCAQCEQgEIhAIRCBQQ8NnBQI1NHxWIFAjwucXDIEaER6BcqBJ4QsW6LuomWgIVK3wR7tkHSIQYvVn/9GiGzmVEYGqFf7of9vkKFDfO59xFJoK616dVa+58vjjdv0MaMAnWHL4GIH0nK+2PjEoaRvuJI5o1kJ8ft8+d7d7HhQtL/TVNacne+vLacAnWHL4+Ay0eLW3p9kqthZInyfb+WNYyXHrxzoD3ZxG1JfSgE+w5PAFC+QloYD/uutk3L+7jqovpAGfYMnhLQRa/PIPeiJh25xp326fq8furFW/5Of2Ak10BtoYQJtlDp63HrkO1WyxlQaEj98LMwJ1BypxDJzJi0/6JOhZZ6iLQyWT+GHSWgjl3LljTtcP0ALpC9Yv37DUQRXD22Sgl0P3gddTqTv9zCuZvfhk3YVF74U5DoutVDS8jUBal67ZH9P3JgN5JZM0Au07DoRA1Q5vK5AZNutkFMpAijQZKMx3YYGmes2wLzkOVMXwlgLpEY+611NW9cRD5ZFXkmoMtA83A41bLLZS0fCWAjlmz0vthLd/836g7gZeie7Kfm2dgSZtc5WqWBCoSuGL/C3MuzpVkkEIVMnwBfwa//m93n2fZD0VOo4mfYIlhy9AIDMBfxbzQ/xm/Qxo0id4cOGzAoEy21zOZNvazECgzDb3i1xBoF0QCIFEIFC9BGoHcErrQzZ3yAIxKyMHECgLECizzTVcILow6eYOUKBCf8oIGIVPKbOo/2DqdSDmIAX6wTb5C6QyUdzvYYcs0M67lSkItJdtIWZx/dee+hIQqEyBzDljETMzotg9u2xLiEnShaVjBSq3E0Egi7ckWqD9KiXO8dme1hMz/Nmtv9PenfOXMuXABSr122P7luQqUMLwZ6f+bnsR6MAEitlMINDi1dftds89BfEBk8RCQiQNf7br72lvowWqahem1yucnd2ZyRipJ4mFzkjcmJUaBQJFhq+sQGZmmCtQ+klimR5IRKBKCmQc8VVJOUks058yEKhCAhkXVFrZFijdJDEEaqpAzkgLodec8wTyuq+Uk8QQqNYCxe7MjVTP1AvmpurpYA+YJIZAdRZo3TjRq+NAIAQSUSeByj2Sd9AC5QcCIZCIOglEF1YCCIRAIhAoM4HK7UHLAoGyEigtdbu4Qhb1ESgFCLQLAqUAgXZBoBQUIVABQymLayTeX9leIxGBUlCIQPnvzMUKYRZbuTnRf2zqI1AKGiCQWWxFXyPav140AmVH2QLpM1n908pmyas+RREnhFlsZe5JlFy/2dN60lK+QGf+VX4/v/+TzenMe0keA60F0p8hAmVF6QK9+roXXCv6Q7r5hSHSCJRYny4sBWXvhS1e/ecf3umqk3NnlDQhMAoEqrVAsRnobnYezMGYJV3fOYpkgRhE11agz393T5p3z5bvJ84p3Y/FcSB24+sqkLP4lTkZWvPAYbSFQMsLDiTWVCBn9Bczn8dM43kI/JTRaIH0vvzI5J4HDqMRqNYCxe6F6cQzOfu3N1Mn8QrP+0GgOgtUQDAEKkugQkEgBDpQEAiBRCAQAolAIAQSgUAIJAKBEEgEAiGQCARCIBEIhEAiEAiBRNRJoFLPyG4qNRIoLRiRBdkKVKkcgEBZwJqpIAKBQAQCgQgEAhEIBCIQCEQgEIhAIBCBQCACgUAEAoEIBAIRCAQiEAhEIBCIQCAQgUAgAoFAhIUQ8yet1uOP9vWtQaAaYCHE9CT0BIFgAwshbk7T1bcGgWpAshD3765DzxAINkgWYvn6eeuR65CeWYNAsEGyEPOn187yjdVSBylBoBpgJ4TlYispQaAagEAgIlmI6fGtM/+S40CwFwshxi3bxVZSgkA1gJ8yQAQCgQgEAhEIBCIQCEQgEIhAIBCBQCACgUAEAoEIBAIRCAQiEAhEIBCIQCAQgUAgAoFABAKBiDIFKhkEygIEAhEIBCIaLBBkAQKBCAQCEQgEIhAIRCAQiEAgEIFAIAKBQAQCgQgEAhGpBcqQoyw31ujwuahhKUSJsZt9PlBNfstFoIaGzwoEamj4rGBQDCIQCEQgEIhAIBCBQCCiEIH0opmKR9dJFddrcmQV+dltcGtN5q0IM3+i1+8zLZomvyMHT1EZKOIz3CrOXaD9zci7FRuxnuh1I9yI91e/O0l+wYGDQAW0YjPW3069iPNn36ZLjYdIoQLNf/Tb49vlhbt2y1T1aSfqscri65If5iaQ+txarVM3XvGt2GzRv/56a9o1Ptlcj7aSFCvQk0uVty+d8eOPy4tLPQTQxaGScdaDgrVAaugxPb7Vz4tvxVaLpiduu3SimwbrqVeVYgV6eh28dY7bVbgfqCnR72V+XZgX2o1XeCu2WnT/zo3ohlK+VptiBXK//2Z/TN+b775XMs5XIOOIr0qxrdhu0fwnWqMrd9e06sPo4gUy40Z3Kd/1d1+Ry3ffbFBtelugQlsRxo1z83vTFNOCSlO4QHqsoccBasB6o7776v3zSvIZfdy4/cSpb43ffRXcihDeYPD49sbknqoPowsXyDH7PDqB/+zqUt1deiW6E/lp9t/9G9VNnAbJT8croxVrTOIZH//TCzI9rvaePD9lgAgEAhEIBCIQCEQgEIhAIBCBQCCipgKt+m3D2V1EjdkgaRvJNaC2AmkWL4fR/7jqJ+mRXAMcBJLUAKcBAq3efqP7sYnqznq6dKQeDJxFt91+8X33K9XT9fRj7YpXZdXXpQO3xqdy/wdVoP4C9bUGk85QSaP0GJ2bJzq/LLrqwcTVS//1qqz6urQzJANZ0QCBet6NMzu7W71VZYvuwBOo5/g3g3WVvl+AQBY0QCDlwayjH2pxFDPdYxmBBk5wE1RxX6BuEMiKhgjk7dQP9EDn7EN3n0B+FQRKRUME6ng7ZIEzERnIfwEC2dIMgQIZZvrA4mxfFxZUQaBUNEMgdxfLGXWGJvm0e+6wekOgoMpaoF7Z/4Mq0BCB3IM8Ovuo+85QaeKM9HGgsEB+lSBljTgOZEGNBYIiQCAQgUAgAoFABAKBCAQCEQgEIhAIRCAQiEAgEPF/JpM04qgESFUAAAAASUVORK5CYII=" style="display: block; margin: auto;" /></p>
</div>
<div id="color-theming-olink_color_discrete-olink_color_gradient-olink_fill_discrete-olink_fill_gradient" class="section level2">
<h2>Color theming (olink_color_discrete, olink_color_gradient,
olink_fill_discrete, olink_fill_gradient)</h2>
<p>These functions sets a coherent coloring theme for the plots by
adding it to a ggplot object. It is mainly used for aesthetic
reasons.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>npx_data1 <span class="sc">%&gt;%</span> </span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>  <span class="fu">filter</span>(OlinkID <span class="sc">==</span> <span class="st">&#39;OID01216&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Treatment, <span class="at">y =</span> NPX, <span class="at">fill =</span> Treatment)) <span class="sc">+</span></span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>  <span class="fu">set_plot_theme</span>() <span class="sc">+</span></span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a>  <span class="fu">olink_fill_discrete</span>()</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAkAAAAEgCAMAAABrWDzDAAABCFBMVEUAAAAAx+EzMzNNTU1NTW5NTY5NbqtNjo5NjshuTU1uTW5uTY5ubk1ubo5ubqtujo5ujshuq6tuq8huq+Rzc3Nzc4xzc6RzjLxzpKRzpNOMc3OMc4yMc6SMjKSMpNOMvLyMvNOMvOmOTU2OTW6OTY6Obk2Ojo6Oq6uOyP+kc3Okc4ykjHOkpIykpKSkpNOkvLyk09Ok0/+rbk2rbm6rbo6rq26r5P+8jHO8jIy8jKS8vIy8vKS86f/Ijk3I5KvI/8jI/+TI///TpHPTpKTT6bzT/9PT/+nT///kq27kyMjk///pvIzp6bzp///+HwT/yI7/06T/5Kv/6bz//8j//9P//+T//+n///83/d8RAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAMSklEQVR4nO2dCXvayhVAX02cOkvTxeDUJF1w0tak28NpnfY9aGteWtPwUlZH//+fdK5GEotBi68WJJ3zJSyTQXcChzsjodF85QAo+KroBkC5QSBQgUCgAoFABQKBCgQCFUkFQjjYAIFABQKBCgQCFQgEKhAIVCAQqEAgUFFfgY6OjopuQhWorUBHRxiUBggEKsKEmD1pNC7t3eNPMeqXCgRKhxAhFheXzuTRtTM5iVe/ZOBPKoQIMTFp5+7q0hmcxqsPdSRCCJOF7t5fe08k6SMQbBAhxOjxp8Wb541HvkNkINgkXIjR8Y0ze3rtLN7exKoPtSNUiJGfeRZv/BSEQLBBmBCSfywIBHsIEUL6LsNEurEXlTsOBOkQIsSgIVw6o0YjSEUIBJvU9qcMSAcEAhUIBCoQCFQgEKhAIFCBQKACgUAFAoEKBAIVCAQqEAhUIBCoQCBQgUCgAoFABQKBCgQCFQgEKhAIVCAQqEAgUIFAoAKBQAUCgQoEAhUIBCoQCFQgEKhAIFCBQKACgUAFAoEKBAIVCAQqYiy2cnfFNRJhHzEWWxmcyJ/o+lBHohdbkWtEz55lcaV6VsupANGLrYg82VxoHIEqQPRiKyuB0l6tB4EqQIzFVqqagfA3DaIXW0EgCCF6sZXKDqIRKA2iF1vJbjcegSpAjMVWFhcZHUhEoApQ5E8ZCFQBEAhUIBCoQCBQgUCgAoFABQKBCgQCFQhUWZbdpuXsdk+NaS9qG9E1EKjSzF/19//jshulR3QNBKo2CJQh9RFo+e5b6cfGpjvrSOnQPOg583az+fL79temp+vIY3HFq7LsSmnPrfE5PAICVRpPoK5oMG71jTRGj+G5fSL5Zd42D8auXvLXq7LsSmmrTwY64PC54AvU8W6c6dnt8p0pm7d7nkAdx7/prap0/QIEOtjwueALZDyYtuShiGOYSo9lBeo5wU1QxX2BuUGgAw6fC+sCeTv1PRnonH1s7xLIr4JAJQifC/cykLPSZV8G8l+AQIcdPhfWBApkmMqBxemuLiyogkAlCJ8LawK5u1jOsNW3yafZcYfVGwIFVVYCdaIiIFClWRfIPcgj2cfct/pGE2cox4HWBfKrBClryHGggw1fERAIVCAQqEAgUIFAoAKBQAUCgQoEAhUIVNPwaYFANQ2fFghUi/DZBUOgWoRHoAyoU/icBfpu30w0BCpX+KP7pB0iEGL5J//RvL13KiMClSv80Q+2yVCgrnc+43BtKqx7dVZZc+Xxp+36KVCDT7Dg8CECyZyvppwYFLUNdxLHflZCfPnQPHe3ex4ULS7k6pqTk5319dTgEyw4fHgGmr/e2dNsFccWSM6Tbf1hXclR48eSgQane+prqcEnWHD4nAXyklDAf911Mu7eX++rr6QGn2DB4WMINP/F72UiYdOead9snpvH7qxVv+Tn8QUaSwbaGEDbZQ6eNx65DlVssZUahA/fC7MCtXsmcfSc8cvPchL0tNWX4rWScfgwaSWEce7csafrB4hAcsH6xVuWOihj+DgZ6FXffeD1VOZOnnkl05efY3dh+/fCHIfFVkoaPo5Aokvb7o/Jvc1AXsk4iUC7jgMhULnDxxXIDpslGa1lIEOSDLTOd+sCTWTNsBccBypj+JgCyYjH3MuUVZl4aDzyShKNgXbhZqBRg8VWSho+pkCO3fMyO+HNX3/ombueVyJd2a9iZ6Bx016lKhQEKlP4PH8L865OFWUQApUyfA6/xn/5ILvv47SnQodRp0+w4PA5CGQn4E9DfojfrJ8CdfoEDy58WiBQapvLmHRbmxoIdKCby3nzDwaBDnRzOW/+wSDQgW4u580/mDWBmgGc0lr85tLdPLMyMgCB0gCBDnRz6W4+l+NAdGGHtLlUNp/rTxkBw2bItV0PWKCCD8QcpED/2yZ7gUwmCvs97JAFuvdupQoC7WRbiGlY/7WjvgYESvK/e9CLQv4X9pyxPTMz9nH/7LItIcZRF5YOFajYTgSB7r8ogUC7VYqc47M9rSfq0vbhAt07fylVDlygQr89cd+STAWKGP7cq3+/vQh0YAKFbCYQaP76m2az456C+IBJYmtCRA1/tuvvaG+tBYp871Rkl4FkvcLp2a2djJF4ktjaGYkbs1L3gUB7w8d49xRkKJCdGeYKlHySWKoHEhEoOzIUyDriq5JwkliqP2UgUHakLpB1waSVbYGSTRJDoLoK5AxFCFlzzhPI674SThJDoEoLFLozNzQ9UyeYmyrTwR4wSQyBqixQSq8OA4EQSEWVBCr2SN5BC5QdCIRAKqokEF1YASAQAqlAoNQEKrYHLQoESkugpORqBHthCHSgwRAIgVQgUJUFymEoFeMaiXdXca+RiEAJyEWg7KOGCmEXWxmcyJ849REoATUQyC62IteI9q8XjUDpUbRAciarf1rZNHrVp32ECWEXW5l5EkXXr/e0nqQUL9CZf5XfLx/+GOd05p1Ej4FWAslniEBpUbhAr7/pBNeK/phsfuEaSQSKrE8XloCi98Lmr//zd+901fG5M4yaELgPBKq0QCElctrqeTAHYxp1fed9RAvEILqyAn35m3vSvHu2fDdyTuluYhwHYje+qgI581/ak6GFBw6jYwi0uOBAYkUFcoZ/tvN57DSeh8BPGbUWSPblhzb3PHAYjUCVFih0L0wSz/js395MncgrPO8GgaosUA7BEKgogXIFgRDoQEEgBFKBQAikAoEQSAUCIZAKBEIgFQiEQCoQCIFUIBACqUAgBFJRJYEKPSO7rlRIoKRgRBqkK1CpcgACpQFrpoIKBAIVCAQqEAhUIBCoQCBQgUCgAoFABQKBCgQCFQgEKhAIVCAQqEAgUIFAoAKBQAUCgQoEAhUxhJg9aTQef4pfPzYIVAFiCDE5WXuCQLBBDCEGp8nqxwaBKkC0EHfvr9eeIRBsEC3E4s3zxiPXIZlZg0CwQbQQs6fXzuJtrKUOEoJAFSCeEDEXW0kIAlUABAIV0UJMjm+c2QuOA8FOYggxasRdbCUhCFQB+CkDVCAQqEAgUIFAoAKBQAUCgQoEAhUIBCoQCFQgEKhAIFCBQKACgUAFAoEKBAIVCAQqEAhUIBCoKFKggkGgNEAgUIFAoKLGAkEaIBCoQCBQgUCgAoFABQKBCgQCFQgEKhAIVCAQqEAgUJFYoBQ5SnNjtQ6fiRoxhSgwdr3PB6rIb7kIVNPwaYFANQ2fFgyKQQUCgQoEAhUIBCoQCFTkIpAsmml4dB1VcbUmR1qRn90Et7FJvRXrzJ7I+n22RZPod+TgySsD7fkMt4ozF2h3M7JuxUasJ7JuhBvx7uq3J9EvOHAQKIdWbMb666kXcfbsH8lS4yGSq0CzH/3m+GZx4a7dMjF92ol5bLL4quSHmQlkPrdG49SNl38rNlv0r7/c2HaNTjbXoy0l+Qr05NLk7Utn9PjT4uJShgBSvFYySntQsBLIDD0mxzfyPP9WbLVocuK2SxLdJFhPvazkK9DT6+Ctc9yuwv1AbYm8l9l1YV5oN17urdhq0d17N6IbyvhabvIVyP3+2/0xubfffa9klK1A1hFflXxbsd2i2U9Eoyt317Tsw+j8BbLjRncp39V335DJd99u0Gx6W6BcW7GOG2fwO9sU24JSk7tAMtaQcYAZsA7Md9+8f15JNqOPgdtPnPrW+N1Xzq1YwxsMHt8MbO4p+zA6d4Ecu88jCfxnV5fm7tIrkU7kp+l/9wemmzgNkp/EK6IVK2ziGR3/0wsyOS73njw/ZYAKBAIVCAQqEAhUIBCoQCBQgUCgoqICLbtNy9ntnhrTXtQ2omtAZQUS5q/6+/9x2Y3SI7oGOAikqQFODQRavvtW+rGx6c46Ujo0D3rOvN1svvy+/bXp6TryWFzxqiy7Utpza3wu9n9QBqovUFc0GLf6Rhqjx/DcPpH8Mm+bB2NXL/nrVVl2pbTVJwPFogYCdbwbZ3p2u3xnyubtnidQx/FveqsqXb8AgWJQA4GMB9OWPBRxDFPpsaxAPSe4Caq4LzA3CBSLmgjk7dT3ZKBz9rG9SyC/CgIloiYCtbwdssCZPRnIfwECxaUeAgUyTOXA4nRXFxZUQaBE1EMgdxfLGbb6Nvk0O+6wekOgoMpKoE7R/4MyUBOB3IM8kn3MfatvNHGGchxoXSC/SpCyhhwHikGFBYI8QCBQgUCgAoFABQKBCgQCFQgEKhAIVCAQqEAgUPF/jpYbwkBrKCcAAAAASUVORK5CYII=" style="display: block; margin: auto;" /></p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
